---
title: "Diversity indices"
author: "Marion Boisseaux"
date: "2023-05-03"
output: html_document
---

Diversity indices

```{r main libraries, results='hide'}
set.seed(123456)

library(tidyr)
library(tidyverse)
library(labdsv)
library(dplyr)
library(FactoMineR)
library(car)
library(missMDA)
library(metabaR)
library(ggplot2)
library(corrplot)
library(factoextra)
library(MASS)
library(cowplot)
library(MuMIn)
library(multcomp)

library(ade4)
library(hillR)
library(adespatial)
library(reldist)
library(bipartite)
library(vegan)
library(scales)
library(phyloseq)
library(reshape2)
#devtools::install_github("kassambara/ggpubr")
library(ggpubr)
library(purrr)
library(stringr)
library(BiocManager)
#BiocManager::install("microbiome")
library(microbiome)  
library(rstatix)

# Enable the r-universe repo
# options(repos = c(
#     fawda123 = 'https://fawda123.r-universe.dev',
#     CRAN = 'https://cloud.r-project.org'))

# Install ggord
#install.packages('ggord')
library(ggord) #simple package for creating ordination plots with ggplot2. Marcus W. Beck (2017). ggord: Ordination Plots with ggplot2. R package version 1.0.0. https://zenodo.org/badge/latestdoi/35334615
library(phyloseq)
setwd("E:/DRYER_greenhouse/")
```


# Endophyte analyses

Bacteria data - 16S
```{r}
#microbio 16S clean
load("/DRYER_greenhouse/results/Metabarlist_serre_16S_AMVG51-230505_clean.Rdata")

# leaf <- AMVG51_230505_16S_clean
# leaf_data <- leaf$reads
# leaf_data <- as.data.frame(leaf_data)

# essai <-  leaf_data %>%
#   rownames_to_column(var = "group") %>%  
#   pivot_longer(cols = -group)
# 
# min_n_seqs <- essai %>%
#   group_by(group)%>%
#   summarize(nseq=sum(value))%>%
#   summarize(min = min(nseq), max = max(nseq), median= median(nseq), mean= mean(nseq)) %>%
#   pull(min)
#  # pull(min) # 851
# 
# essai_df <- essai %>% pivot_wider(names_from = "name", values_from = "value") %>%
#   as.data.frame()
# 
# 
# rownames(essai_df) <- essai_df$group
# essai <- essai_df[,-1]
# 
# 
# rarefy(essai, min_n_seqs) %>%
#   as_tibble(rownames="Group") %>%
#   dplyr::select(Group, vegan=value)
# 
# 
# leaf_rarefy <- rrarefy(essai, min_n_seqs) %>%
#   as_tibble(rownames = "Group") %>%
#               pivot_longer(-Group)
# 
# bacteria_rarefy <- subset_metabarlist(metabarlist = leaf, reads = "samples", indices = leaf$reads > 304)
# 
# leaf$samples$nb_reads <- rowSums(leaf$reads) 
# leaf$pcrs[which(leaf$samples$nb_reads < 1000)]

```

```{r}
bacteria <- AMVG51_230505_16S_clean
rownames(bacteria$samples) <- bacteria$samples$sample_id
bacteria$samples <-bacteria$samples %>%
   mutate(Period = case_when(
     Treatment == "D1" &  Time == "t21" ~ "Drought D1",
      Treatment == "D1" &  Time == "t51" ~ "Recovery D1",
      Treatment == "D2" &  Time == "t27" ~ "Drought D2",
      Treatment == "D2" &  Time == "t57" ~ "Recovery D2",
      Treatment == "D3" &  Time == "t71" ~ "Drought D3",
      Treatment == "D3" &  Time == "t101" ~ "Recovery D3",
      TRUE                      ~ "Control"
    )
  )

drought <- subset_metabarlist(metabarlist = bacteria, table = "samples", indices = bacteria$samples$Time == "Control")
rownames(drought$samples) <- drought$samples$sample_id

data.offset <- drought$reads
data.offset <- data.offset + 1 # apply offset
sum(which(data.offset == 0)) # how many zeroes are there after offset

low.count.removal <- function(
                        data, # OTU count df of size n (sample) x p (OTU)
                        percent=0.01 # cutoff chosen
                        ) 
  {
    keep.otu = which(colSums(data)*100/(sum(colSums(data))) > percent)
    data.filter = data[,keep.otu]
    return(list(data.filter = data.filter, keep.otu = keep.otu))
}

dim(data.offset) # check samples are in rows
# call the function then apply on the offset data
result.filter <- low.count.removal(data.offset, percent=0.05)
data.filter <- result.filter$data.filter

# check the number of variables kept after filtering
# in this particular case we had already filtered the data so no was change made, 
# but from now on we will work with 'data.filter'
length(result.filter$keep.otu) 
lib.size <- apply(data.filter, 1, sum) # determine total count for each sample
barplot(lib.size) # and plot as bar plot
maximum.lib.size <- 2000

data.filter <- data.filter[-which(lib.size > maximum.lib.size),]
# undergo PCA on CLR transformed data
pca.result <- pca(data.filter, logratio = 'CLR') 

data.filter_data <- as.data.frame(data.filter) 
data.filter_data$sample_id <- rownames(data.filter_data)
data.filter_data <- left_join(data.filter_data, bacteria$samples)


plotIndiv(pca.result,  # plot samples
          group = data.filter_data$Species, 
          title = 'bacteria drought and recovery PCA Comps 1&2',
            ellipse = TRUE,  # plot using the ellipses 
          legend = T, ind.names = FALSE)
```


# - Bacteria 16S Analyses 

## alpha diversity

```{r Hill numbers diversity}

#Hill numbers, using HillR package
Shannon_bac <-hill_taxa(AMVG51_230505_16S_clean$reads,q=1)
#equivalent of Shannon entropy

df <- Shannon_bac %>% 
  enframe() %>%
  rename(Shannon_bac = value, 
         sample_id = name)

AMVG51_230505_16S_clean$samples <- left_join(AMVG51_230505_16S_clean$samples, df)
data <- AMVG51_230505_16S_clean$samples

write.csv(x = data, file = "AMVG51_230505_16S_clean_alpha.csv")


```

##Plots

```{r}

##overall plot
data <- AMVG51_230505_16S_clean$samples
overall <- data %>%
 filter(Time %in% c("t21", "t27", "t71")) %>%
# filter(Treatment != "C0")  %>%
 ggplot() +
  aes(x=Treatment, y = Shannon_bac, color = Treatment) +
  geom_boxplot()+
   geom_jitter(shape = "circle", alpha = 0.5) +
  scale_color_manual(values = c( 
      D1 = "#8CB425", D2 = "#ECE320", D3 = "#DF5220", C0 = "#219ebc")) + 
  theme_minimal(base_size = 14) +
   ylab("Alpha diversity")+
  xlab("")+
  stat_compare_means(method = "kruskal.test", label.y = c(90))+
  theme(legend.position = "bottom", axis.text.x = element_blank())+
   ggtitle(expression(italic("Bacteria"))) 
overall

#increasing drought treatments
data <- AMVG51_230505_16S_clean$samples
ef <- data %>%
 filter(Species == "Epe.fal") %>%
 filter(Time %in% c("t21", "t27", "t71")) %>%
# filter(Treatment != "C0")  %>%
 ggplot() +
  aes(x=Treatment, y = Shannon_bac, color = Treatment) +
  geom_boxplot()+
   geom_jitter(shape = "circle", alpha = 0.5) +
  scale_color_manual(values = c( 
      D1 = "#8CB425", D2 = "#ECE320", D3 = "#DF5220", C0 = "#219ebc")) + 
  theme_minimal(base_size = 14) +
   ylab("Alpha diversity")+
  xlab("")+
  stat_compare_means(method = "kruskal.test", label.y = c(50))+
  ggtitle(expression(italic("E. falcata"))) +
  theme(legend.position = "none", axis.text.x = element_blank())+
  ylim(0,60)
ef

ih <- data %>%
 filter(Species == "Iry.hos") %>%
filter(Time %in% c("t21", "t27", "t71")) %>%
 ggplot() +
  aes(x=Treatment, y = Shannon_bac, color = Treatment) +
  geom_boxplot()+
   geom_jitter(shape = "circle", alpha = 0.5) +
  scale_color_manual(values = c( 
      D1 = "#8CB425", D2 = "#ECE320", D3 = "#DF5220", C0 = "#219ebc")) + 
  theme_minimal(base_size = 14) +
   ylab("")+
  xlab("")+
  stat_compare_means(method = "kruskal.test", label.y = c(90), label.x = 1.5)+
  ggtitle(expression(italic("I. hostmanii"))) +
  theme(legend.position = "none", axis.text.x = element_blank())+
  ylim(0,100)
ih

jc <- data %>%
 filter(Species == "Jac.cop") %>%
filter(Time %in% c("t21", "t27", "t71")) %>%
 #filter(Treatment != "C0")  %>%
  ggplot() +
  aes(x=Treatment, y = Shannon_bac, color = Treatment) +
  geom_boxplot()+
   geom_jitter(shape = "circle", alpha = 0.5) +
  scale_color_manual(values = c( 
      D1 = "#8CB425", D2 = "#ECE320", D3 = "#DF5220", C0 = "#219ebc")) + 
  theme_minimal(base_size = 14) +
   ylab("")+
  xlab("")+
  stat_compare_means(method = "kruskal.test", label.y = c(115), label.x = 1.5)+
  ggtitle(expression(italic("J. copaia"))) +
  theme(legend.position = "none", axis.text.x = element_blank())+
  ylim(0,130)
jc

po <- data %>%
 filter(Species == "Pte.off") %>%
 filter(Time %in% c("t21", "t27", "t71")) %>%
# filter(Treatment != "C0")  %>%
 ggplot() +
  aes(x=Treatment, y = Shannon_bac, color = Treatment) +
  geom_boxplot()+
   geom_jitter(shape = "circle", alpha = 0.5) +
  scale_color_manual(values = c( 
      D1 = "#8CB425", D2 = "#ECE320", D3 = "#DF5220", C0 = "#219ebc")) + 
  theme_minimal(base_size = 14) +
   ylab("")+
  xlab("")+
  stat_compare_means(method = "kruskal.test", label.y = c(80), label.x = 1.5)+
   ggtitle(expression(italic("P. officinalis"))) +
  theme(legend.position = "none", axis.text.x = element_blank())+
  ylim(0,90)
po

sg <- data %>%
 filter(Species == "Sym.off") %>%
 filter(Time %in% c("t21", "t27", "t71")) %>%
# filter(Treatment != "C0")  %>%
 ggplot() +
  aes(x=Treatment, y = Shannon_bac, color = Treatment) +
  geom_boxplot()+
   geom_jitter(shape = "circle", alpha = 0.5) +
  scale_color_manual(values = c( 
      D1 = "#8CB425", D2 = "#ECE320", D3 = "#DF5220", C0 = "#219ebc")) + 
  theme_minimal(base_size = 14) +
   ylab("Alpha diversity")+
  xlab("")+
  stat_compare_means(method = "kruskal.test", label.y = c(120), label.x = 1.5)+
  ggtitle(expression(italic("S. globulifera"))) +
  theme(legend.position = "none")+
  ylim(0,130)
sg

tm <- data %>%
 filter(Species == "Tac.mel") %>%
filter(Time %in% c("t21", "t27", "t71")) %>%
# filter(Treatment != "C0")  %>%
 ggplot() +
  aes(x=Treatment, y = Shannon_bac, color = Treatment) +
  geom_boxplot()+
   geom_jitter(shape = "circle", alpha = 0.5) +
  scale_color_manual(values = c( 
      D1 = "#8CB425", D2 = "#ECE320", D3 = "#DF5220", C0 = "#219ebc")) + 
  theme_minimal(base_size = 14) +
   ylab("")+
  xlab("")+
  stat_compare_means(method = "kruskal.test", label.y = c(65), label.x = 1.5) +
  ggtitle(expression(italic("T. melinonii"))) +
  theme(legend.position = "none")+
  ylim(0,70)
tm

vs <- data %>%
 filter(Species == "Vir.sur") %>%
 filter(Time %in% c("t21", "t27", "t71")) %>%
#filter(Treatment != "C0")  %>%
 ggplot() +
  aes(x=Treatment, y = Shannon_bac, color = Treatment) +
  geom_boxplot()+
   geom_jitter(shape = "circle", alpha = 0.5) +
  scale_color_manual(values = c( 
      D1 = "#8CB425", D2 = "#ECE320", D3 = "#DF5220", C0 = "#219ebc")) + 
  theme_minimal(base_size = 14) +
  xlab("")+
  ylab("")+
  stat_compare_means(method = "kruskal.test", label.y = c(60), label.x = 1.5)+
  ggtitle(expression(italic("V. surinamensis"))) +
  theme(legend.position = "none")+
 # theme(legend.position = "bottom", legend.box = "vertical", legend.direction = "vertical")+
  ylim(0,70)
vs

legend <- get_legend(vs)

alpha_bac_drought <- ggarrange(ef, ih, jc, po, sg, tm, vs, as_ggplot(legend), nrow = 2, ncol=4, widths = c(0.2,0.2,0.2,0.2))

alpha_bac_drought <- annotate_figure(alpha_bac_drought, top = text_grob("A. Bacteria", color = "black", face = "bold", size = 20, just = "right" , hjust = 3.2, vjust = 0.5))

ggsave(filename = "./results/microbio/alpha_bac_drought_withC0.jpeg", plot = alpha_bac_drought, width = 10, height = 6)

############################################################
#after drought, during the recovery phase

#prepare data
data <- AMVG51_230505_16S_clean$samples
data <- data %>%  filter(Treatment != "C0") %>% filter(!is.na(Species)) %>%
  mutate(Period = ifelse(Time %in% c("t21","t27", "t71"), "Drought", 
                  ifelse(Time %in% c("t51", "t57","t101"), "Recovery", NA)))
data$Period <- as.factor(data$Period)
data$Treatment <- as.factor(data$Treatment)

#epefal
ef <- data %>% filter(Species == "Epe.fal")
 ef_plot <- ggboxplot(
  ef, x= "Period" , y = "Shannon", fill = "Period", facet.by = "Treatment", alpha =0.7) +           geom_jitter(shape = "circle", alpha = 0.5) +
  scale_fill_manual(values = c(Drought = "#ff9e00", Recovery = "#219ebc"))+
  theme_classic() +
  theme(axis.text.x = element_blank(), axis.title.x = element_blank())+
  ylab("Alpha diversity")+
  theme(legend.position = "none")
ef_plot
stat.test <- data %>%
  filter(Species == "Epe.fal") %>%
  filter(Treatment != "D3") %>%
  group_by(Treatment) %>%
  t_test(Shannon ~ Period) %>%
  adjust_pvalue(method = "bonferroni") %>%
  add_significance()
stat.test 
stat.test <- stat.test %>% add_xy_position(x = "Period") # Make facet and add p-values
ef_plot <- ef_plot + stat_pvalue_manual(stat.test)
ef_plot <- annotate_figure(ef_plot, left = text_grob("E. falcata       ", color = "black", face = "italic", size = 20))

#iryhos
ih <- data %>% filter(Species == "Iry.hos") 
ih_plot <- ggboxplot(
  ih, x= "Period" , y = "Shannon", fill = "Period", facet.by = "Treatment", alpha =0.7) + 
  geom_jitter(shape = "circle", alpha = 0.5) +
  scale_fill_manual(values = c(Drought = "#ff9e00", Recovery = "#219ebc"))+
  theme_classic() +
  theme(axis.text.x = element_blank(), axis.title.x = element_blank())+
  ylab("Alpha diversity")+
  theme(legend.position = "none")
ih_plot
stat.test <- ih %>%
  group_by(Species, Treatment) %>%
  t_test(Shannon ~ Period) %>%
  adjust_pvalue(method = "bonferroni") %>%
  add_significance()
stat.test 
stat.test <- stat.test %>% add_xy_position(x = "Period") 
ih_plot <- ih_plot + stat_pvalue_manual(stat.test)
ih_plot <- annotate_figure(ih_plot, right = text_grob("I. hostmanii", color = "black", face = "italic", size = 20))

#jaccop
jc <- data %>% filter(Species == "Jac.cop") 
 jc_plot <- ggboxplot(
   jc, x= "Period" , y = "Shannon", fill = "Period", facet.by = "Treatment", alpha =0.7) + 
   geom_jitter(shape = "circle", alpha = 0.5) +
   scale_fill_manual(values = c(Drought = "#ff9e00", Recovery = "#219ebc"))+
  theme_classic() +
  theme(axis.text.x = element_blank(), axis.title.x = element_blank())+
  ylab("Alpha diversity")+
  theme(legend.position = "none")
jc_plot
stat.test <- jc %>%
  group_by(Species, Treatment) %>%
  t_test(Shannon ~ Period) %>%
  adjust_pvalue(method = "bonferroni") %>%
  add_significance()
stat.test 
stat.test <- stat.test %>% add_xy_position(x = "Period")
jc_plot <- jc_plot + stat_pvalue_manual(stat.test)
jc_plot <- annotate_figure(jc_plot, left = text_grob("J. copaia        ", color = "black", face = "italic", size = 20))

#pteoff
po <- data %>% filter(Species == "Pte.off") 
 po_plot <- ggboxplot(
   po, x= "Period" , y = "Shannon", fill = "Period", facet.by = "Treatment", alpha =0.7) + 
  geom_jitter(shape = "circle", alpha = 0.5) +
  scale_fill_manual(values = c(Drought = "#ff9e00", Recovery = "#219ebc"))+
  theme_classic() +
  theme(axis.text.x = element_blank(), axis.title.x = element_blank())+
  ylab("Alpha diversity")+
  theme(legend.position = "none")
po_plot
stat.test <- po %>%
  group_by(Species, Treatment) %>%
  t_test(Shannon ~ Period) %>%
  adjust_pvalue(method = "bonferroni") %>%
  add_significance()
stat.test 
stat.test <- stat.test %>% add_xy_position(x = "Period")
po_plot <- po_plot + stat_pvalue_manual(stat.test)
po_plot <- annotate_figure(po_plot, right = text_grob("P. officinalis", color = "black", face = "italic", size = 20))

#symglo
sg <- data %>% filter(Species == "Sym.off") 
 sg_plot <- ggboxplot(
   sg, x= "Period" , y = "Shannon", fill = "Period", facet.by = "Treatment", alpha =0.7) + 
  geom_jitter(shape = "circle", alpha = 0.5) +
  scale_fill_manual(values = c(Drought = "#ff9e00", Recovery = "#219ebc"))+
  theme_classic() +
  theme(axis.text.x = element_blank(), axis.title.x = element_blank())+
  ylab("Alpha diversity")+
  theme(legend.position = "none")
sg_plot
stat.test <- sg %>%
  group_by(Species, Treatment) %>%
  t_test(Shannon ~ Period) %>%
  adjust_pvalue(method = "bonferroni") %>%
  add_significance()
stat.test 
stat.test <- stat.test %>% add_xy_position(x = "Period")
sg_plot <- sg_plot + stat_pvalue_manual(stat.test)
sg_plot <- annotate_figure(sg_plot, left = text_grob("S. globulifera  ", color = "black", face = "italic", size = 20))

#tacmel
tm <- data %>% filter(Species == "Tac.mel") 
tm_plot <- ggboxplot(
  tm, x= "Period" , y = "Shannon", fill = "Period", facet.by = "Treatment", alpha =0.7) + 
  geom_jitter(shape = "circle", alpha = 0.5) +
  scale_fill_manual(values = c(Drought = "#ff9e00", Recovery = "#219ebc"))+
  theme_classic() +
  theme(axis.text.x = element_blank(), axis.title.x = element_blank())+
  ylab("Alpha diversity")+
  theme(legend.position = "none")
tm_plot
stat.test <- tm %>%
  group_by(Species, Treatment) %>%
  t_test(Shannon ~ Period) %>%
  adjust_pvalue(method = "bonferroni") %>%
  add_significance()
stat.test 
stat.test <- stat.test %>% add_xy_position(x = "Period")
tm_plot <- tm_plot + stat_pvalue_manual(stat.test)
tm_plot <- annotate_figure(tm_plot, right = text_grob("T. melinonii", color = "black", face = "italic", size = 20))

#virsur
vs <- data %>% filter(Species == "Vir.sur") 
vs_plot <- ggboxplot(
  vs, x= "Period" , y = "Shannon", fill = "Period", facet.by = "Treatment", alpha =0.7) + 
  geom_jitter(shape = "circle", alpha = 0.5) +
  scale_fill_manual(values = c(Drought = "#ff9e00", Recovery = "#219ebc"))+
  theme_classic() +
  theme(axis.text.x = element_blank(), axis.title.x = element_blank())+
  ylab("Alpha diversity")+
  theme(legend.position = "none", legend.text = element_text(size = 16),
    legend.title = element_text(size = 20))
vs_plot
stat.test <- vs %>%
  filter(!(Treatment == "D3")) %>%
  group_by(Species, Treatment) %>%
  t_test(Shannon ~ Period) %>%
  adjust_pvalue(method = "bonferroni") %>%
  add_significance()
stat.test 
stat.test <- stat.test %>% add_xy_position(x = "Period") 
vs_plot <- vs_plot + stat_pvalue_manual(stat.test)
vs_plot <- annotate_figure(vs_plot, left = text_grob("V. surinamensis", color = "black", face = "italic", size = 20))

#get legend
legend <- get_legend(vs_plot) #get legend work with no complicated plots fso just grab the legend when plot is simpler
as_ggplot(legend)

#plot all
alpha_recov_bac <- ggarrange(ef_plot, ih_plot, jc_plot, po_plot, sg_plot, tm_plot, vs_plot, as_ggplot(legend), ncol=2, nrow = 4)
alpha_recov_bac
alpha_recov_bac <- annotate_figure(alpha_recov_bac, top = text_grob("Bacterial alpha diversity before and after drought", color = "black", face = "bold", size = 24))
ggsave(filename = "./results/microbio/alpha_bac_recov.jpeg", plot = alpha_recov_bac, width = 12, height = 10)

```


##Resistance of alpha

```{r}
resistance_D1 <- function(data) {
  resistance_df <- data.frame(Species = character(),
                              Treatment = character(),
                              Resistance = numeric(),
                              stringsAsFactors = FALSE)
  data <- data %>% filter(Time == "t21")
  unique_species <- unique(data$Species)
  
  for (species in unique_species) {
    species_data <- data[data$Species == species, ]
    
    CT <- mean(species_data$Shannon[species_data$Treatment == "C0"])
    treatments <- "D1"
    
    for (treatment in treatments) {
      DT <- abs(CT - mean(species_data$Shannon[species_data$Treatment == treatment]))
      
      if (length(DT) > 0) {
        Resistance <- 1 - ((2 * abs(DT)) / (CT + abs(DT)))
        
        species_resistance <- data.frame(Species = species,
                                         Treatment = treatment,
                                         Resistance = Resistance,
                                         stringsAsFactors = FALSE)
        
        resistance_df <- rbind(resistance_df, species_resistance)
      }
    }
  }
  
  resistance_df
}
```

```{r}
resistance_D2 <- function(data) {
  resistance_df <- data.frame(Species = character(),
                              
                              Treatment = character(),
                              Resistance = numeric(),
                              stringsAsFactors = FALSE)
  data <- data %>% filter(Time == "t27")
  unique_species <- unique(data$Species)
  
  for (species in unique_species) {
    species_data <- data[data$Species == species, ]
    
    CT <- mean(species_data$Shannon[species_data$Treatment == "C0"])
     treatments <- "D2"
    
    for (treatment in treatments) {
      DT <- abs(CT - mean(species_data$Shannon[species_data$Treatment == treatment]))
      
      if (length(DT) > 0) {
        Resistance <- 1 - ((2 * abs(DT)) / (CT + abs(DT)))
        
        species_resistance <- data.frame(Species = species,
                                         Treatment = treatment,
                                         Resistance = Resistance,
                                         stringsAsFactors = FALSE)
        
        resistance_df <- rbind(resistance_df, species_resistance)
      }
    }
  }
  
  resistance_df
}

```



```{r}
resistance_D3 <- function(data) {
  resistance_df <- data.frame(Species = character(),
                              
                              Treatment = character(),
                              Resistance = numeric(),
                              stringsAsFactors = FALSE)
  data <- data %>% filter(Time == "t71")
  unique_species <- unique(data$Species)
  
  for (species in unique_species) {
    species_data <- data[data$Species == species, ]
    
    CT <- mean(species_data$Shannon[species_data$Treatment == "C0"])
      treatments <- "D3"
    
    for (treatment in treatments) {
      DT <- abs(CT - mean(species_data$Shannon[species_data$Treatment == treatment]))
      
      if (length(DT) > 0) {
        Resistance <- 1 - ((2 * abs(DT)) / (CT + abs(DT)))
        
        species_resistance <- data.frame(Species = species,
                                         Treatment = treatment,
                                         Resistance = Resistance,
                                         stringsAsFactors = FALSE)
        
        resistance_df <- rbind(resistance_df, species_resistance)
      }
    }
  }
  
  resistance_df
}

```

plot and format with reference C0
```{r}
bacteria1 <- resistance_D1(data)
bacteria2 <- resistance_D2(data)
bacteria3 <- resistance_D3(data)

bacteria <- rbind(bacteria1, bacteria2, bacteria3)
bacteria

#plot
#for the labels
bacteria$facet_label <- NA
bacteria$facet_label[bacteria$Species == "Epe.fal"] <- "E. falcata"
bacteria$facet_label[bacteria$Species == "Iry.hos"] <- "I. hostmanii"
bacteria$facet_label[bacteria$Species == "Jac.cop"] <- "J. copaia"
bacteria$facet_label[bacteria$Species == "Pte.off"] <- "P. officinalis"
bacteria$facet_label[bacteria$Species == "Sym.off"] <- "S. globulifera"
bacteria$facet_label[bacteria$Species == "Tac.mel"] <- "T. melinonii"
bacteria$facet_label[bacteria$Species == "Vir.sur"] <- "V. surinamensis"

ef <- bacteria %>%
  filter(Species == "Epe.fal") %>%
  ggplot() +
 aes(x = Treatment, y = Resistance,  colour = Treatment) +
 ylab("Resistance")+
 geom_point(shape = "circle", size = 4L,  alpha = 0.5) +
 scale_color_manual(values = c( 
      D2 = "#ECE320", D3 = "#DF5220")) +
 labs(title = "E. falcata") +
 xlab(" ")+
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
 theme_minimal(base_size =20)+
 theme(title = element_text(face = "italic"), plot.title =   element_text(hjust = 0.5))+
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
 theme(legend.position = "none")

ih <-  bacteria %>%
  filter(Species == "Iry.hos") %>%
  ggplot() +
 aes(x = Treatment, y = Resistance,  colour = Treatment) +
 ylab("Resistance")+
 geom_point(shape = "circle", size = 4L,  alpha = 0.5) +
 scale_color_manual(values = c( 
      D2 = "#ECE320", D3 = "#DF5220")) +
 labs(title = "I. hostmanii") +
 xlab(" ")+
  ylab(" ")+
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
 theme_minimal(base_size =20)+
 theme(title = element_text(face = "italic"), plot.title =   element_text(hjust = 0.5))+
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
   theme(legend.position = "none")

jc <-  bacteria %>%
  filter(Species == "Jac.cop") %>%
  ggplot() +
 aes(x = Treatment, y = Resistance,  colour = Treatment) +
 ylab("Resistance")+
 geom_point(shape = "circle", size = 4L,  alpha = 0.5) +
 scale_color_manual(values = c( 
      D2 = "#ECE320", D3 = "#DF5220")) +
 labs(title = "J. copaia") +
 xlab(" ")+
  ylab(" ")+
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
 theme_minimal(base_size =20)+
 theme(title = element_text(face = "italic"), plot.title =   element_text(hjust = 0.5))+
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
   theme(legend.position = "none")

po <- bacteria %>%
  filter(Species == "Pte.off") %>%
  ggplot() +
 aes(x = Treatment, y = Resistance,  colour = Treatment) +
 ylab("Resistance")+
 geom_point(shape = "circle", size = 4L,  alpha = 0.5) +
 scale_color_manual(values = c( 
      D2 = "#ECE320", D3 = "#DF5220")) +
 labs(title = "P. officinalis") +
 xlab(" ")+
  ylab(" ")+
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
 theme_minimal(base_size =20)+
 theme(title = element_text(face = "italic"), plot.title =   element_text(hjust = 0.5))+
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
   theme(legend.position = "none")

sg <- bacteria %>%
  filter(Species == "Sym.off") %>%
  ggplot() +
 aes(x = Treatment, y = Resistance,  colour = Treatment) +
 ylab("Resistance")+
 geom_point(shape = "circle", size = 4L,  alpha = 0.5) +
 scale_color_manual(values = c( 
      D2 = "#ECE320", D3 = "#DF5220")) +
 labs(title = "S. globulifera") +
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
 theme_minimal(base_size =20)+
 theme(title = element_text(face = "italic"), plot.title =   element_text(hjust = 0.5))+
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
   theme(legend.position = "none")


tm <- bacteria %>%
  filter(Species == "Tac.mel") %>%
  ggplot() +
 aes(x = Treatment, y = Resistance,  colour = Treatment) +
 ylab("Resistance")+
 geom_point(shape = "circle", size = 4L,  alpha = 0.5) +
 scale_color_manual(values = c( 
      D2 = "#ECE320", D3 = "#DF5220")) +
 labs(title = "T. melinonii") +
  ylab(" ")+
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
 theme_minimal(base_size =20)+
 theme(title = element_text(face = "italic"), plot.title =   element_text(hjust = 0.5))+
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
   theme(legend.position = "none")

vs <- bacteria %>%
  filter(Species == "Vir.sur") %>%
  ggplot() +
 aes(x = Treatment, y = Resistance,  colour = Treatment) +
 ylab("Resistance")+
 geom_point(shape = "circle", size = 4L,  alpha = 0.5) +
 scale_color_manual(values = c( 
      D2 = "#ECE320", D3 = "#DF5220")) +
 labs(title = "V. surinamensis") +
  ylab(" ")+
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
 theme_minimal(base_size =20)+
 theme(title = element_text(face = "italic"), plot.title =   element_text(hjust = 0.5))+
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
   theme(legend.position = "none")

#get legend
legend <- get_legend(vs)
legend <- as_ggplot(legend)


resistance_bac <- ggarrange(ef, ih, jc, po, sg, tm, vs, legend, nrow = 2, ncol = 4)
resistance_bac
ggsave(filename = "./results/microbio/resistance_bac.jpeg", plot = resistance_bac, width = 12, height = 10)
```



```{r}
#compare with D1
resistance <- function(data) {
  resistance_df <- data.frame(Species = character(),
                              Treatment = character(),
                              Resistance = numeric(),
                              stringsAsFactors = FALSE)
  
  unique_species <- unique(data$Species)
  
  data <- data %>% filter(Treatment != "C0")
  
  for (species in unique_species) {
    species_data <- data[data$Species == species, ]
    
    CT <- mean(species_data$Shannon[species_data$Treatment == "D1"])
    treatments <- unique(species_data$Treatment[species_data$Treatment != "D1"])
    
    for (treatment in treatments) {
      DT <- abs(CT - mean(species_data$Shannon[species_data$Treatment == treatment]))
      
      if (length(DT) > 0) {
        Resistance <- 1 - ((2 * abs(DT)) / (CT + abs(DT)))
        
        species_resistance <- data.frame(Species = species,
                                         Treatment = treatment,
                                         Resistance = Resistance,
                                         stringsAsFactors = FALSE)
        
        resistance_df <- rbind(resistance_df, species_resistance)
      }
    }
  }
  
  resistance_df
}
data <- AMVG51_230505_16S_clean$samples
bacteria <- resistance_D1(data)
bacteria2 <- resistance_D2(data)


#plot
#for the labels
bacteria$facet_label <- NA
bacteria$facet_label[bacteria$Species == "Epe.fal"] <- "E. falcata"
bacteria$facet_label[bacteria$Species == "Iry.hos"] <- "I. hostmanii"
bacteria$facet_label[bacteria$Species == "Jac.cop"] <- "J. copaia"
bacteria$facet_label[bacteria$Species == "Pte.off"] <- "P. officinalis"
bacteria$facet_label[bacteria$Species == "Sym.off"] <- "S. globulifera"
bacteria$facet_label[bacteria$Species == "Tac.mel"] <- "T. melinonii"
bacteria$facet_label[bacteria$Species == "Vir.sur"] <- "V. surinamensis"

ef <- bacteria %>%
  filter(Species == "Epe.fal") %>%
  ggplot() +
 aes(x = Treatment, y = Resistance,  colour = Treatment) +
 ylab("Resistance")+
 geom_point(shape = "circle", size = 4L,  alpha = 0.5) +
 scale_color_manual(values = c( 
      D2 = "#ECE320", D3 = "#DF5220")) +
 labs(title = "E. falcata") +
 xlab(" ")+
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
 theme_minimal(base_size =20)+
 theme(title = element_text(face = "italic"), plot.title =   element_text(hjust = 0.5))+
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
 theme(legend.position = "none")

ih <-  bacteria %>%
  filter(Species == "Iry.hos") %>%
  ggplot() +
 aes(x = Treatment, y = Resistance,  colour = Treatment) +
 ylab("Resistance")+
 geom_point(shape = "circle", size = 4L,  alpha = 0.5) +
 scale_color_manual(values = c( 
      D2 = "#ECE320", D3 = "#DF5220")) +
 labs(title = "I. hostmanii") +
 xlab(" ")+
  ylab(" ")+
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
 theme_minimal(base_size =20)+
 theme(title = element_text(face = "italic"), plot.title =   element_text(hjust = 0.5))+
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
   theme(legend.position = "none")

jc <-  bacteria %>%
  filter(Species == "Jac.cop") %>%
  ggplot() +
 aes(x = Treatment, y = Resistance,  colour = Treatment) +
 ylab("Resistance")+
 geom_point(shape = "circle", size = 4L,  alpha = 0.5) +
 scale_color_manual(values = c( 
      D2 = "#ECE320", D3 = "#DF5220")) +
 labs(title = "J. copaia") +
 xlab(" ")+
  ylab(" ")+
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
 theme_minimal(base_size =20)+
 theme(title = element_text(face = "italic"), plot.title =   element_text(hjust = 0.5))+
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
   theme(legend.position = "none")

po <- bacteria %>%
  filter(Species == "Pte.off") %>%
  ggplot() +
 aes(x = Treatment, y = Resistance,  colour = Treatment) +
 ylab("Resistance")+
 geom_point(shape = "circle", size = 4L,  alpha = 0.5) +
 scale_color_manual(values = c( 
      D2 = "#ECE320", D3 = "#DF5220")) +
 labs(title = "P. officinalis") +
 xlab(" ")+
  ylab(" ")+
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
 theme_minimal(base_size =20)+
 theme(title = element_text(face = "italic"), plot.title =   element_text(hjust = 0.5))+
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
   theme(legend.position = "none")

sg <- bacteria %>%
  filter(Species == "Sym.off") %>%
  ggplot() +
 aes(x = Treatment, y = Resistance,  colour = Treatment) +
 ylab("Resistance")+
 geom_point(shape = "circle", size = 4L,  alpha = 0.5) +
 scale_color_manual(values = c( 
      D2 = "#ECE320", D3 = "#DF5220")) +
 labs(title = "S. globulifera") +
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
 theme_minimal(base_size =20)+
 theme(title = element_text(face = "italic"), plot.title =   element_text(hjust = 0.5))+
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
   theme(legend.position = "none")


tm <- bacteria %>%
  filter(Species == "Tac.mel") %>%
  ggplot() +
 aes(x = Treatment, y = Resistance,  colour = Treatment) +
 ylab("Resistance")+
 geom_point(shape = "circle", size = 4L,  alpha = 0.5) +
 scale_color_manual(values = c( 
      D2 = "#ECE320", D3 = "#DF5220")) +
 labs(title = "T. melinonii") +
  ylab(" ")+
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
 theme_minimal(base_size =20)+
 theme(title = element_text(face = "italic"), plot.title =   element_text(hjust = 0.5))+
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
   theme(legend.position = "none")

vs <- bacteria %>%
  filter(Species == "Vir.sur") %>%
  ggplot() +
 aes(x = Treatment, y = Resistance,  colour = Treatment) +
 ylab("Resistance")+
 geom_point(shape = "circle", size = 4L,  alpha = 0.5) +
 scale_color_manual(values = c( 
      D2 = "#ECE320", D3 = "#DF5220")) +
 labs(title = "V. surinamensis") +
  ylab(" ")+
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
 theme_minimal(base_size =20)+
 theme(title = element_text(face = "italic"), plot.title =   element_text(hjust = 0.5))+
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
   theme(legend.position = "none")

#get legend
legend <- get_legend(vs)
legend <- as_ggplot(legend)


resistance_bac <- ggarrange(ef, ih, jc, po, sg, tm, vs, legend, nrow = 2, ncol = 4)
resistance_bac
ggsave(filename = "./results/microbio/resistance_bac.jpeg", plot = resistance_bac, width = 12, height = 10)
```

## Recovery index for D1

function with c0
```{r}
Recovery_D1 <- function(data, trait_name) {
  recovery_df <- data.frame(Species = character(),
                            Treatment = character(),
                            Recovery = numeric(),
                            stringsAsFactors = FALSE)
  unique_species <- unique(data$Species)
  Treatments <- c("D1")  # Treatments to calculate Recovery for
  
  for (i in unique_species) {
    species_data <- data %>% filter(Species == i)
    
    CT_drought <- mean(species_data$Shannon[species_data$Time == "t21" & species_data$Treatment == "C0"])
    CT_recovery <- mean(species_data$Shannon[species_data$Time == "t51" & species_data$Treatment == "C0"])
    
    for (j in Treatments) {
      DT1 <- abs(CT_drought - mean(species_data$Shannon[species_data$Time %in% c("t21") & species_data$Treatment == j]))
      DT2 <- abs(CT_recovery - mean(species_data$Shannon[species_data$Time %in% c("t51") & species_data$Treatment == j]))
      
      Recovery <- ((2 * abs(DT1)) / (abs(DT1) + abs(DT2))) - 1
      
      species_recovery <- data.frame(Species = i,
                                     Treatment = j,
                                     Recovery = Recovery,
                                     stringsAsFactors = FALSE)
      
      recovery_df <- rbind(recovery_df, species_recovery)
    }
  }
  
  return(recovery_df)
}

Recovery_D2 <- function(data, trait_name) {
  recovery_df <- data.frame(Species = character(),
                            Treatment = character(),
                            Recovery = numeric(),
                            stringsAsFactors = FALSE)
  unique_species <- unique(data$Species)
  Treatments <- c("D2")  # Treatments to calculate Recovery for
  
  for (i in unique_species) {
    species_data <- data %>% filter(Species == i)
    
    CT_drought <- mean(species_data$Shannon[species_data$Time == "t27" & species_data$Treatment == "C0"])
    CT_recovery <- mean(species_data$Shannon[species_data$Time == "t57" & species_data$Treatment == "C0"])
    
    for (j in Treatments) {
      DT1 <- abs(CT_drought - mean(species_data$Shannon[species_data$Time %in% c("t27") & species_data$Treatment == j]))
      DT2 <- abs(CT_recovery - mean(species_data$Shannon[species_data$Time %in% c("t57") & species_data$Treatment == j]))
      
      Recovery <- ((2 * abs(DT1)) / (abs(DT1) + abs(DT2))) - 1
      
      species_recovery <- data.frame(Species = i,
                                     Treatment = j,
                                     Recovery = Recovery,
                                     stringsAsFactors = FALSE)
      
      recovery_df <- rbind(recovery_df, species_recovery)
    }
  }
  
  return(recovery_df)
}

Recovery_D3 <- function(data, trait_name) {
  recovery_df <- data.frame(Species = character(),
                            Treatment = character(),
                            Recovery = numeric(),
                            stringsAsFactors = FALSE)
  unique_species <- unique(data$Species)
  Treatments <- c("D3")  # Treatments to calculate Recovery for
  
  for (i in unique_species) {
    species_data <- data %>% filter(Species == i)
    
    CT_drought <- mean(species_data$Shannon[species_data$Time == "t71" & species_data$Treatment == "C0"])
    CT_recovery <- mean(species_data$Shannon[species_data$Time == "t101" & species_data$Treatment == "C0"])
    
    for (j in Treatments) {
      DT1 <- abs(CT_drought - mean(species_data$Shannon[species_data$Time %in% c("t71") & species_data$Treatment == j]))
      DT2 <- abs(CT_recovery - mean(species_data$Shannon[species_data$Time %in% c("t101") & species_data$Treatment == j]))
      
      Recovery <- ((2 * abs(DT1)) / (abs(DT1) + abs(DT2))) - 1
      
      species_recovery <- data.frame(Species = i,
                                     Treatment = j,
                                     Recovery = Recovery,
                                     stringsAsFactors = FALSE)
      
      recovery_df <- rbind(recovery_df, species_recovery)
    }
  }
  
  return(recovery_df)
}
```

```{r}
bacteria1 <- Recovery_D1(data)
bacteria2 <- Recovery_D2(data)
bacteria3 <- Recovery_D3(data)

bacteria <- rbind(bacteria1, bacteria2, bacteria3)
bacteria
write.csv(x = bacteria, file = "bactery_recovery_index.csv")
```


##Beta - during drought boxplots


The raw microbial community composition data was processed to calculate the Hellinger transformed Bray-Curtis dissimilarity matrix (bray_curtis_dist) using the vegan package.
The dissimilarity matrix was transformed into a tidy format for further analysis, filtering out self-comparisons and extracting relevant information such as individual IDs, treatment, time, species, and code. The dissimilarity matrix was merged with the information data frame based on individual codes, resulting in a merged data frame (bray_curtis_dist_merged2) that contains the necessary information for analysis. The data was filtered to exclude controles, include only comparisons within the same species and within the desired time periods (t21, t27, t71). 
```{r}
data_16S <- AMVG51_230505_16S_clean

hill <- hill_taxa_parti_pairwise(data_16S$reads,q=1,output="matrix", pairs = "full") 
hill.beta  <- hill$TD_beta
hill.beta.dist <- as.dist(hill.beta)

#bray_curtis_dist <- vegan::vegdist(decostand(data_16S$reads, method= "hellinger" ), method = "bray")

hill.beta.dist <- hill.beta.dist %>%
  as.matrix() %>%
  as_tibble(rownames = "sample_id") %>%
  pivot_longer(-sample_id) %>%
  filter(sample_id < name) %>% #don't repeat the same compared to self
  mutate(indv_a = str_replace(sample_id, ".*__", "")) %>%
  mutate(indv_b = str_replace(name, ".*__", ""))
 
info <- data_16S$samples %>%
  dplyr::select(sample_id, Treatment, Time, Species, code)

dim(info) #276*5
dim(hill.beta.dist) #37950     5

dist_merged <- merge(hill.beta.dist, info, by.x = "indv_a", by.y = "code", all.x = TRUE) %>%
  rename(indv_a_species = Species) %>%
  rename(indv_a_time = Time) %>%
  rename(indv_a_treatment = Treatment) %>%
  dplyr::select(-sample_id.x, -sample_id.y)

dist_merged2 <- merge(dist_merged, info, by.x = "indv_b", by.y = "code", all.x = TRUE) %>%
  rename(indv_b_species = Species) %>%
  rename(indv_b_time = Time) %>%
  rename(indv_b_treatment = Treatment) %>%
  dplyr::select(-sample_id, -name)

beta<- dist_merged2 %>% 
  dplyr::filter(indv_b_species == indv_a_species) %>% #compare within same species
  filter(indv_b_time == indv_a_time) %>%#compare with the same time period
  filter(indv_b_treatment ==indv_a_treatment )

write.csv(x = beta, file = "beta_bac.csv")

library(dplyr)
library(ggplot2)

plot_control <- beta %>%
 filter(indv_a_time %in% c("t21", "t27", "t71")) %>%
 filter(indv_b_time %in% c("t21", "t27", "t71"
)) %>%
 ggplot() +
 aes(x = indv_b_treatment, y = value, fill = indv_b_treatment) +
 geom_boxplot() +
 scale_fill_manual(values = c(C0 = "#6DA0F8", 
D1 = "#8CB425", D2 = "#ECE320", D3 = "#DF5220")) +
 labs(y = "Beta-diversity", fill = "Treatment") +
 theme_classic() +
 theme(axis.title.y = element_text(size = 15L, face = "bold")) +
 facet_wrap(vars(indv_b_species))+
  ggtitle("Bacteria")

#for the labels
beta$facet_label <- NA
beta$facet_label[beta$indv_b_species == "Epe.fal"] <- "E. falcata"
beta$facet_label[beta$indv_b_species == "Iry.hos"] <- "I. hostmanii"
beta$facet_label[beta$indv_b_species == "Jac.cop"] <- "J. copaia"
beta$facet_label[beta$indv_b_species == "Pte.off"] <- "P. officinalis"
beta$facet_label[beta$indv_b_species == "Sym.off"] <- "S. globulifera"
beta$facet_label[beta$indv_b_species == "Tac.mel"] <- "T. melinonii"
beta$facet_label[beta$indv_b_species == "Vir.sur"] <- "V. surinamensis"

plot_withoutC0<- beta %>%
# filter(!(indv_a_treatment %in% "C0")) %>%
 filter(indv_a_time %in% c("t21", "t27", "t71")) %>%
# filter(!(indv_b_treatment %in%  "C0")) %>%
 filter(indv_b_time %in% c("t21", "t27", "t71")) %>%
 ggplot() +
 aes(x = indv_b_treatment, y = value, fill = indv_b_treatment) +
 geom_boxplot(alpha = 0.6) +  # Adding transparency and softer color for boxplots
  geom_jitter(alpha = 0.5, fill = "gray80") +   # Adding transparency and softer color for jitter points
 scale_fill_manual(values = c( 
D1 = "#8CB425", D2 = "#ECE320", D3 = "#DF5220", C0 = "#6DA0F8")) +
 labs(y = "Beta-diversity", fill = "Treatment") +
 theme_classic() +
 theme(
    axis.title.y = element_text(size = 15L, face = "bold"),
    axis.title.x = element_blank(),
   # panel.background = element_rect(fill = "transparent"),  # Making the panel background transparent
    strip.background = element_blank() ,# Removing background from facet labels
    strip.text = element_text(size=12, face= "italic")
   )+
 facet_wrap(vars(facet_label)) +  # Apply the custom labeller function)+
  ggtitle("Bacteria")

plot_withoutC0
```

plots individual to add statisitical significance
```{r}
 #epefal
ef <- beta %>%
 filter(facet_label == "E. falcata") %>%
 #filter(!(indv_a_treatment %in% "C0")) %>%
 filter(indv_a_time %in% c("t21", "t27", "t71")) %>%
 #filter(!(indv_b_treatment %in% "C0")) %>%
 filter(indv_b_time %in% c("t21", "t27", "t71")) %>%
  ggplot() +
 aes(x = indv_b_treatment, y = value, fill = indv_b_treatment) +
 geom_boxplot(alpha = 0.6) + 
  geom_jitter(alpha = 0.5, fill = "gray80") +  
 scale_fill_manual(values = c( 
D1 = "#8CB425", D2 = "#ECE320", D3 = "#DF5220", C0 = "#219ebc")) +
 labs(y = "Beta-diversity", fill = "Treatment") +
 theme_classic(base_size = 14) +
 theme(
    axis.title.y = element_text(size = 15L, face = "bold"),
    axis.title.x = element_blank(),
    axis.text.x = element_blank())+
  stat_compare_means(aes(label = sprintf("Kurskall-Wallis, p < 0.001", as.numeric(..p.format..))),method = "kruskal.test", label.y = c(1.9))+
  stat_compare_means(comparisons = list(c("D1", "D2"), c("C0", "D2")), label="p.signif", method = "wilcox.test", label.y = c(1.5, 1.70))+
  ggtitle(expression(italic("E. falcata"))) +
  theme(legend.position = "none")+
  ylim(1,2)
ef

#iryhos
ih <- beta %>%
 filter(facet_label == "I. hostmanii") %>%
# filter(!(indv_a_treatment %in% "C0")) %>%
 filter(indv_a_time %in% c("t21", "t27", "t71")) %>%
 #filter(!(indv_b_treatment %in% "C0")) %>%
 filter(indv_b_time %in% c("t21", "t27", "t71")) %>%
  ggplot() +
 aes(x = indv_b_treatment, y = value, fill = indv_b_treatment) +
 geom_boxplot(alpha = 0.6) + 
  geom_jitter(alpha = 0.5, fill = "gray80") +  
 scale_fill_manual(values = c( 
D1 = "#8CB425", D2 = "#ECE320", D3 = "#DF5220", C0 = "#219ebc")) +
 labs(y = "Beta-diversity", fill = "Treatment") +
 theme_classic(base_size = 14) +
 theme(
    axis.title.y = element_blank(),
    axis.title.x = element_blank(), axis.text.x = element_blank())+
  stat_compare_means(method = "kruskal.test", label.y = c(1.9), label.x=1.5)+
 # stat_compare_means(comparisons = list(c("D1", "D2"), c("D2", "D3"), c("D1", "D3"), c("C0", "D1"), c("C0", "D2"), c("C0" , "D3")), label="p.signif", method = "wilcox.test", label.y = c(1.5, 1.6,1.7,1.2,1.3,1.25))+
  ggtitle(expression(italic("I. hostmanii"))) +
  theme(legend.position = "none")+
   ylim(1,2)
ih

#jaccop
jc <- beta %>%
 filter(facet_label == "J. copaia") %>%
 filter(indv_a_time %in% c("t21", "t27", "t71")) %>%
 filter(indv_b_time %in% c("t21", "t27", "t71")) %>%
  ggplot() +
 aes(x = indv_b_treatment, y = value, fill = indv_b_treatment) +
 geom_boxplot(alpha = 0.6) + 
  geom_jitter(alpha = 0.5, fill = "gray80") +  
 scale_fill_manual(values = c( 
D1 = "#8CB425", D2 = "#ECE320", D3 = "#DF5220", C0 = "#219ebc")) +
 labs(y = "Beta-diversity", fill = "Treatment") +
 theme_classic(base_size = 14) +
 theme(
    axis.title.y = element_blank(),
    axis.title.x = element_blank(), axis.text.x = element_blank() )+
  stat_compare_means(method = "kruskal.test", label.y = c(1.9), label.x=1.5)+
 # stat_compare_means(comparisons = list(c("D1", "D2"), c("D2", "D3"), c("D1", "D3")), method = "wilcox.test", label.y = c(1.5, 1.6,1.7))+
  ggtitle(expression(italic("J. copaia"))) +
  theme(legend.position = "none")+
   ylim(1,2)
jc

#pteoff
po <-  beta %>%
 filter(facet_label == "P. officinalis") %>%
 filter(indv_a_time %in% c("t21", "t27", "t71")) %>%
 filter(indv_b_time %in% c("t21", "t27", "t71")) %>%
  ggplot() +
 aes(x = indv_b_treatment, y = value, fill = indv_b_treatment) +
 geom_boxplot(alpha = 0.6) + 
  geom_jitter(alpha = 0.5, fill = "gray80") +  
 scale_fill_manual(values = c( 
D1 = "#8CB425", D2 = "#ECE320", D3 = "#DF5220", C0 = "#219ebc")) +
 theme_classic(base_size = 14) +
 theme(
    axis.title.y = element_blank(),
    axis.title.x = element_blank(), axis.text.x = element_blank())+
  stat_compare_means(aes(label = sprintf("Kurskall-Wallis, p = %5.2f", as.numeric(..p.format..))),method = "kruskal.test", label.y = c(1.9), label.x=1.5)+
 # stat_compare_means(comparisons = list(c("D1", "D2"), c("D2", "D3"), c("D1", "D3")), method = "wilcox.test", label.y = c(1.5, 1.6,1.7))+
  ggtitle(expression(italic("P. officinalis"))) +
  theme(legend.position = "none")+
  ylim(1,2)
po

#symglo
sg <-  beta %>%
 filter(facet_label == "S. globulifera") %>%
 filter(indv_a_time %in% c("t21", "t27", "t71")) %>%
 filter(indv_b_time %in% c("t21", "t27", "t71")) %>%
  ggplot() +
 aes(x = indv_b_treatment, y = value, fill = indv_b_treatment) +
 geom_boxplot(alpha = 0.6) + 
  geom_jitter(alpha = 0.5, fill = "gray80") +  
 scale_fill_manual(values = c( 
D1 = "#8CB425", D2 = "#ECE320", D3 = "#DF5220", C0 = "#219ebc")) +
 labs(y = "Beta-diversity", fill = "Treatment") +
 theme_classic(base_size = 14) +
 theme(
    axis.title.y =  element_text(size = 15L, face = "bold"),
    axis.title.x = element_blank(), axis.text.x = element_blank())+
  stat_compare_means(method = "kruskal.test", label.y = c(1.9), label.x=1.3)+
 # stat_compare_means(comparisons = list(c("D1", "D2"), c("D2", "D3"), c("D1", "D3")), method = "wilcox.test", label.y = c(1.5, 1.6,1.7))+
  ggtitle(expression(italic("S. globulifera"))) +
  theme(legend.position = "none")+
  ylim(1,2)
sg

#tacmel
tm <-   beta %>%
 filter(facet_label == "T. melinonii") %>%
 filter(indv_a_time %in% c("t21", "t27", "t71")) %>%
 filter(indv_b_time %in% c("t21", "t27", "t71")) %>%
  ggplot() +
 aes(x = indv_b_treatment, y = value, fill = indv_b_treatment) +
 geom_boxplot(alpha = 0.6) + 
  geom_jitter(alpha = 0.5, fill = "gray80") +  
 scale_fill_manual(values = c( 
D1 = "#8CB425", D2 = "#ECE320", D3 = "#DF5220", C0 = "#219ebc")) +
 labs(y = "Beta-diversity", fill = "Treatment") +
 theme_classic(base_size = 14) +
 theme(
    axis.title.y = element_blank(),
    axis.title.x = element_blank(), axis.text.x = element_blank())+
  stat_compare_means(aes(label = sprintf("Kurskall-Wallis, p < 0.01", as.numeric(..p.format..))),method = "kruskal.test", label.y = c(1.9), label.x=1.5)+
  stat_compare_means(comparisons = list(c("D1", "D2"), c("D1", "D3"),  c("C0", "D2"), c("C0" , "D3")), label="p.signif", method = "wilcox.test", label.y = c(1.5, 1.4,1.6,1.75))+
  ggtitle(expression(italic("T. melinonii"))) +
  theme(legend.position = "none")+
  ylim(1,2)
tm

#virsur
vs <-  beta %>%
 filter(facet_label == "V. surinamensis") %>%
 filter(indv_a_time %in% c("t21", "t27", "t71")) %>%
 filter(indv_b_time %in% c("t21", "t27", "t71")) %>%
  ggplot() +
 aes(x = indv_b_treatment, y = value, fill = indv_b_treatment) +
 geom_boxplot(alpha = 0.6) + 
  geom_jitter(alpha = 0.5, fill = "gray80") +  
 scale_fill_manual(values = c( 
D1 = "#8CB425", D2 = "#ECE320", D3 = "#DF5220", C0 = "#219ebc")) +
 theme_classic(base_size = 14) +
  labs(y = "Beta-diversity", fill = "Treatment") +
 theme(
    axis.title.y = element_blank(),
    axis.title.x = element_blank(), axis.text.x = element_blank())+
  stat_compare_means(aes(label = sprintf("Kurskall-Wallis, p = %5.2f", as.numeric(..p.format..))),method = "kruskal.test", label.y = c(1.9), label.x=1.5)+
  stat_compare_means(comparisons = list(c("C0", "D2"), c("C0" , "D3"), c("C0", "D1")), label="p.signif", method = "wilcox.test", label.y = c(1.6, 1.7, 1.5))+
  ggtitle(expression(italic("V. surinamensis"))) +
  theme(legend.position = "none")+
 #theme(legend.position = "bottom", legend.box = "vertical", legend.direction = "vertical")+
  ylim(1,2)
vs

legend <- get_legend(vs)

beta_bac_drought <- ggarrange(ef, ih, jc, po, sg, tm, vs, as_ggplot(legend),nrow = 2, ncol=4, widths = c(0.11,0.1,0.1,0.1))

beta_bac_drought <- annotate_figure(beta_bac_drought, top = text_grob("A. Bacteria", color = "black", face = "bold", size = 20, just = "right" , hjust = 3, vjust = 0.5))

ggsave(filename = "./results/microbio/beta_bac_drought.jpeg", plot = beta_bac_drought, width = 10, height = 6)

```



##Beta - after drought, recovery boxplots
```{r}
#prepare data
data <- beta %>%  
  #filter(indv_b_treatment != "C0") %>%
   mutate(Period = case_when(
     indv_b_treatment == "D1" &  indv_b_time == "t21" ~ "Drought",
      indv_b_treatment == "D1" &  indv_b_time  == "t51" ~ "Recovery",
      indv_b_treatment == "D2" & indv_b_time  == "t27" ~ "Drought",
      indv_b_treatment == "D2" &  indv_b_time  == "t57" ~ "Recovery",
      indv_b_treatment == "D3" &  indv_b_time  == "t71" ~ "Drought",
      indv_b_treatment == "D3" & indv_b_time  == "t101" ~ "Recovery",
      TRUE                      ~ "Control"
    )
  )
data$Period <- as.factor(data$Period)
data$Treatment <- as.factor(data$indv_b_treatment)

#epefal
ef <- data %>% filter(indv_b_species == "Epe.fal")
 ef_plot <- ggboxplot(
  ef, x= "Period" , y = "value", fill = "Period", facet.by = "indv_b_treatment", alpha =0.7) +           geom_jitter(shape = "circle", alpha = 0.5) +
  scale_fill_manual(values = c(Drought = "#ff9e00", Recovery = "#219ebc", Control = "#e5e5e5"))+
  theme_classic() +
  theme(axis.text.x = element_blank(), axis.title.x = element_blank())+
  ylab("Beta diversity")+
  theme(legend.position = "none")
ef_plot
stat.test <- data %>%
  filter(indv_b_species == "Epe.fal") %>%
  filter(indv_b_treatment != "D3") %>%
  group_by(indv_b_treatment) %>%
  t_test(value ~ Period) %>%
  adjust_pvalue(method = "bonferroni") %>%
  add_significance()

stat.test 
stat.test <- stat.test %>% add_xy_position(x = "Period") # Make facet and add p-values
ef_plot <- ef_plot + stat_pvalue_manual(stat.test)
ef_plot <- annotate_figure(ef_plot, left = text_grob("E. falcata       ", color = "black", face = "italic", size = 20))

#iryhos
ih <- data %>% filter(indv_b_species == "Iry.hos")
 ih_plot <- ggboxplot(
  ih, x= "Period" , y = "value", fill = "Period", facet.by = "indv_b_treatment", alpha =0.7) +           geom_jitter(shape = "circle", alpha = 0.5) +
  scale_fill_manual(values = c(Drought = "#ff9e00", Recovery = "#219ebc"))+
  theme_classic() +
  theme(axis.text.x = element_blank(), axis.title.x = element_blank())+
  ylab(" ")+
  theme(legend.position = "none")
ih_plot
stat.test <- data %>%
  filter(indv_b_species == "Iry.hos") %>%
  group_by(indv_b_treatment) %>%
  filter(Treatment != "D3")%>% 
  t_test(value ~ Period) %>%
  adjust_pvalue(method = "bonferroni") %>%
  add_significance()

stat.test 
stat.test <- stat.test %>% add_xy_position(x = "Period") 
ih_plot <- ih_plot + stat_pvalue_manual(stat.test)
ih_plot <- annotate_figure(ih_plot, right = text_grob("I. hostmannii", color = "black", face = "italic", size = 20))

#jaccop
jc <- data %>% filter(indv_b_species == "Jac.cop")
 jc_plot <- ggboxplot(
  jc, x= "Period" , y = "value", fill = "Period", facet.by = "indv_b_treatment", alpha =0.7) +           geom_jitter(shape = "circle", alpha = 0.5) +
  scale_fill_manual(values = c(Drought = "#ff9e00", Recovery = "#219ebc"))+
  theme_classic() +
  theme(axis.text.x = element_blank(), axis.title.x = element_blank())+
  ylab("Beta diversity")+
  theme(legend.position = "none")
jc_plot
stat.test <- jc %>%
  group_by(indv_b_treatment) %>%
  t_test(value ~ Period) %>%
  adjust_pvalue(method = "bonferroni") %>%
  add_significance()
stat.test 
stat.test <- stat.test %>% add_xy_position(x = "Period")
jc_plot <- jc_plot + stat_pvalue_manual(stat.test)
jc_plot <- annotate_figure(jc_plot, left = text_grob("J. copaia        ", color = "black", face = "italic", size = 20))

#pteoff
po <- data %>% filter(indv_b_species == "Pte.off")
 po_plot <- ggboxplot(
  po, x= "Period" , y = "value", fill = "Period", facet.by = "indv_b_treatment", alpha =0.7) +           geom_jitter(shape = "circle", alpha = 0.5) +
  scale_fill_manual(values = c(Drought = "#ff9e00", Recovery = "#219ebc"))+
  theme_classic() +
  theme(axis.text.x = element_blank(), axis.title.x = element_blank())+
  ylab(" ")+
  theme(legend.position = "none")
po_plot
stat.test <- po %>%
  filter(indv_b_treatment != "D3") %>%
 group_by(indv_b_treatment) %>%
  t_test(value ~ Period) %>%
  adjust_pvalue(method = "bonferroni") %>%
  add_significance()
stat.test 
stat.test <- stat.test %>% add_xy_position(x = "Period")
po_plot <- po_plot + stat_pvalue_manual(stat.test)
po_plot <- annotate_figure(po_plot, right = text_grob("P. officinalis", color = "black", face = "italic", size = 20))

#symglo
sg <- data %>% filter(indv_b_species == "Sym.off")
 sg_plot <- ggboxplot(
  sg, x= "Period" , y = "value", fill = "Period", facet.by = "indv_b_treatment", alpha =0.7) +           geom_jitter(shape = "circle", alpha = 0.5) +
  scale_fill_manual(values = c(Drought = "#ff9e00", Recovery = "#219ebc"))+
  theme_classic() +
  theme(axis.text.x = element_blank(), axis.title.x = element_blank())+
  ylab("Beta diversity")+
  theme(legend.position = "none")
sg_plot
stat.test <- sg %>%
  group_by(indv_b_treatment) %>%
  t_test(value ~ Period) %>%
  adjust_pvalue(method = "bonferroni") %>%
  add_significance()
stat.test 
stat.test <- stat.test %>% add_xy_position(x = "Period")
sg_plot <- sg_plot + stat_pvalue_manual(stat.test)
sg_plot <- annotate_figure(sg_plot, left = text_grob("S. globulifera  ", color = "black", face = "italic", size = 20))

#tacmel
tm <- data %>% filter(indv_b_species == "Tac.mel")
 tm_plot <- ggboxplot(
  tm, x= "Period" , y = "value", fill = "Period", facet.by = "indv_b_treatment", alpha =0.7) +           geom_jitter(shape = "circle", alpha = 0.5) +
  scale_fill_manual(values = c(Drought = "#ff9e00", Recovery = "#219ebc"))+
  theme_classic() +
  theme(axis.text.x = element_blank(), axis.title.x = element_blank())+
  ylab(" ")+
  theme(legend.position = "none")
tm_plot
stat.test <- tm %>%
 group_by(indv_b_treatment) %>%
  t_test(value ~ Period) %>%
  adjust_pvalue(method = "bonferroni") %>%
  add_significance()
stat.test 
stat.test <- stat.test %>% add_xy_position(x = "Period")
tm_plot <- tm_plot + stat_pvalue_manual(stat.test)
tm_plot <- annotate_figure(tm_plot, right = text_grob("T. melinonii", color = "black", face = "italic", size = 20))

#virsur
vs <- data %>% filter(indv_b_species == "Vir.sur")
 vs_plot <- ggboxplot(
  vs, x= "Period" , y = "value", fill = "Period", facet.by = "indv_b_treatment", alpha =0.7) +           geom_jitter(shape = "circle", alpha = 0.5) +
  scale_fill_manual(values = c(Drought = "#ff9e00", Recovery = "#219ebc"))+
  theme_classic() +
  theme(axis.text.x = element_blank(), axis.title.x = element_blank())+
  ylab("Beta diversity")+
  theme(legend.position = "none")
vs_plot
stat.test <- vs %>%
  filter(indv_b_treatment != "D3") %>%
   group_by(indv_b_treatment) %>%
  t_test(value ~ Period) %>%
  adjust_pvalue(method = "bonferroni") %>%
  add_significance()
stat.test 
stat.test <- stat.test %>% add_xy_position(x = "Period") 
vs_plot <- vs_plot + stat_pvalue_manual(stat.test)
vs_plot <- annotate_figure(vs_plot, left = text_grob("V. surinamensis", color = "black", face = "italic", size = 20))

#get legend
legend <- get_legend(vs_plot) #get legend work with no complicated plots fso just grab the legend when plot is simpler


#plot all
beta_recov_bac <- ggarrange(ef_plot, ih_plot, jc_plot, po_plot, sg_plot, tm_plot, vs_plot, as_ggplot(legend), ncol=2, nrow = 4)
beta_recov_bac
ggsave(filename = "./results/microbio/beta_bac_recov.jpeg", plot = beta_recov_bac, width = 12, height = 10)
```

#PCOA -RECVOERY

```{r}
bacteria <-AMVG51_230505_16S_clean
rownames(bacteria$samples) <- bacteria$samples$sample_id
bacteria$samples <- bacteria$samples %>%
   mutate(Period = case_when(
     Treatment == "D1" &  Time == "t21" ~ "Drought D1",
      Treatment == "D1" &  Time == "t51" ~ "Recovery D1",
      Treatment == "D2" &  Time == "t27" ~ "Drought D2",
      Treatment == "D2" &  Time == "t57" ~ "Recovery D2",
      Treatment == "D3" &  Time == "t71" ~ "Drought D3",
      Treatment == "D3" &  Time == "t101" ~ "Recovery D3",
      TRUE                      ~ "Control"
    )
  )

bacteria$samples$Name <- NA
bacteria$samples$Name[bacteria$samples$Species == "Epe.fal"] <- "E. falcata"
bacteria$samples$Name[bacteria$samples$Species == "Iry.hos"] <- "I. hostmannii"
bacteria$samples$Name[bacteria$samples$Species == "Jac.cop"] <- "J. copaia"
bacteria$samples$Name[bacteria$samples$Species == "Pte.off"] <- "P. officinalis"
bacteria$samples$Name[bacteria$samples$Species == "Sym.off"] <- "S. globulifera"
bacteria$samples$Name[bacteria$samples$Species == "Tac.mel"] <- "T. melinonii"
bacteria$samples$Name[bacteria$samples$Species == "Vir.sur"] <- "V. surinamensis"

bacteria_D1 <- subset_metabarlist(bacteria, table = "samples", indices = bacteria$samples$Time %in% c("t21", "t51"))

#load metadata and taxonomy file
meta<-bacteria_D1$samples
hill <- hill_taxa_parti_pairwise(bacteria_D1$reads,q=1,output="matrix", pairs = "full") 
hill.beta  <- hill$TD_beta
hill.beta.dist <- as.dist(hill.beta)

# tests whether there are differences in dispersion
bd_treatment <- betadisper(d = hill.beta.dist, group = bacteria_D1$samples$Period)
bd_treatment
anova(bd_treatment)
permutest(bd_treatment) #are there significant dispersions between the treatments? no: p-value 0.413. having no significant effects in the dispersions, easier and less problematic to interpret afterwards the adonis result. 

bd_Name <- betadisper(d = hill.beta.dist, group = bacteria_D1$samples$Period)
bd_Name
anova(bd_Name)
permutest(bd_Name)

bd_Name_period <- betadisper(d = hill.beta.dist, group = paste0(bacteria_D1$samples$Period, bacteria_D1$samples$Name))
bd_Name_period
anova(bd_Name_period)

# test whether there are differences in fungal data between different water treatment periods while accounting for variation among host plant species.
Permanova  <-
  adonis2(hill.beta.dist~ bacteria_D1$samples$Period, permutations = 999, strata=bacteria_D1$samples$Name)  #permutation done within the same name

Permanova_bacteria <-data.frame(Permanova, check.names = F) %>% 
  rename("R^2" =  "R2", "p-value"="Pr(>F)") %>% 
  rownames_to_column( var="Factor")

Permanova_bacteria[1,1] <- "Treatment"

library(ggpubr)
perm_D1 <- Permanova_bacteria %>%
mutate_at(3, funs(round(., 0)))  %>%
    mutate_at(5:6, as.numeric) %>% 
  mutate_at(4:5, funs(round(., 2))) %>% 
  mutate_at(5:6, ~replace(., is.na(.), "")) %>% 
    ggtexttable(., rows = NULL,  theme = ttheme(
             colnames.style = colnames_style(color = "black", fill = "white",hjust=0, x=0.1, size = 10), tbody.style = tbody_style(color = "black",  fill = c("white"), hjust=0, x=0.1))) %>%
  tab_add_hline(at.row = 1:2, row.side = "top", linewidth = 2)%>%
  table_cell_font(., row = 2, column = 6, face = "bold") %>% 
  tab_add_hline(at.row = c(4), row.side = "bottom", 
                linewidth = 1, linetype = 1) %>% 
  table_cell_font(., row = 2:4, column = 1,
                     size = 11)%>%
 tab_add_title(text = "D1", face = "bold", padding = unit(0.5, "line"))

library(pairwiseAdonis)
pairwise_d1 <- pairwise.adonis2(x = hill.beta.dist~ Period,data = bacteria_D1$samples,nperm = 999, strata = "Name")

pairwise_d1_RD <- pairwise_d1$`Recovery D1_vs_Drought D1` %>% as_tibble() %>%
  dplyr::select(R2, `Pr(>F)`)%>% 
  rename("R^2" =  "R2", "p-value"="Pr(>F)") 

subtitle <- paste0(
  "drought", 
  " vs.",
  " recovery") %>%
  strwrap(width = 8) %>%
  paste(collapse = "\n")

pw_d1<- pairwise_d1_RD  %>%
    mutate_at(1:2, as.numeric) %>% 
  mutate_at(1:2, funs(round(., 2))) %>% 
  mutate_at(1:2, ~replace(., is.na(.), "")) %>% 
    ggtexttable(., rows = NULL,  theme = ttheme(
             colnames.style = colnames_style(color = "black", fill = "white",hjust=0, x=0.1, size = 10), tbody.style = tbody_style(color = "black",  fill = c("white"), hjust=0, x=0.1))) %>%
  tab_add_hline(at.row = 1:2, row.side = "top", linewidth = 2)%>%
  tab_add_hline(at.row = c(4), row.side = "bottom", 
                linewidth = 1, linetype = 1) %>% 
  table_cell_font(., row = 2:4, column = 1,
                     size = 11)%>%
 tab_add_title(text = subtitle, face = "bold", padding = unit(0.5, "line"))

pairwise_d1_RC <- pairwise_d1$`Recovery D1_vs_Control` %>% as_tibble() %>%
  dplyr::select(R2, `Pr(>F)`)%>% 
  rename("R^2" =  "R2", "p-value"="Pr(>F)") 

subtitle <- paste0(
  "control", 
  " vs.",
  " recovery") %>%
  strwrap(width = 8) %>%
  paste(collapse = "\n")

pw_d1_RC<- pairwise_d1_RC  %>%
    mutate_at(1:2, as.numeric) %>% 
  mutate_at(1:2, funs(round(., 2))) %>% 
  mutate_at(1:2, ~replace(., is.na(.), "")) %>% 
    ggtexttable(., rows = NULL,  theme = ttheme(
             colnames.style = colnames_style(color = "black", fill = "white",hjust=0, x=0.1, size = 10), tbody.style = tbody_style(color = "black",  fill = c("white"), hjust=0, x=0.1))) %>%
  tab_add_hline(at.row = 1:2, row.side = "top", linewidth = 2)%>%
  table_cell_font(., row = 2, column = 2, face = "bold") %>% 
  tab_add_hline(at.row = c(4), row.side = "bottom", 
                linewidth = 1, linetype = 1) %>% 
  table_cell_font(., row = 2:4, column = 1,
                     size = 11)%>%
 tab_add_title(text = subtitle, face = "bold", padding = unit(0.5, "line"))

pairwise_d1_DC <- pairwise_d1$`Drought D1_vs_Control` %>% as_tibble() %>%
  dplyr::select(R2, `Pr(>F)`)%>% 
  rename("R^2" =  "R2", "p-value"="Pr(>F)") 

subtitle <- paste0(
  "control", 
  " vs.",
  " drought") %>%
  strwrap(width = 8) %>%
  paste(collapse = "\n")

pw_d1_DC<- pairwise_d1_DC  %>%
    mutate_at(1:2, as.numeric) %>% 
  mutate_at(1:2, funs(round(., 3))) %>% 
  mutate_at(1:2, ~replace(., is.na(.), "")) %>% 
    ggtexttable(., rows = NULL,  theme = ttheme(
             colnames.style = colnames_style(color = "black", fill = "white",hjust=0, x=0.1, size = 10), tbody.style = tbody_style(color = "black",  fill = c("white"), hjust=0, x=0.1))) %>%
  tab_add_hline(at.row = 1:2, row.side = "top", linewidth = 2)%>%
  table_cell_font(., row = 2, column = 2, face = "bold") %>% 
  tab_add_hline(at.row = c(4), row.side = "bottom", 
                linewidth = 1, linetype = 1) %>% 
  table_cell_font(., row = 2:4, column = 1,
                     size = 11)%>%
 tab_add_title(text = subtitle, face = "bold", padding = unit(0.5, "line"))

D1 <- ggarrange(perm_D1, pw_d1, pw_d1_DC, pw_d1_RC, widths = c(1, 0.2, 0.2, 0.2), nrow=1, ncol=4)
D1


#D2
bacteria_D2 <- subset_metabarlist(bacteria, table = "samples", indices = bacteria$samples$Time %in% c("t27", "t57"))

meta<-bacteria_D2$samples
hill <- hill_taxa_parti_pairwise(bacteria_D2$reads,q=1,output="matrix", pairs = "full") 
hill.beta  <- hill$TD_beta
hill.beta.dist <- as.dist(hill.beta)

# tests whether there are differences in dispersion
bd_treatment <- betadisper(d = hill.beta.dist, group = bacteria_D2$samples$Period)
bd_treatment
anova(bd_treatment)
permutest(bd_treatment) #are there significant dispersions between the treatments? no: p-value 0.413. having no significant effects in the dispersions, easier and less problematic to interpret afterwards the adonis result. 

bd_Name <- betadisper(d = hill.beta.dist, group = bacteria_D2$samples$Period)
bd_Name
anova(bd_Name)
permutest(bd_Name) #0.059 .

bd_Name_period <- betadisper(d = hill.beta.dist, group = paste0(bacteria_D2$samples$Period, bacteria_D2$samples$Name))
bd_Name_period
anova(bd_Name_period) #1.414e-05 ***


# test whether there are differences in fungal data between different water treatment periods while accounting for variation among host plant species.
Permanova  <-
  adonis2(hill.beta.dist~ bacteria_D2$samples$Period, permutations = 999, strata=bacteria_D2$samples$Name)  #permutation done within the same name

Permanova_bacteria <-data.frame(Permanova, check.names = F) %>% 
  rename("R^2" =  "R2", "p-value"="Pr(>F)") %>% 
  rownames_to_column( var="Factor")

Permanova_bacteria[1,1] <- "Treatment"

perm_D2 <- Permanova_bacteria %>%
mutate_at(3, funs(round(., 0)))  %>%
    mutate_at(5:6, as.numeric) %>% 
  mutate_at(4:5, funs(round(., 2))) %>% 
  mutate_at(5:6, ~replace(., is.na(.), "")) %>% 
    ggtexttable(., rows = NULL,  theme = ttheme(
             colnames.style = colnames_style(color = "black", fill = "white",hjust=0, x=0.1, size = 10), tbody.style = tbody_style(color = "black",  fill = c("white"), hjust=0, x=0.1))) %>%
  tab_add_hline(at.row = 1:2, row.side = "top", linewidth = 2)%>%
  tab_add_hline(at.row = c(4), row.side = "bottom", 
                linewidth = 1, linetype = 1) %>% 
  table_cell_font(., row = 2:4, column = 1,
                     size = 11)%>%
  tab_add_title(text = "D2", face = "bold", padding = unit(0.5, "line"))


perm_D2


#D3 
bacteria_D3 <- subset_metabarlist(bacteria, table = "samples", indices = bacteria$samples$Time %in% c("t71", "t101"))
meta<-bacteria_D3$samples
hill <- hill_taxa_parti_pairwise(bacteria_D3$reads,q=1,output="matrix", pairs = "full") 
hill.beta  <- hill$TD_beta
hill.beta.dist <- as.dist(hill.beta)

# tests whether there are differences in dispersion
bd_treatment <- betadisper(d = hill.beta.dist, group = bacteria_D3$samples$Period)
bd_treatment
anova(bd_treatment)
permutest(bd_treatment) #are there significant dispersions between the treatments? no: p-value 0.413. having no significant effects in the dispersions, easier and less problematic to interpret afterwards the adonis result. 

bd_Name <- betadisper(d = hill.beta.dist, group = bacteria_D3$samples$Period)
bd_Name
anova(bd_Name)
permutest(bd_Name) 

bd_Name_period <- betadisper(d = hill.beta.dist, group = paste0(bacteria_D3$samples$Period, bacteria_D3$samples$Name))
bd_Name_period
anova(bd_Name_period) 

# test whether there are differences in fungal data between different water treatment periods while accounting for variation among host plant species.
Permanova  <-
  adonis2(hill.beta.dist~ bacteria_D3$samples$Period, permutations = 999, strata=bacteria_D3$samples$Name)  #permutation done within the same name


Permanova_bacteria <-data.frame(Permanova, check.names = F) %>% 
  rename("R^2" =  "R2", "p-value"="Pr(>F)") %>% 
  rownames_to_column( var="Factor")

Permanova_bacteria[1,1] <- "Treatment"

library(ggpubr)
perm_D3 <- Permanova_bacteria %>%
mutate_at(3, funs(round(., 0)))  %>%
    mutate_at(5:6, as.numeric) %>% 
  mutate_at(4:5, funs(round(., 2))) %>% 
  mutate_at(5:6, ~replace(., is.na(.), "")) %>% 
    ggtexttable(., rows = NULL,  theme = ttheme(
             colnames.style = colnames_style(color = "black", fill = "white",hjust=0, x=0.1, size = 10), tbody.style = tbody_style(color = "black",  fill = c("white"), hjust=0, x=0.1))) %>%
  tab_add_hline(at.row = 1:2, row.side = "top", linewidth = 2)%>%
  tab_add_hline(at.row = c(4), row.side = "bottom", 
                linewidth = 1, linetype = 1) %>% 
  table_cell_font(., row = 2:4, column = 1,
                     size = 11)%>%
  tab_add_title(text = "D3", face = "bold", padding = unit(0.5, "line"))

pw_d1_all<- (pw_d1+ pw_d1_DC+pw_d1_RC)+plot_layout(ncol = 3)
D1 <- ggarrange(perm_D1,pw_d1_all, nrow=1)


ggsave("./results/D1_bacteria_option3.jpeg", plot=D1, width = 10, height = 5)
ggsave("./results/D2_bacteria_option3.jpeg", plot=perm_D2, width = 10, height = 5)

ggsave("./results/D3_bacteria_option3.jpeg", plot=perm_D3, width = 10, height = 5)

```



#-Fungi ITS Analyses

```{r}
#microbio clean
load("./results/Metabarlist_serre_ITS_AMVG42-230505_clean.Rdata")
data_ITS <- AMVG42_230505_ITS_clean
rownames(data_ITS$samples) <- data_ITS$samples$sample_id
```

```{r}
fungi <- data_ITS
fungi$samples <- fungi$samples %>%
   mutate(Period = case_when(
     Treatment == "D1" &  Time == "t21" ~ "Drought D1",
      Treatment == "D1" &  Time == "t51" ~ "Recovery D1",
      Treatment == "D2" &  Time == "t27" ~ "Drought D2",
      Treatment == "D2" &  Time == "t57" ~ "Recovery D2",
      Treatment == "D3" &  Time == "t71" ~ "Drought D3",
      Treatment == "D3" &  Time == "t101" ~ "Recovery D3",
      TRUE                      ~ "Control"
    )
  )


data.offset <- fungi$reads
data.offset <- data.offset + 1 # apply offset
sum(which(data.offset == 0)) # how many zeroes are there after offset

low.count.removal <- function(
                        data, # OTU count df of size n (sample) x p (OTU)
                        percent=0.01 # cutoff chosen
                        ) 
  {
    keep.otu = which(colSums(data)*100/(sum(colSums(data))) > percent)
    data.filter = data[,keep.otu]
    return(list(data.filter = data.filter, keep.otu = keep.otu))
}

dim(data.offset) # check samples are in rows
# call the function then apply on the offset data
result.filter <- low.count.removal(data.offset, percent=0.01)
data.filter <- result.filter$data.filter

# check the number of variables kept after filtering
# in this particular case we had already filtered the data so no was change made, 
# but from now on we will work with 'data.filter'
length(result.filter$keep.otu) 
lib.size <- apply(data.filter, 1, sum) # determine total count for each sample
barplot(lib.size) # and plot as bar plot
maximum.lib.size <- 10000

data.filter <- data.filter[-which(lib.size > maximum.lib.size),]
# undergo PCA on CLR transformed data
pca.result <- pca(data.filter, logratio = 'CLR') 

data.filter_data <- as.data.frame(data.filter) 
data.filter_data$sample_id <- rownames(data.filter_data)
data.filter_data <- left_join(data.filter_data, fungi$samples)


plotIndiv(pca.result,  # plot samples
          group = data.filter_data$Time, 
          title = 'Fungi drought d3 and recovery PCA Comps 1&2',
            ellipse = TRUE,  # plot using the ellipses 
          legend = T, ind.names = FALSE, ellipse.level = 0.8)
```


## alpha diversity

```{r Hill numbers diversity}
 
#Hill numbers, using HillR package
Shannon_fun <-hill_taxa(AMVG42_230505_ITS_clean$reads,q=1) #equivalent of Shannon entropy

df <- Shannon_fun %>% 
  enframe() %>%
  rename(Shannon_fun = value, 
         sample_id = name)
AMVG42_230505_ITS_clean$samples <- left_join(AMVG42_230505_ITS_clean$samples, df)
data <- AMVG42_230505_ITS_clean$samples 

write.csv(x = data, file = "AMVG42_230505_ITS_clean_alpha.csv")

##OTHER
set.seed(12)
plant_phy_rar <- rarefy(AMVG42_230505_ITS_clean$reads, sample = 1104)
alpha_plant <- hill_taxa(AMVG42_230505_ITS_clean$reads,q=1) 

Shannon_fun <- alpha_plant %>% as_vector()

df <- Shannon_fun %>% 
  enframe() %>%
  rename(Shannon_fun = value, 
         sample_id = name)
AMVG42_230505_ITS_clean$samples <- left_join(AMVG42_230505_ITS_clean$samples, df)
data <- AMVG42_230505_ITS_clean$samples 

plant_alpha_diverity_clean <- subset(plant_alpha_diverity,source !="bulk_soil")
plant_alpha_diverity_clean$bacteria <- ifelse(plant_alpha_diverity_clean$bacteria=="Yes","SPMX","Without SPMX")
plant_alpha_diverity_clean$water <- factor(plant_alpha_diverity_clean$water,levels = c("Yes","No"))
plant_alpha_diverity_clean$bacteria <- factor(plant_alpha_diverity_clean$bacteria,levels = c("Without SPMX","SPMX"))
plant_alpha_diverity_clean$source <- factor(plant_alpha_diverity_clean$source,levels = c("root_surface","rhizosphere"))
colnames(plant_alpha_diverity_clean)[3] <- "Shannon_index"


a <- ggplot(plant_alpha_diverity_clean,aes(x=water,y=Shannon_index,fill=water))+geom_boxplot()+
  scale_fill_manual(values=c( "#E69F00", "#56B4E9"))+
  facet_grid(source~bacteria) +
  xlab("")+theme_light()+theme(axis.text.x=element_blank())


a
fit <- aov(Shannon_fun~paste0(Treatment,"_",Species), data =data)

anova(fit)
TukeyHSD(fit)
```

##Plots
```{r}
data <- AMVG42_230505_ITS_clean$samples 

##overall plot
data <- AMVG42_230505_ITS_clean$samples
overall_fun <- data %>%
 filter(Time %in% c("t21", "t27", "t71")) %>%
# filter(Treatment != "C0")  %>%
 ggplot() +
  aes(x=Treatment, y = Shannon_fun, color = Treatment) +
  geom_boxplot()+
   geom_jitter(shape = "circle", alpha = 0.5) +
  scale_color_manual(values = c( 
      D1 = "#8CB425", D2 = "#ECE320", D3 = "#DF5220", C0 = "#219ebc")) + 
  theme_minimal(base_size = 14) +
   ylab("Alpha diversity")+
  xlab("")+
  stat_compare_means(method = "kruskal.test", label.y = c(90))+
  theme(legend.position = "bottom", axis.text.x = element_blank())+
   ggtitle(expression(italic("Fungi"))) 
overall_fun

##################
#during drought
##################

data <- data %>%
 #filter(Treatment != "C0") %>%
 filter(!is.na(Species)) %>%
 filter(Time %in% c("t21", "t27", "t71") | is.na(Time))

ef <- data %>%
 filter(Species == "Epe.fal") %>%
# filter(Treatment != "C0")  %>%
 ggplot() +
  aes(x = Treatment, y = Shannon_fun, color = Treatment) +
  geom_boxplot()+
   geom_jitter(shape = "circle", alpha = 0.5) +
  scale_color_manual(values = c( 
      D1 = "#8CB425", D2 = "#ECE320", D3 = "#DF5220",  C0 = "#219ebc")) + 
  theme_minimal(base_size = 14) +
   ylab("Alpha diversity")+
  xlab("")+
  stat_compare_means(method = "kruskal.test", label.y = c(55), label.x = 1.5)+
   #stat_compare_means(comparisons = list(c("D1", "D2"), c("D1", "D3"), c("D2", "D3")), method = "wilcox.test", label.y = c(30,40,35))+
  ggtitle(expression(italic("E. falcata"))) +
  theme(legend.position = "none", axis.text.x = element_blank())+
  ylim(0,60)
ef


ih <- data %>%
 filter(Species == "Iry.hos") %>%
# filter(Treatment != "C0")  %>%
 ggplot() +
  aes(x = Treatment, y = Shannon_fun, color = Treatment) +
  geom_boxplot()+
   geom_jitter(shape = "circle", alpha = 0.5) +
  scale_color_manual(values = c( 
      D1 = "#8CB425", D2 = "#ECE320", D3 = "#DF5220", C0 = "#219ebc")) + 
  theme_minimal(base_size = 14) +
   ylab("")+
  xlab("")+
  stat_compare_means(method = "kruskal.test", label.y = c(95), label.x = 1.5)+
   # stat_compare_means(comparisons = list(c("D1", "D2"), c("D1", "D3"), c("D2", "D3")), method = "wilcox.test", label.y = c(65,80,70))+
  ggtitle(expression(italic("I. hostmanii"))) +
  theme(legend.position = "none", axis.text.x = element_blank())+
  ylim(0,100)
ih

jc <- data %>%
 filter(Species == "Jac.cop") %>%
# filter(Treatment != "C0")  %>%
  ggplot() +
  aes(x = Treatment, y = Shannon_fun, color = Treatment) +
  geom_boxplot()+
   geom_jitter(shape = "circle", alpha = 0.5) +
  scale_color_manual(values = c( 
      D1 = "#8CB425", D2 = "#ECE320", D3 = "#DF5220",  C0 = "#219ebc")) + 
  theme_minimal(base_size = 14) +
   ylab("")+
  xlab("")+
  stat_compare_means(method = "kruskal.test", label.y = c(115), label.x = 1.5)+
  # stat_compare_means(comparisons = list(c("D1", "D2"), c("D1", "D3"), c("D2", "D3")), method = "wilcox.test", label.y = c(70,90,80))+
  ggtitle(expression(italic("J. copaia"))) +
  theme(legend.position = "none", axis.text.x = element_blank())+
  ylim(0,130)
jc

po <- data %>%
 filter(Species == "Pte.off") %>%
 #filter(Treatment != "C0")  %>%
 ggplot() +
  aes(x = Treatment, y = Shannon_fun, color = Treatment) +
  geom_boxplot()+
   geom_jitter(shape = "circle", alpha = 0.5) +
  scale_color_manual(values = c( 
      D1 = "#8CB425", D2 = "#ECE320", D3 = "#DF5220",  C0 = "#219ebc")) + 
  theme_minimal(base_size = 14) +
   ylab("")+
  xlab("")+
  stat_compare_means(method = "kruskal.test", label.y = c(50), label.x = 1.5)+
   stat_compare_means(comparisons = list(c("D1", "D2"), c("D1", "D3")), method = "wilcox.test", label.y = c(25,40))+
  ggtitle(expression(italic("P. officinalis"))) +
  theme(legend.position = "none", axis.text.x = element_blank())+
  ylim(0,60)
po

sg <- data %>%
 filter(Species == "Sym.off") %>%
# filter(Treatment != "C0")  %>%
 ggplot() +
  aes(x = Treatment, y = Shannon_fun, color = Treatment)+  geom_boxplot()+
   geom_jitter(shape = "circle", alpha = 0.5) +
  scale_color_manual(values = c( 
      D1 = "#8CB425", D2 = "#ECE320", D3 = "#DF5220", C0 = "#219ebc")) + 
  theme_minimal(base_size = 14) +
   ylab("Alpha diversity")+
  xlab("")+
  stat_compare_means(method = "kruskal.test", label.y = c(115), label.x = 1.5)+
  stat_compare_means(comparisons = list(c("D2", "D3"), c("C0", "D2")), method = "wilcox.test", label.y = c(75, 90))+
  ggtitle(expression(italic("S. globulifera"))) +
  theme(legend.position = "none")+
  ylim(0,130)
sg

tm <- data %>%
 filter(Species == "Tac.mel") %>%
 #filter(Treatment != "C0")  %>%
 ggplot() +
  aes(x = Treatment, y = Shannon_fun, color = Treatment) +
  geom_boxplot()+
   geom_jitter(shape = "circle", alpha = 0.5) +
  scale_color_manual(values = c( 
      D1 = "#8CB425", D2 = "#ECE320", D3 = "#DF5220", C0 = "#219ebc")) + 
  theme_minimal(base_size = 14) +
   ylab("")+
  xlab("")+
  stat_compare_means(method = "kruskal.test", label.y = c(87), label.x = 1.5) +
  ggtitle(expression(italic("T. melinonii"))) +
  theme(legend.position = "none")+
  ylim(0,90)
tm

vs <- data %>%
 filter(Species == "Vir.sur") %>%
# filter(Treatment != "C0")  %>%
 ggplot() +
  aes(x = Treatment, y = Shannon_fun, color = Treatment) +
  geom_boxplot()+
   geom_jitter(shape = "circle", alpha = 0.5) +
  scale_color_manual(values = c( 
      D1 = "#8CB425", D2 = "#ECE320", D3 = "#DF5220",  C0 = "#219ebc")) + 
  theme_minimal(base_size = 14) +
  xlab("")+
  ylab("")+
  stat_compare_means(method = "kruskal.test", label.y = c(60), label.x = 1.5)+
  ggtitle(expression(italic("V. surinamensis"))) +
 theme(legend.position = "none")+
  # theme(legend.position = "bottom", legend.box = "vertical", legend.direction = "vertical")+
  ylim(0,70)
vs

legend <- get_legend(vs)

alpha_fun_drought <- ggarrange(ef, ih, jc, po, sg, tm, vs, as_ggplot(legend),nrow = 2, ncol=4, widths = c(0.2,0.2,0.2,0.2))

alpha_fun_drought <- annotate_figure(alpha_fun_drought, top = text_grob("B. Fungi", color = "black", face = "bold", size = 20, just = "right" , hjust = 4.2, vjust = 0.5))

ggsave(filename = "./results/microbio/alpha_fun_drought_withC0.jpeg", plot = alpha_fun_drought, width = 10, height = 6)

```

combine with bacteria for one big plot

```{r}
alpha_drought <- ggarrange(alpha_bac_drought, alpha_fun_drought, nrow=2, ncol=1)

alpha_drought

ggsave(filename = "./results/microbio/alpha_drought.jpeg", plot = alpha_drought, width = 10, height = 12)
```


```{r}

##################
# after drought 
##################

#prepare data
data <- AMVG42_230505_ITS_clean$samples
data <- data %>%  filter(Treatment != "C0") %>% filter(!is.na(Species)) %>%
  mutate(Period = ifelse(Time %in% c("t21","t27", "t71"), "Drought", 
                  ifelse(Time %in% c("t51", "t57","t101"), "Recovery", NA)))
data$Period <- as.factor(data$Period)
data$Treatment <- as.factor(data$Treatment)

#epefal
ef <- data %>% filter(Species == "Epe.fal")
 ef_plot <- ggboxplot(
  ef, x= "Period" , y = "Shannon_fun", fill = "Period", facet.by = "Treatment", alpha =0.7) +           geom_jitter(shape = "circle", alpha = 0.5) +
  scale_fill_manual(values = c(Drought = "#ff9e00", Recovery = "#219ebc"))+
  theme_classic() +
  theme(axis.text.x = element_blank(), axis.title.x = element_blank())+
  ylab("Alpha diversity")+
  theme(legend.position = "none")
ef_plot
stat.test <- data %>%
  filter(Species == "Epe.fal") %>%
  filter(Treatment != "D3") %>%
  group_by(Treatment) %>%
  t_test(Shannon_fun ~ Period) %>%
  adjust_pvalue(method = "bonferroni") %>%
  add_significance()
stat.test 
stat.test <- stat.test %>% add_xy_position(x = "Period") # Make facet and add p-values
ef_plot <- ef_plot + stat_pvalue_manual(stat.test)
ef_plot <- annotate_figure(ef_plot, left = text_grob("E. falcata       ", color = "black", face = "italic", size = 20))

#iryhos
ih <- data %>% filter(Species == "Iry.hos") 
ih_plot <- ggboxplot(
  ih, x= "Period" , y = "Shannon_fun", fill = "Period", facet.by = "Treatment", alpha =0.7) + 
  geom_jitter(shape = "circle", alpha = 0.5) +
  scale_fill_manual(values = c(Drought = "#ff9e00", Recovery = "#219ebc"))+
  theme_classic() +
  theme(axis.text.x = element_blank(), axis.title.x = element_blank())+
  ylab("Alpha diversity")+
  theme(legend.position = "none")
ih_plot
stat.test <- ih %>%
  group_by(Species, Treatment) %>%
  t_test(Shannon_fun ~ Period) %>%
  adjust_pvalue(method = "bonferroni") %>%
  add_significance()
stat.test 
stat.test <- stat.test %>% add_xy_position(x = "Period") 
ih_plot <- ih_plot + stat_pvalue_manual(stat.test)
ih_plot <- annotate_figure(ih_plot, right = text_grob("I. hostmanii", color = "black", face = "italic", size = 20))

#jaccop
jc <- data %>% filter(Species == "Jac.cop") 
 jc_plot <- ggboxplot(
   jc, x= "Period" , y = "Shannon_fun", fill = "Period", facet.by = "Treatment", alpha =0.7) + 
   geom_jitter(shape = "circle", alpha = 0.5) +
   scale_fill_manual(values = c(Drought = "#ff9e00", Recovery = "#219ebc"))+
  theme_classic() +
  theme(axis.text.x = element_blank(), axis.title.x = element_blank())+
  ylab("Alpha diversity")+
  theme(legend.position = "none")
jc_plot
stat.test <- jc %>%
  group_by(Species, Treatment) %>%
  t_test(Shannon_fun ~ Period) %>%
  adjust_pvalue(method = "bonferroni") %>%
  add_significance()
stat.test 
stat.test <- stat.test %>% add_xy_position(x = "Period")
jc_plot <- jc_plot + stat_pvalue_manual(stat.test)
jc_plot <- annotate_figure(jc_plot, left = text_grob("J. copaia        ", color = "black", face = "italic", size = 20))

#pteoff
po <- data %>% filter(Species == "Pte.off") 
 po_plot <- ggboxplot(
   po, x= "Period" , y = "Shannon_fun", fill = "Period", facet.by = "Treatment", alpha =0.7) + 
  geom_jitter(shape = "circle", alpha = 0.5) +
  scale_fill_manual(values = c(Drought = "#ff9e00", Recovery = "#219ebc"))+
  theme_classic() +
  theme(axis.text.x = element_blank(), axis.title.x = element_blank())+
  ylab("Alpha diversity")+
  theme(legend.position = "none")
po_plot
stat.test <- po %>%
  group_by(Species, Treatment) %>%
  t_test(Shannon_fun ~ Period) %>%
  adjust_pvalue(method = "bonferroni") %>%
  add_significance()
stat.test 
stat.test <- stat.test %>% add_xy_position(x = "Period")
po_plot <- po_plot + stat_pvalue_manual(stat.test)
po_plot <- annotate_figure(po_plot, right = text_grob("P. officinalis", color = "black", face = "italic", size = 20))

#symglo
sg <- data %>% filter(Species == "Sym.off") 
 sg_plot <- ggboxplot(
   sg, x= "Period" , y = "Shannon_fun", fill = "Period", facet.by = "Treatment", alpha =0.7) + 
  geom_jitter(shape = "circle", alpha = 0.5) +
  scale_fill_manual(values = c(Drought = "#ff9e00", Recovery = "#219ebc"))+
  theme_classic() +
  theme(axis.text.x = element_blank(), axis.title.x = element_blank())+
  ylab("Alpha diversity")+
  theme(legend.position = "none")
sg_plot
stat.test <- sg %>%
  group_by(Species, Treatment) %>%
  t_test(Shannon_fun ~ Period) %>%
  adjust_pvalue(method = "bonferroni") %>%
  add_significance()
stat.test 
stat.test <- stat.test %>% add_xy_position(x = "Period")
sg_plot <- sg_plot + stat_pvalue_manual(stat.test)
sg_plot <- annotate_figure(sg_plot, left = text_grob("S. globulifera  ", color = "black", face = "italic", size = 20))

#tacmel
tm <- data %>% filter(Species == "Tac.mel") 
tm_plot <- ggboxplot(
  tm, x= "Period" , y = "Shannon_fun", fill = "Period", facet.by = "Treatment", alpha =0.7) + 
  geom_jitter(shape = "circle", alpha = 0.5) +
  scale_fill_manual(values = c(Drought = "#ff9e00", Recovery = "#219ebc"))+
  theme_classic() +
  theme(axis.text.x = element_blank(), axis.title.x = element_blank())+
  ylab("Alpha diversity")+
  theme(legend.position = "none")
tm_plot
stat.test <- tm %>%
  group_by(Species, Treatment) %>%
  t_test(Shannon_fun ~ Period) %>%
  adjust_pvalue(method = "bonferroni") %>%
  add_significance()
stat.test 
stat.test <- stat.test %>% add_xy_position(x = "Period")
tm_plot <- tm_plot + stat_pvalue_manual(stat.test)
tm_plot <- annotate_figure(tm_plot, right = text_grob("T. melinonii", color = "black", face = "italic", size = 20))

#virsur
vs <- data %>% filter(Species == "Vir.sur") 
vs_plot <- ggboxplot(
  vs, x= "Period" , y = "Shannon_fun", fill = "Period", facet.by = "Treatment", alpha =0.7) + 
  geom_jitter(shape = "circle", alpha = 0.5) +
  scale_fill_manual(values = c(Drought = "#ff9e00", Recovery = "#219ebc"))+
  theme_classic() +
  theme(axis.text.x = element_blank(), axis.title.x = element_blank())+
  ylab("Alpha diversity")+
  theme(legend.position = "none", legend.text = element_text(size = 16),
    legend.title = element_text(size = 20))
vs_plot
stat.test <- vs %>%
  filter(!(Treatment == "D3")) %>%
  group_by(Species, Treatment) %>%
  t_test(Shannon_fun ~ Period) %>%
  adjust_pvalue(method = "bonferroni") %>%
  add_significance()
stat.test 
stat.test <- stat.test %>% add_xy_position(x = "Period") 
vs_plot <- vs_plot + stat_pvalue_manual(stat.test)
vs_plot <- annotate_figure(vs_plot, left = text_grob("V. surinamensis", color = "black", face = "italic", size = 20))

#get legend
legend <- get_legend(vs_plot) #get legend work with no complicated plots fso just grab the legend when plot is simpler
as_ggplot(legend)

#plot all
alpha_recov_fun <- ggarrange(ef_plot, ih_plot, jc_plot, po_plot, sg_plot, tm_plot, vs_plot, as_ggplot(legend), ncol=2, nrow = 4)
alpha_recov_fun
alpha_recov_fun <- annotate_figure(alpha_recov_fun, top = text_grob("Fungal alpha diversity before and after drought", color = "black", face = "bold", size = 24))
ggsave(filename = "./results/microbio/alpha_fun_recov.jpeg", plot = alpha_recov_fun, width = 12, height = 10)

```

##Resistance of alpha

compare with C0
```{r}
resistance_D1 <- function(data) {
  resistance_df <- data.frame(Species = character(),
                              Treatment = character(),
                              Resistance = numeric(),
                              stringsAsFactors = FALSE)
  data <- data %>% filter(Time == "t21")
  unique_species <- unique(data$Species)
  
  for (species in unique_species) {
    species_data <- data[data$Species == species, ]
    
    CT <- mean(species_data$Shannon[species_data$Treatment == "C0"])
    treatments <- "D1"
    
    for (treatment in treatments) {
      DT <- abs(CT - mean(species_data$Shannon[species_data$Treatment == treatment]))
      
      if (length(DT) > 0) {
        Resistance <- 1 - ((2 * abs(DT)) / (CT + abs(DT)))
        
        species_resistance <- data.frame(Species = species,
                                         Treatment = treatment,
                                         Resistance = Resistance,
                                         stringsAsFactors = FALSE)
        
        resistance_df <- rbind(resistance_df, species_resistance)
      }
    }
  }
  
  resistance_df
}
```

```{r}
resistance_D2 <- function(data) {
  resistance_df <- data.frame(Species = character(),
                              
                              Treatment = character(),
                              Resistance = numeric(),
                              stringsAsFactors = FALSE)
  data <- data %>% filter(Time == "t27")
  unique_species <- unique(data$Species)
  
  for (species in unique_species) {
    species_data <- data[data$Species == species, ]
    
    CT <- mean(species_data$Shannon[species_data$Treatment == "C0"])
     treatments <- "D2"
    
    for (treatment in treatments) {
      DT <- abs(CT - mean(species_data$Shannon[species_data$Treatment == treatment]))
      
      if (length(DT) > 0) {
        Resistance <- 1 - ((2 * abs(DT)) / (CT + abs(DT)))
        
        species_resistance <- data.frame(Species = species,
                                         Treatment = treatment,
                                         Resistance = Resistance,
                                         stringsAsFactors = FALSE)
        
        resistance_df <- rbind(resistance_df, species_resistance)
      }
    }
  }
  
  resistance_df
}

```



```{r}
resistance_D3 <- function(data) {
  resistance_df <- data.frame(Species = character(),
                              
                              Treatment = character(),
                              Resistance = numeric(),
                              stringsAsFactors = FALSE)
  data <- data %>% filter(Time == "t71")
  unique_species <- unique(data$Species)
  
  for (species in unique_species) {
    species_data <- data[data$Species == species, ]
    
    CT <- mean(species_data$Shannon[species_data$Treatment == "C0"])
      treatments <- "D3"
    
    for (treatment in treatments) {
      DT <- abs(CT - mean(species_data$Shannon[species_data$Treatment == treatment]))
      
      if (length(DT) > 0) {
        Resistance <- 1 - ((2 * abs(DT)) / (CT + abs(DT)))
        
        species_resistance <- data.frame(Species = species,
                                         Treatment = treatment,
                                         Resistance = Resistance,
                                         stringsAsFactors = FALSE)
        
        resistance_df <- rbind(resistance_df, species_resistance)
      }
    }
  }
  
  resistance_df
}

```

format plot
```{r}

data <- AMVG42_230505_ITS_clean$samples
fungi1 <- resistance_D1(data)
fungi2 <- resistance_D2(data)
fungi3 <- resistance_D3(data)

fungi <- rbind(fungi1, fungi2, fungi3)

#plot
#for the labels
fungi$facet_label <- NA
fungi$facet_label[fungi$Species == "Epe.fal"] <- "E. falcata"
fungi$facet_label[fungi$Species == "Iry.hos"] <- "I. hostmanii"
fungi$facet_label[fungi$Species == "Jac.cop"] <- "J. copaia"
fungi$facet_label[fungi$Species == "Pte.off"] <- "P. officinalis"
fungi$facet_label[fungi$Species == "Sym.off"] <- "S. globulifera"
fungi$facet_label[fungi$Species == "Tac.mel"] <- "T. melinonii"
fungi$facet_label[fungi$Species == "Vir.sur"] <- "V. surinamensis"

ef <- fungi %>%
  filter(Species == "Epe.fal") %>%
  ggplot() +
 aes(x = Treatment, y = Resistance,  colour = Treatment) +
 ylab("Resistance")+
 geom_point(shape = "circle", size = 4L,  alpha = 0.5) +
 scale_color_manual(values = c( 
      D2 = "#ECE320", D3 = "#DF5220")) +
 labs(title = "E. falcata") +
 xlab(" ")+
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
 theme_minimal(base_size =20)+
 theme(title = element_text(face = "italic"), plot.title =   element_text(hjust = 0.5))+
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
 theme(legend.position = "none")

ih <- fungi %>%
  filter(Species == "Iry.hos") %>%
  ggplot() +
 aes(x = Treatment, y = Resistance,  colour = Treatment) +
 ylab("Resistance")+
 geom_point(shape = "circle", size = 4L,  alpha = 0.5) +
 scale_color_manual(values = c( 
      D2 = "#ECE320", D3 = "#DF5220")) +
 labs(title = "I. hostmanii") +
 xlab(" ")+
  ylab(" ")+
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
 theme_minimal(base_size =20)+
 theme(title = element_text(face = "italic"), plot.title =   element_text(hjust = 0.5))+
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
   theme(legend.position = "none")

jc <- fungi %>%
  filter(Species == "Jac.cop") %>%
  ggplot() +
 aes(x = Treatment, y = Resistance,  colour = Treatment) +
 ylab("Resistance")+
 geom_point(shape = "circle", size = 4L,  alpha = 0.5) +
 scale_color_manual(values = c( 
      D2 = "#ECE320", D3 = "#DF5220")) +
 labs(title = "J. copaia") +
 xlab(" ")+
  ylab(" ")+
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
 theme_minimal(base_size =20)+
 theme(title = element_text(face = "italic"), plot.title =   element_text(hjust = 0.5))+
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
   theme(legend.position = "none")

po <- fungi %>%
  filter(Species == "Pte.off") %>%
  ggplot() +
 aes(x = Treatment, y = Resistance,  colour = Treatment) +
 ylab("Resistance")+
 geom_point(shape = "circle", size = 4L,  alpha = 0.5) +
 scale_color_manual(values = c( 
      D2 = "#ECE320", D3 = "#DF5220")) +
 labs(title = "P. officinalis") +
 xlab(" ")+
  ylab(" ")+
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
 theme_minimal(base_size =20)+
 theme(title = element_text(face = "italic"), plot.title =   element_text(hjust = 0.5))+
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
   theme(legend.position = "none")

sg <- fungi %>%
  filter(Species == "Sym.off") %>%
  ggplot() +
 aes(x = Treatment, y = Resistance,  colour = Treatment) +
 ylab("Resistance")+
 geom_point(shape = "circle", size = 4L,  alpha = 0.5) +
 scale_color_manual(values = c( 
      D2 = "#ECE320", D3 = "#DF5220")) +
 labs(title = "S. globulifera") +
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
 theme_minimal(base_size =20)+
 theme(title = element_text(face = "italic"), plot.title =   element_text(hjust = 0.5))+
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
   theme(legend.position = "none")


tm <- fungi %>%
  filter(Species == "Tac.mel") %>%
  ggplot() +
 aes(x = Treatment, y = Resistance,  colour = Treatment) +
 ylab("Resistance")+
 geom_point(shape = "circle", size = 4L,  alpha = 0.5) +
 scale_color_manual(values = c( 
      D2 = "#ECE320", D3 = "#DF5220")) +
 labs(title = "T. melinonii") +
  ylab(" ")+
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
 theme_minimal(base_size =20)+
 theme(title = element_text(face = "italic"), plot.title =   element_text(hjust = 0.5))+
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
   theme(legend.position = "none")

vs <- fungi %>%
  filter(Species == "Vir.sur") %>%
  ggplot() +
 aes(x = Treatment, y = Resistance,  colour = Treatment) +
 ylab("Resistance")+
 geom_point(shape = "circle", size = 4L,  alpha = 0.5) +
 scale_color_manual(values = c( 
      D2 = "#ECE320", D3 = "#DF5220")) +
 labs(title = "V. surinamensis") +
  ylab(" ")+
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
 theme_minimal(base_size =20)+
 theme(title = element_text(face = "italic"), plot.title =   element_text(hjust = 0.5))+
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
   theme(legend.position = "none")

#get legend
legend <- get_legend(vs)
legend <- as_ggplot(legend)


resistance_fun <- ggarrange(ef, ih, jc, po, sg, tm, vs, legend, nrow = 2, ncol = 4)
resistance_fun
ggsave(filename = "./results/microbio/resistance_fun.jpeg", plot = resistance_fun, width = 12, height = 10)
```


```{r}

#compare with D1
resistance <- function(data) {
  resistance_df <- data.frame(Species = character(),
                              Treatment = character(),
                              Resistance = numeric(),
                              stringsAsFactors = FALSE)
  
  unique_species <- unique(data$Species)
  
  data <- data %>% filter(Treatment != "C0")
  
  for (species in unique_species) {
    species_data <- data[data$Species == species, ]
    
    CT <- mean(species_data$Shannon[species_data$Treatment == "D1"])
    treatments <- unique(species_data$Treatment[species_data$Treatment != "D1"])
    
    for (treatment in treatments) {
      DT <- abs(CT - mean(species_data$Shannon[species_data$Treatment == treatment]))
      
      if (length(DT) > 0) {
        Resistance <- 1 - ((2 * abs(DT)) / (CT + abs(DT)))
        
        species_resistance <- data.frame(Species = species,
                                         Treatment = treatment,
                                         Resistance = Resistance,
                                         stringsAsFactors = FALSE)
        
        resistance_df <- rbind(resistance_df, species_resistance)
      }
    }
  }
  
  resistance_df
}

data <- AMVG42_230505_ITS_clean$samples
fungi <- resistance_D1(data)


#plot
#for the labels
fungi$facet_label <- NA
fungi$facet_label[fungi$Species == "Epe.fal"] <- "E. falcata"
fungi$facet_label[fungi$Species == "Iry.hos"] <- "I. hostmanii"
fungi$facet_label[fungi$Species == "Jac.cop"] <- "J. copaia"
fungi$facet_label[fungi$Species == "Pte.off"] <- "P. officinalis"
fungi$facet_label[fungi$Species == "Sym.off"] <- "S. globulifera"
fungi$facet_label[fungi$Species == "Tac.mel"] <- "T. melinonii"
fungi$facet_label[fungi$Species == "Vir.sur"] <- "V. surinamensis"

ef <- fungi %>%
  filter(Species == "Epe.fal") %>%
  ggplot() +
 aes(x = Treatment, y = Resistance,  colour = Treatment) +
 ylab("Resistance")+
 geom_point(shape = "circle", size = 4L,  alpha = 0.5) +
 scale_color_manual(values = c( 
      D2 = "#ECE320", D3 = "#DF5220")) +
 labs(title = "E. falcata") +
 xlab(" ")+
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
 theme_minimal(base_size =20)+
 theme(title = element_text(face = "italic"), plot.title =   element_text(hjust = 0.5))+
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
 theme(legend.position = "none")

ih <- fungi %>%
  filter(Species == "Iry.hos") %>%
  ggplot() +
 aes(x = Treatment, y = Resistance,  colour = Treatment) +
 ylab("Resistance")+
 geom_point(shape = "circle", size = 4L,  alpha = 0.5) +
 scale_color_manual(values = c( 
      D2 = "#ECE320", D3 = "#DF5220")) +
 labs(title = "I. hostmanii") +
 xlab(" ")+
  ylab(" ")+
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
 theme_minimal(base_size =20)+
 theme(title = element_text(face = "italic"), plot.title =   element_text(hjust = 0.5))+
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
   theme(legend.position = "none")

jc <- fungi %>%
  filter(Species == "Jac.cop") %>%
  ggplot() +
 aes(x = Treatment, y = Resistance,  colour = Treatment) +
 ylab("Resistance")+
 geom_point(shape = "circle", size = 4L,  alpha = 0.5) +
 scale_color_manual(values = c( 
      D2 = "#ECE320", D3 = "#DF5220")) +
 labs(title = "J. copaia") +
 xlab(" ")+
  ylab(" ")+
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
 theme_minimal(base_size =20)+
 theme(title = element_text(face = "italic"), plot.title =   element_text(hjust = 0.5))+
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
   theme(legend.position = "none")

po <- fungi %>%
  filter(Species == "Pte.off") %>%
  ggplot() +
 aes(x = Treatment, y = Resistance,  colour = Treatment) +
 ylab("Resistance")+
 geom_point(shape = "circle", size = 4L,  alpha = 0.5) +
 scale_color_manual(values = c( 
      D2 = "#ECE320", D3 = "#DF5220")) +
 labs(title = "P. officinalis") +
 xlab(" ")+
  ylab(" ")+
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
 theme_minimal(base_size =20)+
 theme(title = element_text(face = "italic"), plot.title =   element_text(hjust = 0.5))+
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
   theme(legend.position = "none")

sg <- fungi %>%
  filter(Species == "Sym.off") %>%
  ggplot() +
 aes(x = Treatment, y = Resistance,  colour = Treatment) +
 ylab("Resistance")+
 geom_point(shape = "circle", size = 4L,  alpha = 0.5) +
 scale_color_manual(values = c( 
      D2 = "#ECE320", D3 = "#DF5220")) +
 labs(title = "S. globulifera") +
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
 theme_minimal(base_size =20)+
 theme(title = element_text(face = "italic"), plot.title =   element_text(hjust = 0.5))+
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
   theme(legend.position = "none")


tm <- fungi %>%
  filter(Species == "Tac.mel") %>%
  ggplot() +
 aes(x = Treatment, y = Resistance,  colour = Treatment) +
 ylab("Resistance")+
 geom_point(shape = "circle", size = 4L,  alpha = 0.5) +
 scale_color_manual(values = c( 
      D2 = "#ECE320", D3 = "#DF5220")) +
 labs(title = "T. melinonii") +
  ylab(" ")+
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
 theme_minimal(base_size =20)+
 theme(title = element_text(face = "italic"), plot.title =   element_text(hjust = 0.5))+
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
   theme(legend.position = "none")

vs <- fungi %>%
  filter(Species == "Vir.sur") %>%
  ggplot() +
 aes(x = Treatment, y = Resistance,  colour = Treatment) +
 ylab("Resistance")+
 geom_point(shape = "circle", size = 4L,  alpha = 0.5) +
 scale_color_manual(values = c( 
      D2 = "#ECE320", D3 = "#DF5220")) +
 labs(title = "V. surinamensis") +
  ylab(" ")+
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
 theme_minimal(base_size =20)+
 theme(title = element_text(face = "italic"), plot.title =   element_text(hjust = 0.5))+
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
   theme(legend.position = "none")

#get legend
legend <- get_legend(vs)
legend <- as_ggplot(legend)


resistance_fun <- ggarrange(ef, ih, jc, po, sg, tm, vs, legend, nrow = 2, ncol = 4)
resistance_fun
ggsave(filename = "./results/microbio/resistance_fun.jpeg", plot = resistance_fun, width = 12, height = 10)
```

# Rec#overy
```{r}
data <- AMVG42_230505_ITS_clean$samples
fungi1 <- Recovery_D1(data)
fungi2 <- Recovery_D2(data)
fungi3 <- Recovery_D3(data)

fungi <- rbind(fungi1, fungi2, fungi3)
write.csv(x = fungi, file = "fungi_recovery_index.csv")
```


## beta diversity

```{r}
data_ITS <- AMVG42_230505_ITS_clean
rownames(data_ITS$samples) <- data_ITS$samples$sample_id
str(data_ITS)

nmds <- metaMDS(data_ITS$reads,distance = "bray")
envfit_ITS <- envfit(nmds, data_ITS$samples %>% dplyr::select(Treatment, Species, Time), na.rm = TRUE) 
plot(nmds)
site.scrs <- as.data.frame(scores(nmds, display = "sites")) #save NMDS results into dataframe
site.scrs <- cbind(site.scrs, Treatment = data_ITS$samples$Treatment) #add grouping variable "Management" to dataframe
site.scrs <- cbind(site.scrs, Time = data_ITS$samples$Time)
site.scrs <- cbind(site.scrs, Species = data_ITS$samples$Species)
head(site.scrs)


nmds.plot.dune <- ggplot(site.scrs, aes(x=NMDS1, y=NMDS2))+ #sets up the plot
  geom_point(aes(NMDS1, NMDS2, colour = factor(Treatment)), size = 2)+ #adds site points to plot, shape determined by Landuse, colour determined by Management
  coord_fixed()+
  theme_classic()+ 
  theme(panel.background = element_rect(fill = NA, colour = "black", size = 1, linetype = "solid"))+
  labs(colour = "Treatment")+ 
  theme(legend.position = "right", legend.text = element_text(size = 12), legend.title = element_text(size = 12), axis.text = element_text(size = 10)) # add legend at right of plot
nmds.plot.dune
```


##Beta - boxplots
```{r}
data_ITS <-  AMVG42_230505_ITS_clean
#bray_curtis_dist <- vegan::vegdist(decostand(data_ITS$reads, method= "hellinger" ), method = "bray")

hill <- hill_taxa_parti_pairwise(data_ITS$reads,q=1,output="matrix", pairs = "full") 
hill.beta  <- hill$TD_beta
hill.beta.dist <- as.dist(hill.beta)


hill.beta.dist <- hill.beta.dist %>%
  as.matrix() %>%
  as_tibble(rownames = "sample_id") %>%
  pivot_longer(-sample_id) %>%
  filter(sample_id < name) %>% #don't repeat the same compared to self
  mutate(indv_a = str_replace(sample_id, ".*__", "")) %>%
  mutate(indv_b = str_replace(name, ".*__", ""))
 
 
info <- data_ITS$samples %>%
  dplyr::select(sample_id, Treatment, Time, Species, code)

dim(info) #266   5
dim(hill.beta.dist) #35245     5

dist_merged <- merge(hill.beta.dist, info, by.x = "indv_a", by.y = "code", all.x = TRUE) %>%
  rename(indv_a_species = Species) %>%
  rename(indv_a_time = Time) %>%
  rename(indv_a_treatment = Treatment) %>%
  dplyr::select(-sample_id.x, -sample_id.y)

dist_merged2 <- merge(dist_merged, info, by.x = "indv_b", by.y = "code", all.x = TRUE) %>%
  rename(indv_b_species = Species) %>%
  rename(indv_b_time = Time) %>%
  rename(indv_b_treatment = Treatment) %>%
  dplyr::select(-sample_id, -name)

beta<- dist_merged2 %>% 
  dplyr::filter(indv_b_species == indv_a_species) %>% #compare within same species
  filter(indv_b_time == indv_a_time) %>%#compare with the same time period
  filter(indv_b_treatment ==indv_a_treatment )
  
write.csv(x = beta, file = "beta_fun.csv")

library(dplyr)
library(ggplot2)

plot_control <- beta %>%
 filter(indv_a_time %in% c("t21", "t27", "t71")) %>%
 filter(indv_b_time %in% c("t21", "t27", "t71")) %>%
 ggplot() +
 aes(x = indv_b_treatment, y = value, fill = indv_b_treatment) +
 geom_boxplot() +
 scale_fill_manual(values = c(C0 = "#6DA0F8", 
D1 = "#8CB425", D2 = "#ECE320", D3 = "#DF5220")) +
 labs(y = "Beta-diversity", fill = "Treatment") +
 theme_classic() +
 theme(axis.title.y = element_text(size = 15L, face = "bold")) +
 facet_wrap(vars(indv_b_species))+
  ggtitle("Bacteria")

#for the labels
beta$facet_label <- NA
beta$facet_label[beta$indv_b_species == "Epe.fal"] <- "E. falcata"
beta$facet_label[beta$indv_b_species == "Iry.hos"] <- "I. hostmanii"
beta$facet_label[beta$indv_b_species == "Jac.cop"] <- "J. copaia"
beta$facet_label[beta$indv_b_species == "Pte.off"] <- "P. officinalis"
beta$facet_label[beta$indv_b_species == "Sym.off"] <- "S. globulifera"
beta$facet_label[beta$indv_b_species == "Tac.mel"] <- "T. melinonii"
beta$facet_label[beta$indv_b_species == "Vir.sur"] <- "V. surinamensis"

plot_withoutC0<- beta %>%
 filter(!(indv_a_treatment %in% "C0")) %>%
 filter(indv_a_time %in% c("t21", "t27", "t71")) %>%
 filter(!(indv_b_treatment %in% 
 "C0")) %>%
 filter(indv_b_time %in% c("t21", "t27", "t71")) %>%
 ggplot() +
 aes(x = indv_b_treatment, y = value, fill = indv_b_treatment) +
 geom_boxplot(alpha = 0.6) +  # Adding transparency and softer color for boxplots
  geom_jitter(alpha = 0.5, fill = "gray80") +   # Adding transparency and softer color for jitter points
 scale_fill_manual(values = c( 
D1 = "#8CB425", D2 = "#ECE320", D3 = "#DF5220")) +
 labs(y = "Beta-diversity", fill = "Treatment") +
 theme_classic() +
 theme(
    axis.title.y = element_text(size = 15L, face = "bold"),
    axis.title.x = element_blank(),
   # panel.background = element_rect(fill = "transparent"),  # Making the panel background transparent
    strip.background = element_blank() ,# Removing background from facet labels
    strip.text = element_text(size=12, face= "italic")
   )+
 facet_wrap(vars(facet_label)) +  # Apply the custom labeller function)+
  ggtitle("Fungi")

plot_withoutC0
```

plots individual to add statisitical significance
```{r}
 #epefal
ef <- beta %>%
 filter(facet_label == "E. falcata") %>%
 filter(indv_a_time %in% c("t21", "t27", "t71")) %>%
 filter(indv_b_time %in% c("t21", "t27", "t71")) %>%
  ggplot() +
 aes(x = indv_b_treatment, y = value, fill = indv_b_treatment) +
 geom_boxplot(alpha = 0.6) + 
  geom_jitter(alpha = 0.5, fill = "gray80") +  
 scale_fill_manual(values = c( 
D1 = "#8CB425", D2 = "#ECE320", D3 = "#DF5220",C0 = "#6DA0F8")) +
 labs(y = "Beta-diversity", fill = "Treatment") +
 theme_classic(base_size = 14) +
 theme(
    axis.title.y = element_text(size = 15L, face = "bold"),
    axis.title.x = element_blank(),
    axis.text.x = element_blank())+
  stat_compare_means(aes(label = sprintf("Kurskall-Wallis, p = %1.2f", as.numeric(..p.format..))), method = "kruskal.test", label.y = c(1.95), label.x = 1.5)+
 # stat_compare_means(comparisons = list(c("D1", "D2")), method = "wilcox.test", label.y = c(1.75))+
  ggtitle(expression(italic("E. facalta"))) +
  theme(legend.position = "none")+
  ylim(1,2.1)
ef

#iryhos
ih <- beta %>%
 filter(facet_label == "I. hostmanii") %>%
 filter(indv_a_time %in% c("t21", "t27", "t71")) %>%
 filter(indv_b_time %in% c("t21", "t27", "t71")) %>%
  ggplot() +
 aes(x = indv_b_treatment, y = value, fill = indv_b_treatment) +
 geom_boxplot(alpha = 0.6) + 
  geom_jitter(alpha = 0.5, fill = "gray80") +  
 scale_fill_manual(values = c( 
D1 = "#8CB425", D2 = "#ECE320", D3 = "#DF5220",C0 = "#6DA0F8")) +
 labs(y = "Beta-diversity", fill = "Treatment") +
 theme_classic(base_size = 14) +
 theme(
    axis.title.y = element_blank(),
    axis.title.x = element_blank(), axis.text.x = element_blank())+
  stat_compare_means(aes(label = sprintf("Kurskall-Wallis, p < 0.01", as.numeric(..p.format..))),method = "kruskal.test", label.y = c(1.95), label.x = 1.5)+
 stat_compare_means(comparisons = list(c("D1", "D2"), c("D1", "D3")), label="p.signif", method = "wilcox.test", label.y = c(1.65,1.75))+
  ggtitle(expression(italic("I. hostmanii"))) +
  theme(legend.position = "none")+
   ylim(1,2.1)
ih

#jaccop
jc <- beta %>%
 filter(facet_label == "J. copaia") %>%
 filter(indv_a_time %in% c("t21", "t27", "t71")) %>%
 filter(indv_b_time %in% c("t21", "t27", "t71")) %>%
  ggplot() +
 aes(x = indv_b_treatment, y = value, fill = indv_b_treatment) +
 geom_boxplot(alpha = 0.6) + 
  geom_jitter(alpha = 0.5, fill = "gray80") +  
 scale_fill_manual(values = c( 
D1 = "#8CB425", D2 = "#ECE320", D3 = "#DF5220",C0 = "#6DA0F8")) +
 labs(y = "Beta-diversity", fill = "Treatment") +
 theme_classic(base_size = 14) +
 theme(
    axis.title.y = element_blank(),
    axis.title.x = element_blank(), axis.text.x = element_blank() )+
  stat_compare_means(method = "kruskal.test", label.y = c(1.95), label.x = 1.5)+
 # stat_compare_means(comparisons = list(c("D1", "D2"), c("D2", "D3"), c("D1", "D3")), method = "wilcox.test", label.y = c(1.5, 1.6,1.7))+
  ggtitle(expression(italic("J. copaia"))) +
  theme(legend.position = "none")+
   ylim(1,2.1)
jc

#pteoff
po <-  beta %>%
 filter(facet_label == "P. officinalis") %>%
 filter(indv_a_time %in% c("t21", "t27", "t71")) %>%
 filter(indv_b_time %in% c("t21", "t27", "t71")) %>%
  ggplot() +
 aes(x = indv_b_treatment, y = value, fill = indv_b_treatment) +
 geom_boxplot(alpha = 0.6) + 
  geom_jitter(alpha = 0.5, fill = "gray80") +  
 scale_fill_manual(values = c( 
D1 = "#8CB425", D2 = "#ECE320", D3 = "#DF5220",C0 = "#6DA0F8")) +
 labs(y = "Beta-diversity", fill = "Treatment") +
 theme_classic(base_size = 14) +
 theme(
    axis.title.y = element_blank(),
    axis.title.x = element_blank(), axis.text.x = element_blank())+
  stat_compare_means(aes(label = sprintf("Kurskall-Wallis, p < 0.01", as.numeric(..p.format..))),method = "kruskal.test", label.y = c(1.95), label.x = 1.5)+
 stat_compare_means(comparisons = list(c("C0", "D1"), c("D1", "D3")), label="p.signif", method = "wilcox.test", label.y = c( 1.65,1.75))+
  ggtitle(expression(italic("P. officinalis"))) +
  theme(legend.position = "none")+
  ylim(1,2.1)
po

#symglo
sg <-  beta %>%
 filter(facet_label == "S. globulifera") %>%
 filter(indv_a_time %in% c("t21", "t27", "t71")) %>%
 filter(indv_b_time %in% c("t21", "t27", "t71")) %>%
  ggplot() +
 aes(x = indv_b_treatment, y = value, fill = indv_b_treatment) +
 geom_boxplot(alpha = 0.6) + 
  geom_jitter(alpha = 0.5, fill = "gray80") +  
 scale_fill_manual(values = c( 
D1 = "#8CB425", D2 = "#ECE320", D3 = "#DF5220",C0 = "#6DA0F8")) +
 labs(y = "Beta-diversity", fill = "Treatment") +
 theme_classic(base_size = 14) +
 theme(
    axis.title.y = element_text(size = 15L, face = "bold"),
    axis.title.x = element_blank(), axis.text.x = element_blank())+
  stat_compare_means(aes(label = sprintf("Kurskall-Wallis, p < 0.01", as.numeric(..p.format..))),method = "kruskal.test", label.y = c(1.2), label.x = 1.5)+
  stat_compare_means(comparisons = list(c("D1", "D2"), c("D2", "D3"), c("C0", "D2")), label="p.signif", method = "wilcox.test", label.y = c(1.75, 1.85,1.99))+
  ggtitle(expression(italic("S. globulifera"))) +
  theme(legend.position = "none")+
  ylim(1,2.1)
sg

#tacmel
tm <-   beta %>%
 filter(facet_label == "T. melinonii") %>%
 filter(indv_a_time %in% c("t21", "t27", "t71")) %>%
 filter(indv_b_time %in% c("t21", "t27", "t71")) %>%
  ggplot() +
 aes(x = indv_b_treatment, y = value, fill = indv_b_treatment) +
 geom_boxplot(alpha = 0.6) + 
  geom_jitter(alpha = 0.5, fill = "gray80") +  
 scale_fill_manual(values = c( 
D1 = "#8CB425", D2 = "#ECE320", D3 = "#DF5220",C0 = "#6DA0F8")) +
 labs(y = "Beta-diversity", fill = "Treatment") +
 theme_classic(base_size = 14) +
 theme(
    axis.title.y = element_blank(),
    axis.title.x = element_blank(), axis.text.x = element_blank())+
  stat_compare_means(aes(label = sprintf("Kurskall-Wallis, p = %1.2f", as.numeric(..p.format..))),method = "kruskal.test", label.y = c(1.2), label.x=1.5)+
  stat_compare_means(comparisons = list(c("D1", "D2"), c("D1", "D3")), label="p.signif", method = "wilcox.test", label.y = c(1.85, 1.99))+
  ggtitle(expression(italic("T. melinonii"))) +
  theme(legend.position = "none")+
  ylim(1,2.1)
tm

#virsur
vs <-  beta %>%
 filter(facet_label == "V. surinamensis") %>%
 filter(indv_a_time %in% c("t21", "t27", "t71")) %>%
 filter(indv_b_time %in% c("t21", "t27", "t71")) %>%
  ggplot() +
 aes(x = indv_b_treatment, y = value, fill = indv_b_treatment) +
 geom_boxplot(alpha = 0.6) + 
  geom_jitter(alpha = 0.5, fill = "gray80") +  
 scale_fill_manual(values = c( 
D1 = "#8CB425", D2 = "#ECE320", D3 = "#DF5220",C0 = "#6DA0F8")) +
 labs(y = "Beta-diversity", fill = "Treatment") +
 theme_classic(base_size = 14) +
 theme(
    axis.title.y = element_blank(),
    axis.title.x = element_blank(), axis.text.x = element_blank())+
  stat_compare_means(aes(label = sprintf("Kurskall-Wallis, p = %5.2f", as.numeric(..p.format..))),method = "kruskal.test", label.y = c(1.9), label.x=1.5)+
 # stat_compare_means(comparisons = list(c("D1", "D2"), c("D2", "D3"), c("D1", "D3")), method = "wilcox.test", label.y = c(1.5, 1.6,1.7))+
  ggtitle(expression(italic("V. surinamensis"))) +
  theme(legend.position = "none")+
# theme(legend.position = "bottom", legend.box = "vertical", legend.direction = "vertical")+
  ylim(1,2.1)
vs

legend <- get_legend(vs)
beta_fun_drought <- ggarrange(ef, ih, jc, po, sg, tm, vs, as_ggplot(legend),nrow = 2, ncol=4)

beta_fun_drought <- annotate_figure(beta_fun_drought, top = text_grob("B. Fungi", color = "black", face = "bold", size = 20, just = "right" , hjust = 4, vjust = 0.8))

ggsave(filename = "./results/microbio/beta_fun_drought.jpeg", plot = beta_fun_drought, width = 10, height = 6)

```

combine with bacteria for one big plot

```{r}
beta_drought <- ggarrange(beta_bac_drought, beta_fun_drought, nrow=2, ncol=1)

beta_drought

ggsave(filename = "./results/microbio/beta_drought.jpeg", plot = beta_drought, width = 10, height = 12)
```


##Beta - after drought, recovery boxplots
```{r}
#prepare data
data <- beta %>%  
  filter(indv_b_treatment != "C0") %>%
  mutate(Period = ifelse(indv_b_time %in% c("t21","t27", "t71"), "Drought", 
                  ifelse(indv_b_time %in% c("t51", "t57","t101"), "Recovery", NA)))
data$Period <- as.factor(data$Period)
data$Treatment <- as.factor(data$indv_b_treatment)

#epefal
ef <- data %>% filter(indv_b_species == "Epe.fal")
 ef_plot <- ggboxplot(
  ef, x= "Period" , y = "value", fill = "Period", facet.by = "indv_b_treatment", alpha =0.7) +           geom_jitter(shape = "circle", alpha = 0.5) +
  scale_fill_manual(values = c(Drought = "#ff9e00", Recovery = "#219ebc"))+
  theme_classic() +
  theme(axis.text.x = element_blank(), axis.title.x = element_blank())+
  ylab("Beta diversity")+
  theme(legend.position = "none")
ef_plot
stat.test <- data %>%
  filter(indv_b_species == "Epe.fal") %>%
  filter(indv_b_treatment != "D3") %>%
  group_by(indv_b_treatment) %>%
  t_test(value ~ Period) %>%
  adjust_pvalue(method = "bonferroni") %>%
  add_significance()

stat.test 
stat.test <- stat.test %>% add_xy_position(x = "Period") # Make facet and add p-values
ef_plot <- ef_plot + stat_pvalue_manual(stat.test)
ef_plot <- annotate_figure(ef_plot, left = text_grob("E. falcata       ", color = "black", face = "italic", size = 20))

#iryhos
ih <- data %>% filter(indv_b_species == "Iry.hos")
 ih_plot <- ggboxplot(
  ih, x= "Period" , y = "value", fill = "Period", facet.by = "indv_b_treatment", alpha =0.7) +           geom_jitter(shape = "circle", alpha = 0.5) +
  scale_fill_manual(values = c(Drought = "#ff9e00", Recovery = "#219ebc"))+
  theme_classic() +
  theme(axis.text.x = element_blank(), axis.title.x = element_blank())+
  ylab(" ")+
  theme(legend.position = "none")
ih_plot
stat.test <- data %>%
  filter(indv_b_species == "Iry.hos") %>%
  group_by(indv_b_treatment) %>%
  t_test(value ~ Period) %>%
  adjust_pvalue(method = "bonferroni") %>%
  add_significance()

stat.test 
stat.test <- stat.test %>% add_xy_position(x = "Period") 
ih_plot <- ih_plot + stat_pvalue_manual(stat.test)
ih_plot <- annotate_figure(ih_plot, right = text_grob("I. hostmannii", color = "black", face = "italic", size = 20))

#jaccop
jc <- data %>% filter(indv_b_species == "Jac.cop")
 jc_plot <- ggboxplot(
  jc, x= "Period" , y = "value", fill = "Period", facet.by = "indv_b_treatment", alpha =0.7) +           geom_jitter(shape = "circle", alpha = 0.5) +
  scale_fill_manual(values = c(Drought = "#ff9e00", Recovery = "#219ebc"))+
  theme_classic() +
  theme(axis.text.x = element_blank(), axis.title.x = element_blank())+
  ylab("Beta diversity")+
  theme(legend.position = "none")
jc_plot
stat.test <- jc %>%
  group_by(indv_b_treatment) %>%
  t_test(value ~ Period) %>%
  adjust_pvalue(method = "bonferroni") %>%
  add_significance()
stat.test 
stat.test <- stat.test %>% add_xy_position(x = "Period")
jc_plot <- jc_plot + stat_pvalue_manual(stat.test)
jc_plot <- annotate_figure(jc_plot, left = text_grob("J. copaia        ", color = "black", face = "italic", size = 20))

#pteoff
po <- data %>% filter(indv_b_species == "Pte.off")
 po_plot <- ggboxplot(
  po, x= "Period" , y = "value", fill = "Period", facet.by = "indv_b_treatment", alpha =0.7) +           geom_jitter(shape = "circle", alpha = 0.5) +
  scale_fill_manual(values = c(Drought = "#ff9e00", Recovery = "#219ebc"))+
  theme_classic() +
  theme(axis.text.x = element_blank(), axis.title.x = element_blank())+
  ylab(" ")+
  theme(legend.position = "none")
po_plot
stat.test <- po %>%
 group_by(indv_b_treatment) %>%
  t_test(value ~ Period) %>%
  adjust_pvalue(method = "bonferroni") %>%
  add_significance()
stat.test 
stat.test <- stat.test %>% add_xy_position(x = "Period")
po_plot <- po_plot + stat_pvalue_manual(stat.test)
po_plot <- annotate_figure(po_plot, right = text_grob("P. officinalis", color = "black", face = "italic", size = 20))

#symglo
sg <- data %>% filter(indv_b_species == "Sym.off")
 sg_plot <- ggboxplot(
  sg, x= "Period" , y = "value", fill = "Period", facet.by = "indv_b_treatment", alpha =0.7) +           geom_jitter(shape = "circle", alpha = 0.5) +
  scale_fill_manual(values = c(Drought = "#ff9e00", Recovery = "#219ebc"))+
  theme_classic() +
  theme(axis.text.x = element_blank(), axis.title.x = element_blank())+
  ylab("Beta diversity")+
  theme(legend.position = "none")
sg_plot
stat.test <- sg %>%
  group_by(indv_b_treatment) %>%
  t_test(value ~ Period) %>%
  adjust_pvalue(method = "bonferroni") %>%
  add_significance()
stat.test 
stat.test <- stat.test %>% add_xy_position(x = "Period")
sg_plot <- sg_plot + stat_pvalue_manual(stat.test)
sg_plot <- annotate_figure(sg_plot, left = text_grob("S. globulifera  ", color = "black", face = "italic", size = 20))

#tacmel
tm <- data %>% filter(indv_b_species == "Tac.mel")
 tm_plot <- ggboxplot(
  tm, x= "Period" , y = "value", fill = "Period", facet.by = "indv_b_treatment", alpha =0.7) +           geom_jitter(shape = "circle", alpha = 0.5) +
  scale_fill_manual(values = c(Drought = "#ff9e00", Recovery = "#219ebc"))+
  theme_classic() +
  theme(axis.text.x = element_blank(), axis.title.x = element_blank())+
  ylab(" ")+
  theme(legend.position = "none")
tm_plot
stat.test <- tm %>%
  filter(indv_b_treatment == "D1")%>%
 group_by(indv_b_treatment) %>%
  t_test(value ~ Period) %>%
  adjust_pvalue(method = "bonferroni") %>%
  add_significance()
stat.test 
stat.test <- stat.test %>% add_xy_position(x = "Period")
tm_plot <- tm_plot + stat_pvalue_manual(stat.test)
tm_plot <- annotate_figure(tm_plot, right = text_grob("T. melinonii", color = "black", face = "italic", size = 20))

#virsur
vs <- data %>% filter(indv_b_species == "Vir.sur")
 vs_plot <- ggboxplot(
  vs, x= "Period" , y = "value", fill = "Period", facet.by = "indv_b_treatment", alpha =0.7) +           geom_jitter(shape = "circle", alpha = 0.5) +
  scale_fill_manual(values = c(Drought = "#ff9e00", Recovery = "#219ebc"))+
  theme_classic() +
  theme(axis.text.x = element_blank(), axis.title.x = element_blank())+
  ylab("Beta diversity")+
  theme(legend.position = "none")
vs_plot
stat.test <- vs %>%
    filter(indv_b_treatment != "D3") %>%
   group_by(indv_b_treatment) %>%
  t_test(value ~ Period) %>%
  adjust_pvalue(method = "bonferroni") %>%
  add_significance()
stat.test 
stat.test <- stat.test %>% add_xy_position(x = "Period") 
vs_plot <- vs_plot + stat_pvalue_manual(stat.test)
vs_plot <- annotate_figure(vs_plot, left = text_grob("V. surinamensis", color = "black", face = "italic", size = 20))

#get legend
legend <- get_legend(vs_plot) #get legend work with no complicated plots fso just grab the legend when plot is simpler


#plot all
beta_recov_fun <- ggarrange(ef_plot, ih_plot, jc_plot, po_plot, sg_plot, tm_plot, vs_plot, as_ggplot(legend), ncol=2, nrow = 4)
beta_recov_fun
beta_recov_fun <- annotate_figure(beta_recov_fun, top = text_grob("Fungal beta diversity during and after drought", color = "black", face = "bold", size = 24))
ggsave(filename = "./results/microbio/beta_fun_recov.jpeg", plot = beta_recov_fun, width = 12, height = 10)
```



#PCoa -RECOVERY

```{r}
fungi <- AMVG42_230505_ITS_clean
rownames(fungi$samples) <- fungi$samples$sample_id
fungi$samples <- fungi$samples %>%
   mutate(Period = case_when(
     Treatment == "D1" &  Time == "t21" ~ "Drought D1",
      Treatment == "D1" &  Time == "t51" ~ "Recovery D1",
      Treatment == "D2" &  Time == "t27" ~ "Drought D2",
      Treatment == "D2" &  Time == "t57" ~ "Recovery D2",
      Treatment == "D3" &  Time == "t71" ~ "Drought D3",
      Treatment == "D3" &  Time == "t101" ~ "Recovery D3",
      TRUE                      ~ "Control"
    )
  )

fungi$samples$Name <- NA
fungi$samples$Name[fungi$samples$Species == "Epe.fal"] <- "E. falcata"
fungi$samples$Name[fungi$samples$Species == "Iry.hos"] <- "I. hostmannii"
fungi$samples$Name[fungi$samples$Species == "Jac.cop"] <- "J. copaia"
fungi$samples$Name[fungi$samples$Species == "Pte.off"] <- "P. officinalis"
fungi$samples$Name[fungi$samples$Species == "Sym.off"] <- "S. globulifera"
fungi$samples$Name[fungi$samples$Species == "Tac.mel"] <- "T. melinonii"
fungi$samples$Name[fungi$samples$Species == "Vir.sur"] <- "V. surinamensis"

fungi_D1 <- subset_metabarlist(fungi, table = "samples", indices = fungi$samples$Time %in% c("t21", "t51"))

#load metadata and taxonomy file
meta<-fungi_D1$samples
hill <- hill_taxa_parti_pairwise(fungi_D1$reads,q=1,output="matrix", pairs = "full") 
hill.beta  <- hill$TD_beta
hill.beta.dist <- as.dist(hill.beta)

# tests whether there are differences in dispersion
bd_treatment <- betadisper(d = hill.beta.dist, group = fungi_D1$samples$Period)
bd_treatment
anova(bd_treatment)
permutest(bd_treatment) #are there significant dispersions between the treatments? no: p-value 0.413. having no significant effects in the dispersions, easier and less problematic to interpret afterwards the adonis result. 

bd_Name <- betadisper(d = hill.beta.dist, group = fungi_D1$samples$Period)
bd_Name
anova(bd_Name)
permutest(bd_Name)

bd_Name_period <- betadisper(d = hill.beta.dist, group = paste0(fungi_D1$samples$Period, fungi_D1$samples$Name))
bd_Name_period
anova(bd_Name_period)
table_perm <- permutest(bd_Name_period, pairwise = TRUE) 

# test whether there are differences in fungal data between different water treatment periods while accounting for variation among host plant species.
Permanova  <-
  adonis2(hill.beta.dist~ fungi_D1$samples$Period, permutations = 999, strata=fungi_D1$samples$Name)  #permutation done within the same name

Permanova_fungi <-data.frame(Permanova, check.names = F) %>% 
  rename("R^2" =  "R2", "p-value"="Pr(>F)") %>% 
  rownames_to_column( var="Factor")

Permanova_fungi[1,1] <- "Treatment"

library(ggpubr)
perm_D1 <- Permanova_fungi %>%
mutate_at(3, funs(round(., 0)))  %>%
    mutate_at(5:6, as.numeric) %>% 
  mutate_at(4:5, funs(round(., 2))) %>% 
  mutate_at(5:6, ~replace(., is.na(.), "")) %>% 
    ggtexttable(., rows = NULL,  theme = ttheme(
             colnames.style = colnames_style(color = "black", fill = "white",hjust=0, x=0.1, size = 10), tbody.style = tbody_style(color = "black",  fill = c("white"), hjust=0, x=0.1))) %>%
  tab_add_hline(at.row = 1:2, row.side = "top", linewidth = 2)%>%
  table_cell_font(., row = 2, column = 6, face = "bold") %>% 
  tab_add_hline(at.row = c(4), row.side = "bottom", 
                linewidth = 1, linetype = 1) %>% 
  table_cell_font(., row = 2:4, column = 1,
                     size = 11)%>%
 tab_add_title(text = "D1", face = "bold", padding = unit(0.5, "line"))

library(pairwiseAdonis)
pairwise_d1 <- pairwise.adonis2(x = hill.beta.dist~ Period,data = fungi_D1$samples,nperm = 999, strata = "Name")

pairwise_d1_RD <- pairwise_d1$`Recovery D1_vs_Drought D1` %>% as_tibble() %>%
  dplyr::select(R2, `Pr(>F)`)%>% 
  rename("R^2" =  "R2", "p-value"="Pr(>F)") 

subtitle <- paste0(
  "drought", 
  " vs.",
  " recovery") %>%
  strwrap(width = 8) %>%
  paste(collapse = "\n")

pw_d1<- pairwise_d1_RD  %>%
    mutate_at(1:2, as.numeric) %>% 
  mutate_at(1:2, funs(round(., 2))) %>% 
  mutate_at(1:2, ~replace(., is.na(.), "")) %>% 
    ggtexttable(., rows = NULL,  theme = ttheme(
             colnames.style = colnames_style(color = "black", fill = "white",hjust=0, x=0.1, size = 10), tbody.style = tbody_style(color = "black",  fill = c("white"), hjust=0, x=0.1))) %>%
  tab_add_hline(at.row = 1:2, row.side = "top", linewidth = 2)%>%
  tab_add_hline(at.row = c(4), row.side = "bottom", 
                linewidth = 1, linetype = 1) %>% 
  table_cell_font(., row = 2:4, column = 1,
                     size = 11)%>%
 tab_add_title(text = subtitle, face = "bold", padding = unit(0.5, "line"))

pairwise_d1_RC <- pairwise_d1$`Recovery D1_vs_Control` %>% as_tibble() %>%
  dplyr::select(R2, `Pr(>F)`)%>% 
  rename("R^2" =  "R2", "p-value"="Pr(>F)") 

subtitle <- paste0(
  "control", 
  " vs.",
  " recovery") %>%
  strwrap(width = 8) %>%
  paste(collapse = "\n")

pw_d1_RC<- pairwise_d1_RC  %>%
    mutate_at(1:2, as.numeric) %>% 
  mutate_at(1:2, funs(round(., 2))) %>% 
  mutate_at(1:2, ~replace(., is.na(.), "")) %>% 
    ggtexttable(., rows = NULL,  theme = ttheme(
             colnames.style = colnames_style(color = "black", fill = "white",hjust=0, x=0.1, size = 10), tbody.style = tbody_style(color = "black",  fill = c("white"), hjust=0, x=0.1))) %>%
  tab_add_hline(at.row = 1:2, row.side = "top", linewidth = 2)%>%
  table_cell_font(., row = 2, column = 2, face = "bold") %>% 
  tab_add_hline(at.row = c(4), row.side = "bottom", 
                linewidth = 1, linetype = 1) %>% 
  table_cell_font(., row = 2:4, column = 1,
                     size = 11)%>%
 tab_add_title(text = subtitle, face = "bold", padding = unit(0.5, "line"))

pairwise_d1_DC <- pairwise_d1$`Drought D1_vs_Control` %>% as_tibble() %>%
  dplyr::select(R2, `Pr(>F)`)%>% 
  rename("R^2" =  "R2", "p-value"="Pr(>F)") 

subtitle <- paste0(
  "control", 
  " vs.",
  " drought") %>%
  strwrap(width = 8) %>%
  paste(collapse = "\n")

pw_d1_DC<- pairwise_d1_DC  %>%
    mutate_at(1:2, as.numeric) %>% 
  mutate_at(1:2, funs(round(., 3))) %>% 
  mutate_at(1:2, ~replace(., is.na(.), "")) %>% 
    ggtexttable(., rows = NULL,  theme = ttheme(
             colnames.style = colnames_style(color = "black", fill = "white",hjust=0, x=0.1, size = 10), tbody.style = tbody_style(color = "black",  fill = c("white"), hjust=0, x=0.1))) %>%
  tab_add_hline(at.row = 1:2, row.side = "top", linewidth = 2)%>%
  table_cell_font(., row = 2, column = 2, face = "bold") %>% 
  tab_add_hline(at.row = c(4), row.side = "bottom", 
                linewidth = 1, linetype = 1) %>% 
  table_cell_font(., row = 2:4, column = 1,
                     size = 11)%>%
 tab_add_title(text = subtitle, face = "bold", padding = unit(0.5, "line"))

D1 <- ggarrange(perm_D1, pw_d1, pw_d1_DC, pw_d1_RC, widths = c(1, 0.2, 0.2, 0.2), nrow=1, ncol=4)
D1


#D2
fungi_D2 <- subset_metabarlist(fungi, table = "samples", indices = fungi$samples$Time %in% c("t27", "t57"))

meta<-fungi_D2$samples
hill <- hill_taxa_parti_pairwise(fungi_D2$reads,q=1,output="matrix", pairs = "full") 
hill.beta  <- hill$TD_beta
hill.beta.dist <- as.dist(hill.beta)

# tests whether there are differences in dispersion
bd_treatment <- betadisper(d = hill.beta.dist, group = fungi_D2$samples$Period)
bd_treatment
anova(bd_treatment)
permutest(bd_treatment) #are there significant dispersions between the treatments? no: p-value 0.413. having no significant effects in the dispersions, easier and less problematic to interpret afterwards the adonis result. 

bd_Name <- betadisper(d = hill.beta.dist, group = fungi_D2$samples$Period)
bd_Name
anova(bd_Name)
permutest(bd_Name) #0.059 .

bd_Name_period <- betadisper(d = hill.beta.dist, group = paste0(fungi_D2$samples$Period, fungi_D2$samples$Name))
bd_Name_period
anova(bd_Name_period) #1.414e-05 ***


# test whether there are differences in fungal data between different water treatment periods while accounting for variation among host plant species.
Permanova  <-
  adonis2(hill.beta.dist~ fungi_D2$samples$Period, permutations = 999, strata=fungi_D2$samples$Name)  #permutation done within the same name

Permanova_fungi <-data.frame(Permanova, check.names = F) %>% 
  rename("R^2" =  "R2", "p-value"="Pr(>F)") %>% 
  rownames_to_column( var="Factor")

Permanova_fungi[1,1] <- "Treatment"

perm_D2 <- Permanova_fungi %>%
mutate_at(3, funs(round(., 0)))  %>%
    mutate_at(5:6, as.numeric) %>% 
  mutate_at(4:5, funs(round(., 2))) %>% 
  mutate_at(5:6, ~replace(., is.na(.), "")) %>% 
    ggtexttable(., rows = NULL,  theme = ttheme(
             colnames.style = colnames_style(color = "black", fill = "white",hjust=0, x=0.1, size = 10), tbody.style = tbody_style(color = "black",  fill = c("white"), hjust=0, x=0.1))) %>%
  tab_add_hline(at.row = 1:2, row.side = "top", linewidth = 2)%>%
  table_cell_font(., row = 2, column = 6, face = "bold") %>% 
  tab_add_hline(at.row = c(4), row.side = "bottom", 
                linewidth = 1, linetype = 1) %>% 
  table_cell_font(., row = 2:4, column = 1,
                     size = 11)%>%
  tab_add_title(text = "D2", face = "bold", padding = unit(0.5, "line"))

pairwise_d2 <- pairwise.adonis2(x = hill.beta.dist~ Period,data = fungi_D2$samples,nperm = 999, strata = "Name")

pairwise_d2_RD <- pairwise_d2$`Drought D2_vs_Recovery D2` %>% as_tibble() %>%
  dplyr::select(R2, `Pr(>F)`)%>% 
  rename("R^2" =  "R2", "p-value"="Pr(>F)") 

subtitle <- paste0(
  "drought", 
  " vs.",
  " recovery") %>%
  strwrap(width = 8) %>%
  paste(collapse = "\n")

pw_d2<- pairwise_d2_RD  %>%
    mutate_at(1:2, as.numeric) %>% 
  mutate_at(1:2, funs(round(., 2))) %>% 
  mutate_at(1:2, ~replace(., is.na(.), "")) %>% 
    ggtexttable(., rows = NULL,  theme = ttheme(
             colnames.style = colnames_style(color = "black", fill = "white",hjust=0, x=0.1, size = 10), tbody.style = tbody_style(color = "black",  fill = c("white"), hjust=0, x=0.1))) %>%
  tab_add_hline(at.row = 1:2, row.side = "top", linewidth = 2)%>%
  tab_add_hline(at.row = c(4), row.side = "bottom", 
                linewidth = 1, linetype = 1) %>% 
  table_cell_font(., row = 2:4, column = 1,
                     size = 11)%>%
 tab_add_title(text = subtitle, face = "bold", padding = unit(0.5, "line"))

pairwise_d2_RC <- pairwise_d2$`Recovery D2_vs_Control` %>% as_tibble() %>%
  dplyr::select(R2, `Pr(>F)`)%>% 
  rename("R^2" =  "R2", "p-value"="Pr(>F)") 

subtitle <- paste0(
  "control", 
  " vs.",
  " recovery") %>%
  strwrap(width = 8) %>%
  paste(collapse = "\n")

pw_d2_RC<- pairwise_d2_RC  %>%
    mutate_at(1:2, as.numeric) %>% 
  mutate_at(1:2, funs(round(., 2))) %>% 
  mutate_at(1:2, ~replace(., is.na(.), "")) %>% 
    ggtexttable(., rows = NULL,  theme = ttheme(
             colnames.style = colnames_style(color = "black", fill = "white",hjust=0, x=0.1, size = 10), tbody.style = tbody_style(color = "black",  fill = c("white"), hjust=0, x=0.1))) %>%
  tab_add_hline(at.row = 1:2, row.side = "top", linewidth = 2)%>%
  table_cell_font(., row = 2, column = 2, face = "bold") %>% 
  tab_add_hline(at.row = c(4), row.side = "bottom", 
                linewidth = 1, linetype = 1) %>% 
  table_cell_font(., row = 2:4, column = 1,
                     size = 11)%>%
 tab_add_title(text = subtitle, face = "bold", padding = unit(0.5, "line"))

pairwise_d2_DC <- pairwise_d2$`Drought D2_vs_Control` %>% as_tibble() %>%
  dplyr::select(R2, `Pr(>F)`)%>% 
  rename("R^2" =  "R2", "p-value"="Pr(>F)") 

subtitle <- paste0(
  "control", 
  " vs.",
  " drought") %>%
  strwrap(width = 8) %>%
  paste(collapse = "\n")

pw_d2_DC<- pairwise_d2_DC  %>%
    mutate_at(1:2, as.numeric) %>% 
  mutate_at(1:2, funs(round(., 3))) %>% 
  mutate_at(1:2, ~replace(., is.na(.), "")) %>% 
    ggtexttable(., rows = NULL,  theme = ttheme(
             colnames.style = colnames_style(color = "black", fill = "white",hjust=0, x=0.1, size = 10), tbody.style = tbody_style(color = "black",  fill = c("white"), hjust=0, x=0.1))) %>%
  tab_add_hline(at.row = 1:2, row.side = "top", linewidth = 2)%>%
  table_cell_font(., row = 2, column = 2, face = "bold") %>% 
  tab_add_hline(at.row = c(4), row.side = "bottom", 
                linewidth = 1, linetype = 1) %>% 
  table_cell_font(., row = 2:4, column = 1,
                     size = 11)%>%
 tab_add_title(text = subtitle, face = "bold", padding = unit(0.5, "line"))

D2 <- ggarrange(perm_D2, pw_d2, pw_d2_DC, pw_d2_RC, widths = c(1, 0.2, 0.2, 0.2), nrow=1, ncol=4)
D2


#D3 
fungi_D3 <- subset_metabarlist(fungi, table = "samples", indices = fungi$samples$Time %in% c("t71", "t101"))
meta<-fungi_D3$samples
hill <- hill_taxa_parti_pairwise(fungi_D3$reads,q=1,output="matrix", pairs = "full") 
hill.beta  <- hill$TD_beta
hill.beta.dist <- as.dist(hill.beta)

# tests whether there are differences in dispersion
bd_treatment <- betadisper(d = hill.beta.dist, group = fungi_D3$samples$Period)
bd_treatment
anova(bd_treatment)
permutest(bd_treatment) #are there significant dispersions between the treatments? no: p-value 0.413. having no significant effects in the dispersions, easier and less problematic to interpret afterwards the adonis result. 

bd_Name <- betadisper(d = hill.beta.dist, group = fungi_D3$samples$Period)
bd_Name
anova(bd_Name)
permutest(bd_Name) 

bd_Name_period <- betadisper(d = hill.beta.dist, group = paste0(fungi_D3$samples$Period, fungi_D3$samples$Name))
bd_Name_period
anova(bd_Name_period) 

# test whether there are differences in fungal data between different water treatment periods while accounting for variation among host plant species.
Permanova  <-
  adonis2(hill.beta.dist~ fungi_D3$samples$Period, permutations = 999, strata=fungi_D3$samples$Name)  #permutation done within the same name


Permanova_fungi <-data.frame(Permanova, check.names = F) %>% 
  rename("R^2" =  "R2", "p-value"="Pr(>F)") %>% 
  rownames_to_column( var="Factor")

Permanova_fungi[1,1] <- "Treatment"

library(ggpubr)
perm_D3 <- Permanova_fungi %>%
mutate_at(3, funs(round(., 0)))  %>%
    mutate_at(5:6, as.numeric) %>% 
  mutate_at(4:5, funs(round(., 2))) %>% 
  mutate_at(5:6, ~replace(., is.na(.), "")) %>% 
    ggtexttable(., rows = NULL,  theme = ttheme(
             colnames.style = colnames_style(color = "black", fill = "white",hjust=0, x=0.1, size = 10), tbody.style = tbody_style(color = "black",  fill = c("white"), hjust=0, x=0.1))) %>%
  tab_add_hline(at.row = 1:2, row.side = "top", linewidth = 2)%>%
  tab_add_hline(at.row = c(4), row.side = "bottom", 
                linewidth = 1, linetype = 1) %>% 
  table_cell_font(., row = 2:4, column = 1,
                     size = 11)%>%
  tab_add_title(text = "D3", face = "bold", padding = unit(0.5, "line"))

pw_d1_all<- (pw_d1+ pw_d1_DC+pw_d1_RC)+plot_layout(ncol = 3)
D1 <- ggarrange(perm_D1,pw_d1_all, nrow=1)

pw_d2_all<- (pw_d2+ pw_d2_DC+pw_d2_RC)+plot_layout(ncol = 3)
D2 <- ggarrange(perm_D2,pw_d2_all, nrow=1)

ggsave("./results/D1_fungi_option3.jpeg", plot=D1, width = 10, height = 5)
ggsave("./results/D2_fungi_option3.jpeg", plot=D2, width = 10, height = 5)

ggsave("./results/D3_fungi_option3.jpeg", plot=perm_D3, width = 10, height = 5)

```


#NDV - ITS
```{r}
load("E:/DRYER_greenhouse/results/microbio/DRYER_GH_NDV_ITS_bray.RData")

NDV_ITS <- as.data.frame(bray_abund_null_dev_fun)

NDV_ITS$sample_id <- rownames(NDV_ITS)

plant_info <- AMVG42_230505_ITS_clean$samples %>% dplyr::select(sample_id, code, Species, Time, Treatment)

NDV_ITS <- left_join(NDV_ITS, plant_info, by = "sample_id") 

#for the labels
NDV_ITS$facet_label <- NA
NDV_ITS$facet_label[NDV_ITS$Species == "Epe.fal"] <- "E. falcata"
NDV_ITS$facet_label[NDV_ITS$Species == "Iry.hos"] <- "I. hostmannii"
NDV_ITS$facet_label[NDV_ITS$Species == "Jac.cop"] <- "J. copaia"
NDV_ITS$facet_label[NDV_ITS$Species == "Pte.off"] <- "P. officinalis"
NDV_ITS$facet_label[NDV_ITS$Species == "Sym.off"] <- "S. globulifera"
NDV_ITS$facet_label[NDV_ITS$Species == "Tac.mel"] <- "T. melinonii"
NDV_ITS$facet_label[NDV_ITS$Species == "Vir.sur"] <- "V. surinamensis"

plot_NDV <- NDV_ITS %>%
 filter(Time %in% c("t21", "t27", "t71")) %>%
 ggplot() +
  aes(x = Time, y = bray_abund_null_dev_fun, fill = Treatment) +
  geom_boxplot(alpha = 0.6) +
  geom_jitter(alpha = 0.5, fill = "gray80") + 
  scale_fill_manual(
    values = c(C0 = "#6DD4F8",
    D1 = "#8CB425", D2 = "#ECE320", D3 = "#DF5220")
  ) +
  labs(y = "NDV") +
  labs(x="")+
  theme_classic() +
  theme(
    axis.title.y = element_text(size = 16L,
    face = "bold"),
    strip.text = element_text(size=12, face= "italic"),
    strip.background = element_blank()
  ) +
 facet_wrap(vars(facet_label)) +
  ggtitle("Fungi")

plot_NDV_withoutC0 <- NDV_ITS %>%
 filter(Time %in% c("t21", "t27", "t71")) %>%
  filter(Treatment != "C0") %>%
 ggplot() +
  aes(x = Time, y = bray_abund_null_dev_fun, fill = Treatment) +
  geom_boxplot(alpha = 0.6) +
  geom_jitter(alpha = 0.5, fill = "gray80") + 
  scale_fill_manual(
    values = c(
    D1 = "#8CB425", D2 = "#ECE320", D3 = "#DF5220")
  ) +
  labs(y = "NDV") +
  labs(x="")+
  theme_classic() +
  theme(
    axis.title.y = element_text(size = 16L,
    face = "bold"),
    strip.text = element_text(size=12, face= "italic"),
    strip.background = element_blank()
  ) +
 facet_wrap(vars(facet_label)) +
  ggtitle("Fungi")
plot_NDV_withoutC0
```

```{r}
#increasing drought treatments
ef <-  NDV_ITS  %>%
 filter(Species == "Epe.fal") %>%
 filter(Time %in% c("t21", "t27", "t71")) %>%
 #filter(Treatment != "C0")  %>%
 ggplot() +
  aes(x = Treatment, y = bray_abund_null_dev_fun, fill = Treatment) +
  geom_boxplot(alpha = 0.6)+
   geom_jitter(shape = "circle", alpha = 0.5) +
  scale_fill_manual(values = c(C0 = "#6DD4F8", 
      D1 = "#8CB425", D2 = "#ECE320", D3 = "#DF5220")) + 
  theme_minimal(base_size = 14) +
   ylab("NDV")+
  xlab("")+
  stat_compare_means(method = "kruskal.test", label.y = c(0.85), label.x = 1.5)+
   #stat_compare_means(comparisons = list(c("D1", "D2"), c("D1", "D3"), c("D2", "D3")), method = "wilcox.test", label.y = c(0.65,0.7,0.73))+
  ggtitle(expression(italic("E. falcata"))) +
  theme(legend.position = "none")+
   ylim(0.6,0.9)
ef

ih <- NDV_ITS  %>%
 filter(Species == "Iry.hos") %>%
 filter(Time %in% c("t21", "t27", "t71")) %>%
 #filter(Treatment != "C0")  %>%
 ggplot() +
  aes(x = Treatment, y = bray_abund_null_dev_fun, fill = Treatment) +
  geom_boxplot(alpha = 0.6)+
   geom_jitter(shape = "circle", alpha = 0.5) +
  scale_fill_manual(values = c(C0 = "#6DD4F8",
      D1 = "#8CB425", D2 = "#ECE320", D3 = "#DF5220")) + 
  theme_minimal(base_size = 14) +
   ylab(" ")+
  xlab("")+
  stat_compare_means(method = "kruskal.test", label.y = c(0.85), label.x = 1.5)+
  ggtitle(expression(italic("I. hostmannii"))) +
  theme(legend.position = "none")+
   ylim(0.6,0.9)
ih

jc <-NDV_ITS  %>%
 filter(Species == "Jac.cop") %>%
 filter(Time %in% c("t21", "t27", "t71")) %>%
# filter(Treatment != "C0")  %>%
 ggplot() +
  aes(x = Treatment, y = bray_abund_null_dev_fun, fill = Treatment) +
  geom_boxplot(alpha = 0.6)+
   geom_jitter(shape = "circle", alpha = 0.5) +
  scale_fill_manual(values = c(C0 = "#6DD4F8", 
      D1 = "#8CB425", D2 = "#ECE320", D3 = "#DF5220")) + 
  theme_minimal(base_size = 14) +
   ylab(" ")+
  xlab("")+
  stat_compare_means(method = "kruskal.test", label.y = c(0.85), label.x = 1.5)+
  ggtitle(expression(italic("J. copaia"))) +
  theme(legend.position = "none")+
   ylim(0.6,0.9)
jc

po <-NDV_ITS  %>%
 filter(Species == "Pte.off") %>%
 filter(Time %in% c("t21", "t27", "t71")) %>%
 #filter(Treatment != "C0")  %>%
 ggplot() +
  aes(x = Treatment, y = bray_abund_null_dev_fun, fill = Treatment) +
  geom_boxplot(alpha = 0.6)+
   geom_jitter(shape = "circle", alpha = 0.5) +
  scale_fill_manual(values = c( C0 = "#6DD4F8",
      D1 = "#8CB425", D2 = "#ECE320", D3 = "#DF5220")) + 
  theme_minimal(base_size = 14) +
   ylab("")+
  xlab("")+
  stat_compare_means(method = "kruskal.test", label.y = c(0.85), label.x = 1.5)+
  ggtitle(expression(italic("P. officinalis"))) +
  theme(legend.position = "none")+
   ylim(0.6,0.9)
po

sg <- NDV_ITS  %>%
 filter(Species == "Sym.off") %>%
 filter(Time %in% c("t21", "t27", "t71")) %>%
# filter(Treatment != "C0")  %>%
 ggplot() +
  aes(x = Treatment, y = bray_abund_null_dev_fun, fill = Treatment) +
  geom_boxplot(alpha = 0.6)+
   geom_jitter(shape = "circle", alpha = 0.5) +
  scale_fill_manual(values = c(C0 = "#6DD4F8",
      D1 = "#8CB425", D2 = "#ECE320", D3 = "#DF5220")) + 
  theme_minimal(base_size = 14) +
   ylab("NDV")+
  xlab("")+
  stat_compare_means(method = "kruskal.test", label.y = c(0.85), label.x = 1.5)+
  ggtitle(expression(italic("S. globulifera"))) +
  theme(legend.position = "none")+
   ylim(0.6,0.9)
sg

tm <-NDV_ITS  %>%
 filter(Species == "Tac.mel") %>%
 filter(Time %in% c("t21", "t27", "t71")) %>%
# filter(Treatment != "C0")  %>%
 ggplot() +
  aes(x = Treatment, y = bray_abund_null_dev_fun, fill = Treatment) +
  geom_boxplot(alpha = 0.6)+
   geom_jitter(shape = "circle", alpha = 0.5) +
  scale_fill_manual(values = c( C0 = "#6DD4F8",
      D1 = "#8CB425", D2 = "#ECE320", D3 = "#DF5220")) + 
  theme_minimal(base_size = 14) +
   ylab("NDV")+
  xlab("")+
  stat_compare_means(method = "kruskal.test", label.y = c(0.85), label.x = 1.5)+
  ggtitle(expression(italic("T. melinonii"))) +
  theme(legend.position = "none")+
   ylim(0.6,0.9)
tm

vs <-NDV_ITS  %>%
 filter(Species == "Vir.sur") %>%
 filter(Time %in% c("t21", "t27", "t71")) %>%
 #filter(Treatment != "C0")  %>%
 ggplot() +
  aes(x = Treatment, y = bray_abund_null_dev_fun, fill = Treatment) +
  geom_boxplot(alpha = 0.6)+
   geom_jitter(shape = "circle", alpha = 0.5) +
  scale_fill_manual(values = c( C0 = "#6DD4F8",
      D1 = "#8CB425", D2 = "#ECE320", D3 = "#DF5220")) + 
  theme_minimal(base_size = 14) +
   ylab("")+
  xlab("")+
  stat_compare_means(method = "kruskal.test", label.y = c(0.85), label.x = 1.25)+
  ggtitle(expression(italic("V. surinamensis"))) +
  theme(legend.position = "none" )+
  ylim(0.6,0.9)#+
  #theme(legend.position = "bottom", legend.box = "vertical", legend.direction = "vertical")
vs

legend <- get_legend(vs)

ndv_fun_drought <- ggarrange(ef, ih, jc, po, sg, tm, vs, as_ggplot(legend), nrow=2, ncol=4)

ndv_fun_drought <- annotate_figure(ndv_fun_drought, top = text_grob("B. Fungi", color = "black", face = "bold", size = 20, just = "right" , hjust = 4, vjust = 0.8))


ggsave(filename = "./results/microbio/ndv_fun_drought.jpeg", plot = ndv_fun_drought, width = 10, height = 6)
```

##after drought, recovery ITS
```{r}
#after drought, during the recovery phase

#prepare data
data <- NDV_ITS %>%  filter(Treatment != "C0") %>% filter(!is.na(Species)) %>%
  mutate(Period = ifelse(Time %in% c("t21","t27", "t71"), "Drought", 
                  ifelse(Time %in% c("t51", "t57","t101"), "Recovery", NA)))
data$Period <- as.factor(data$Period)
data$Treatment <- as.factor(data$Treatment)

#epefal
ef <- data %>% filter(Species == "Epe.fal")
 ef_plot <- ggboxplot(
  ef, x= "Period" , y = "bray_abund_null_dev_fun", fill = "Period", facet.by = "Treatment", alpha =0.7) +           geom_jitter(shape = "circle", alpha = 0.5) +
  scale_fill_manual(values = c(Drought = "#ff9e00", Recovery = "#219ebc"))+
  theme_classic() +
  theme(axis.text.x = element_blank(), axis.title.x = element_blank())+
  ylab("NDV")+
  theme(legend.position = "none")
ef_plot
stat.test <- data %>%
  filter(Species == "Epe.fal") %>%
  filter(Treatment != "D3") %>%
  group_by(Treatment) %>%
  t_test(bray_abund_null_dev_fun ~ Period) %>%
  adjust_pvalue(method = "bonferroni") %>%
  add_significance()
stat.test 
stat.test <- stat.test %>% add_xy_position(x = "Period") # Make facet and add p-values
ef_plot <- ef_plot + stat_pvalue_manual(stat.test)
ef_plot <- annotate_figure(ef_plot, left = text_grob("E. falcata       ", color = "black", face = "italic", size = 20))

#iryhos
ih <- data %>% filter(Species == "Iry.hos") 
ih_plot <- ggboxplot(
  ih, x= "Period" , y = "bray_abund_null_dev_fun", fill = "Period", facet.by = "Treatment", alpha =0.7) + 
  geom_jitter(shape = "circle", alpha = 0.5) +
  scale_fill_manual(values = c(Drought = "#ff9e00", Recovery = "#219ebc"))+
  theme_classic() +
  theme(axis.text.x = element_blank(), axis.title.x = element_blank())+
  ylab("NDV")+
  theme(legend.position = "none")
ih_plot
stat.test <- ih %>%
  group_by(Species, Treatment) %>%
  t_test(bray_abund_null_dev_fun ~ Period) %>%
  adjust_pvalue(method = "bonferroni") %>%
  add_significance()
stat.test 
stat.test <- stat.test %>% add_xy_position(x = "Period") 
ih_plot <- ih_plot + stat_pvalue_manual(stat.test)
ih_plot <- annotate_figure(ih_plot, right = text_grob("I. hostmanii", color = "black", face = "italic", size = 20))

#jaccop
jc <- data %>% filter(Species == "Jac.cop") 
 jc_plot <- ggboxplot(
   jc, x= "Period" , y = "bray_abund_null_dev_fun", fill = "Period", facet.by = "Treatment", alpha =0.7) + 
   geom_jitter(shape = "circle", alpha = 0.5) +
   scale_fill_manual(values = c(Drought = "#ff9e00", Recovery = "#219ebc"))+
  theme_classic() +
  theme(axis.text.x = element_blank(), axis.title.x = element_blank())+
  ylab("NDV")+
  theme(legend.position = "none")
jc_plot
stat.test <- jc %>%
  group_by(Species, Treatment) %>%
  t_test(bray_abund_null_dev_fun ~ Period) %>%
  adjust_pvalue(method = "bonferroni") %>%
  add_significance()
stat.test 
stat.test <- stat.test %>% add_xy_position(x = "Period")
jc_plot <- jc_plot + stat_pvalue_manual(stat.test)
jc_plot <- annotate_figure(jc_plot, left = text_grob("J. copaia        ", color = "black", face = "italic", size = 20))

#pteoff
po <- data %>% filter(Species == "Pte.off") 
 po_plot <- ggboxplot(
   po, x= "Period" , y = "bray_abund_null_dev_fun", fill = "Period", facet.by = "Treatment", alpha =0.7) + 
  geom_jitter(shape = "circle", alpha = 0.5) +
  scale_fill_manual(values = c(Drought = "#ff9e00", Recovery = "#219ebc"))+
  theme_classic() +
  theme(axis.text.x = element_blank(), axis.title.x = element_blank())+
  ylab("NDV")+
  theme(legend.position = "none")
po_plot
stat.test <- po %>%
  group_by(Species, Treatment) %>%
  t_test(bray_abund_null_dev_fun ~ Period) %>%
  adjust_pvalue(method = "bonferroni") %>%
  add_significance()
stat.test 
stat.test <- stat.test %>% add_xy_position(x = "Period")
po_plot <- po_plot + stat_pvalue_manual(stat.test)
po_plot <- annotate_figure(po_plot, right = text_grob("P. officinalis", color = "black", face = "italic", size = 20))

#symglo
sg <- data %>% filter(Species == "Sym.off") 
 sg_plot <- ggboxplot(
   sg, x= "Period" , y = "bray_abund_null_dev_fun", fill = "Period", facet.by = "Treatment", alpha =0.7) + 
  geom_jitter(shape = "circle", alpha = 0.5) +
  scale_fill_manual(values = c(Drought = "#ff9e00", Recovery = "#219ebc"))+
  theme_classic() +
  theme(axis.text.x = element_blank(), axis.title.x = element_blank())+
  ylab("NDV")+
  theme(legend.position = "none")
sg_plot
stat.test <- sg %>%
  group_by(Species, Treatment) %>%
  t_test(bray_abund_null_dev_fun ~ Period) %>%
  adjust_pvalue(method = "bonferroni") %>%
  add_significance()
stat.test 
stat.test <- stat.test %>% add_xy_position(x = "Period")
sg_plot <- sg_plot + stat_pvalue_manual(stat.test)
sg_plot <- annotate_figure(sg_plot, left = text_grob("S. globulifera  ", color = "black", face = "italic", size = 20))

#tacmel
tm <- data %>% filter(Species == "Tac.mel") 
tm_plot <- ggboxplot(
  tm, x= "Period" , y = "bray_abund_null_dev_fun", fill = "Period", facet.by = "Treatment", alpha =0.7) + 
  geom_jitter(shape = "circle", alpha = 0.5) +
  scale_fill_manual(values = c(Drought = "#ff9e00", Recovery = "#219ebc"))+
  theme_classic() +
  theme(axis.text.x = element_blank(), axis.title.x = element_blank())+
  ylab("NDV")+
  theme(legend.position = "none")
tm_plot
stat.test <- tm %>%
  group_by(Species, Treatment) %>%
  t_test(bray_abund_null_dev_fun ~ Period) %>%
  adjust_pvalue(method = "bonferroni") %>%
  add_significance()
stat.test 
stat.test <- stat.test %>% add_xy_position(x = "Period")
tm_plot <- tm_plot + stat_pvalue_manual(stat.test)
tm_plot <- annotate_figure(tm_plot, right = text_grob("T. melinonii", color = "black", face = "italic", size = 20))

#virsur
vs <- data %>% filter(Species == "Vir.sur") 
vs_plot <- ggboxplot(
  vs, x= "Period" , y = "bray_abund_null_dev_fun", fill = "Period", facet.by = "Treatment", alpha =0.7) + 
  geom_jitter(shape = "circle", alpha = 0.5) +
  scale_fill_manual(values = c(Drought = "#ff9e00", Recovery = "#219ebc"))+
  theme_classic() +
  theme(axis.text.x = element_blank(), axis.title.x = element_blank())+
  ylab("NDV ")+
  theme(legend.position = "none", legend.text = element_text(size = 16),
    legend.title = element_text(size = 20))
vs_plot
stat.test <- vs %>%
  filter(!(Treatment == "D3")) %>%
  group_by(Species, Treatment) %>%
  t_test(bray_abund_null_dev_fun ~ Period) %>%
  adjust_pvalue(method = "bonferroni") %>%
  add_significance()
stat.test 
stat.test <- stat.test %>% add_xy_position(x = "Period") 
vs_plot <- vs_plot + stat_pvalue_manual(stat.test)
vs_plot <- annotate_figure(vs_plot, left = text_grob("V. surinamensis", color = "black", face = "italic", size = 20))

#get legend
legend <- get_legend(vs_plot) #get legend work with no complicated plots fso just grab the legend when plot is simpler
as_ggplot(legend)

#plot all
ndv_recov_fun <- ggarrange(ef_plot, ih_plot, jc_plot, po_plot, sg_plot, tm_plot, vs_plot, as_ggplot(legend), ncol=2, nrow = 4)
ndv_recov_fun
ndv_recov_fun <- annotate_figure(ndv_recov_fun, top = text_grob("Stochastic and deterministic processes of fungal communities during and after drought", color = "black", face = "bold", size = 20))
ggsave(filename = "./results/microbio/ndv_fun_recov.jpeg", plot = ndv_recov_fun, width = 12, height = 10)

```



#NDV- bacteria
```{r}

load("E:/DRYER_greenhouse/results/microbio/DRYER_GH_NDV_16S_bray.RData")

NDV_bac <- as.data.frame(bray_abund_null_dev_fun)

NDV_bac$sample_id <- rownames(NDV_bac)

plant_info <- AMVG51_230505_16S_clean$samples %>% dplyr::select(sample_id, code, Species, Time, Treatment)

NDV_bac <- left_join(NDV_bac, plant_info, by = "sample_id") 

#for the labels
NDV_bac$facet_label <- NA
NDV_bac$facet_label[NDV_bac$Species == "Epe.fal"] <- "E. falcata"
NDV_bac$facet_label[NDV_bac$Species == "Iry.hos"] <- "I. hostmannii"
NDV_bac$facet_label[NDV_bac$Species == "Jac.cop"] <- "J. copaia"
NDV_bac$facet_label[NDV_bac$Species == "Pte.off"] <- "P. officinalis"
NDV_bac$facet_label[NDV_bac$Species == "Sym.off"] <- "S. globulifera"
NDV_bac$facet_label[NDV_bac$Species == "Tac.mel"] <- "T. melinonii"
NDV_bac$facet_label[NDV_bac$Species == "Vir.sur"] <- "V. surinamensis"

plot_NDV <- NDV_bac %>%
 filter(Time %in% c("t21", "t27", "t71")) %>%
 ggplot() +
  aes(x = Time, y = bray_abund_null_dev_fun, fill = Treatment) +
  geom_boxplot(alpha = 0.6) +
  geom_jitter(alpha = 0.5, fill = "gray80") + 
  scale_fill_manual(
    values = c(C0 = "#6DD4F8",
    D1 = "#8CB425", D2 = "#ECE320", D3 = "#DF5220")
  ) +
  labs(y = "NDV") +
  labs(x="")+
  theme_classic() +
  theme(
    axis.title.y = element_text(size = 16L,
    face = "bold"),
    strip.text = element_text(size=12, face= "italic"),
    strip.background = element_blank()
  ) +
 facet_wrap(vars(facet_label)) +
  ggtitle("Bacteria")
plot_NDV

plot_NDV_withoutC0 <- NDV_bac %>%
 filter(Time %in% c("t21", "t27", "t71")) %>%
  filter(Treatment != "C0") %>%
 ggplot() +
  aes(x = Time, y = bray_abund_null_dev_fun, fill = Treatment) +
  geom_boxplot(alpha = 0.6) +
  geom_jitter(alpha = 0.5, fill = "gray80") + 
  scale_fill_manual(
    values = c(
    D1 = "#8CB425", D2 = "#ECE320", D3 = "#DF5220")
  ) +
  labs(y = "NDV") +
  labs(x="")+
  theme_classic() +
  theme(
    axis.title.y = element_text(size = 16L,
    face = "bold"),
    strip.text = element_text(size=12, face= "italic"),
    strip.background = element_blank()
  ) +
 facet_wrap(vars(facet_label)) +
  ggtitle("Bacteria")
plot_NDV_withoutC0
```

```{r}
#increasing drought treatments
ef <-  NDV_bac  %>%
 filter(Species == "Epe.fal") %>%
 filter(Time %in% c("t21", "t27", "t71")) %>%
 #filter(Treatment != "C0")  %>%
 ggplot() +
  aes(x = Treatment, y = bray_abund_null_dev_fun, fill = Treatment) +
  geom_boxplot(alpha = 0.6)+
   geom_jitter(shape = "circle", alpha = 0.5) +
  scale_fill_manual(values = c(C0 = "#6DD4F8", 
      D1 = "#8CB425", D2 = "#ECE320", D3 = "#DF5220")) + 
  theme_minimal(base_size = 14) +
   ylab("NDV")+
  xlab("")+
  stat_compare_means(method = "kruskal.test", label.y = c(0.5))+
  ggtitle(expression(italic("E. falcata"))) +
  theme(legend.position = "none")
ef

ih <- NDV_bac  %>%
 filter(Species == "Iry.hos") %>%
 filter(Time %in% c("t21", "t27", "t71")) %>%
 #filter(Treatment != "C0")  %>%
 ggplot() +
  aes(x = Treatment, y = bray_abund_null_dev_fun, fill = Treatment) +
  geom_boxplot(alpha = 0.6)+
   geom_jitter(shape = "circle", alpha = 0.5) +
  scale_fill_manual(values = c(C0 = "#6DD4F8", 
      D1 = "#8CB425", D2 = "#ECE320", D3 = "#DF5220")) + 
  theme_minimal(base_size = 14) +
   ylab(" ")+
  xlab("")+
  stat_compare_means(method = "kruskal.test", label.y = c(0.57), label.x = c(1.5))+
  ggtitle(expression(italic("I. hostmannii"))) +
  theme(legend.position = "none")+
  ylim(0.25,0.6)
ih

jc <-NDV_bac  %>%
 filter(Species == "Jac.cop") %>%
 filter(Time %in% c("t21", "t27", "t71")) %>%
 #filter(Treatment != "C0")  %>%
 ggplot() +
  aes(x = Treatment, y = bray_abund_null_dev_fun, fill = Treatment) +
  geom_boxplot(alpha = 0.6)+
   geom_jitter(shape = "circle", alpha = 0.5) +
  scale_fill_manual(values = c(C0 = "#6DD4F8",
      D1 = "#8CB425", D2 = "#ECE320", D3 = "#DF5220")) + 
  theme_minimal(base_size = 14) +
   ylab(" ")+
  xlab("")+
  stat_compare_means(method = "kruskal.test", label.y = c(0.65), label.x = c(1.5))+
  ggtitle(expression(italic("J. copaia"))) +
  theme(legend.position = "none")+
  ylim(0.25,0.72)
jc

po <-NDV_bac  %>%
 filter(Species == "Pte.off") %>%
 filter(Time %in% c("t21", "t27", "t71")) %>%
 #filter(Treatment != "C0")  %>%
 ggplot() +
  aes(x = Treatment, y = bray_abund_null_dev_fun, fill = Treatment) +
  geom_boxplot(alpha = 0.6)+
   geom_jitter(shape = "circle", alpha = 0.5) +
  scale_fill_manual(values = c(C0 = "#6DD4F8",
      D1 = "#8CB425", D2 = "#ECE320", D3 = "#DF5220")) + 
  theme_minimal(base_size = 14) +
   ylab("")+
  xlab("")+
  stat_compare_means(method = "kruskal.test", label.y = c(0.65), label.x = c(1.5))+
  ggtitle(expression(italic("P. officinalis"))) +
  theme(legend.position = "none")+ylim(0.25,0.70)
po

sg <- NDV_bac  %>%
 filter(Species == "Sym.off") %>%
 filter(Time %in% c("t21", "t27", "t71")) %>%
 #filter(Treatment != "C0")  %>%
 ggplot() +
  aes(x = Treatment, y = bray_abund_null_dev_fun, fill = Treatment) +
  geom_boxplot(alpha = 0.6)+
   geom_jitter(shape = "circle", alpha = 0.5) +
  scale_fill_manual(values = c(C0 = "#6DD4F8", 
      D1 = "#8CB425", D2 = "#ECE320", D3 = "#DF5220")) + 
  theme_minimal(base_size = 14) +
   ylab("NDV")+
  xlab("")+
  stat_compare_means(method = "kruskal.test", label.y = c(0.65), label.x = 1.5)+
  ggtitle(expression(italic("S. globulifera"))) +
  theme(legend.position = "none")+
  ylim(0.29,0.72)
sg

tm <-NDV_bac  %>%
 filter(Species == "Tac.mel") %>%
 filter(Time %in% c("t21", "t27", "t71")) %>%
 #filter(Treatment != "C0")  %>%
 ggplot() +
  aes(x = Treatment, y = bray_abund_null_dev_fun, fill = Treatment) +
  geom_boxplot(alpha = 0.6)+
   geom_jitter(shape = "circle", alpha = 0.5) +
  scale_fill_manual(values = c(C0 = "#6DD4F8", 
      D1 = "#8CB425", D2 = "#ECE320", D3 = "#DF5220")) + 
  theme_minimal(base_size = 14) +
   ylab(" ")+
  xlab("")+
  stat_compare_means(method = "kruskal.test", label.y = c(0.57), label.x = c(1.5))+
  ggtitle(expression(italic("T. melinonii"))) +
  theme(legend.position = "none")+
  ylim(0.29,0.6)
tm

vs <-NDV_bac  %>%
 filter(Species == "Vir.sur") %>%
 filter(Time %in% c("t21", "t27", "t71")) %>%
 #filter(Treatment != "C0")  %>%
 ggplot() +
  aes(x = Treatment, y = bray_abund_null_dev_fun, fill = Treatment) +
  geom_boxplot(alpha = 0.6)+
   geom_jitter(shape = "circle", alpha = 0.5) +
  scale_fill_manual(values = c(C0 = "#6DD4F8", 
      D1 = "#8CB425", D2 = "#ECE320", D3 = "#DF5220")) + 
  theme_minimal(base_size = 14) +
   ylab("NDV")+
  xlab("")+
  stat_compare_means(method = "kruskal.test", label.y = c(0.57), label.x = 1.5)+
  ggtitle(expression(italic("V. surinamensis"))) +
  theme(legend.position = "none")+
  ylim(0.29,0.6)
  #theme(legend.position = "bottom", legend.box = "vertical", legend.direction = "vertical")
vs

legend <- get_legend(vs)

ndv_bac_drought <- ggarrange(ef, ih, jc, po, sg, tm, vs, as_ggplot(legend), nrow=2, ncol=4)

ndv_bac_drought <- annotate_figure(ndv_bac_drought, top = text_grob("A. Bacteria", color = "black", face = "bold", size = 20, just = "right" , hjust = 3, vjust = 0.5))

ggsave(filename = "./results/microbio/ndv_bac_drought.jpeg", plot = ndv_bac_drought, width = 10, height = 6)
```

##both fungi and bacteria 
```{r}
ndv_drought <- ggarrange(ndv_bac_drought,ndv_fun_drought, nrow=2, ncol=1)
ggsave(filename = "./results/microbio/ndv_drought.jpeg", plot = ndv_drought, width = 10, height = 11)
```




```{r}
#after drought, during the recovery phase

#prepare data
data <- NDV_bac %>%  filter(Treatment != "C0") %>% filter(!is.na(Species)) %>%
  mutate(Period = ifelse(Time %in% c("t21","t27", "t71"), "Drought", 
                  ifelse(Time %in% c("t51", "t57","t101"), "Recovery", NA)))
data$Period <- as.factor(data$Period)
data$Treatment <- as.factor(data$Treatment)

#epefal
ef <- data %>% filter(Species == "Epe.fal")
 ef_plot <- ggboxplot(
  ef, x= "Period" , y = "bray_abund_null_dev_fun", fill = "Period", facet.by = "Treatment", alpha =0.7) +           geom_jitter(shape = "circle", alpha = 0.5) +
  scale_fill_manual(values = c(Drought = "#ff9e00", Recovery = "#219ebc"))+
  theme_classic() +
  theme(axis.text.x = element_blank(), axis.title.x = element_blank())+
  ylab("NDV")+
  theme(legend.position = "none")
ef_plot
stat.test <- data %>%
  filter(Species == "Epe.fal") %>%
  filter(Treatment != "D3") %>%
  group_by(Treatment) %>%
  t_test(bray_abund_null_dev_fun ~ Period) %>%
  adjust_pvalue(method = "bonferroni") %>%
  add_significance()
stat.test 
stat.test <- stat.test %>% add_xy_position(x = "Period") # Make facet and add p-values
ef_plot <- ef_plot + stat_pvalue_manual(stat.test)
ef_plot <- annotate_figure(ef_plot, left = text_grob("E. falcata       ", color = "black", face = "italic", size = 20))

#iryhos
ih <- data %>% filter(Species == "Iry.hos") 
ih_plot <- ggboxplot(
  ih, x= "Period" , y = "bray_abund_null_dev_fun", fill = "Period", facet.by = "Treatment", alpha =0.7) + 
  geom_jitter(shape = "circle", alpha = 0.5) +
  scale_fill_manual(values = c(Drought = "#ff9e00", Recovery = "#219ebc"))+
  theme_classic() +
  theme(axis.text.x = element_blank(), axis.title.x = element_blank())+
  ylab("NDV")+
  theme(legend.position = "none")
ih_plot
stat.test <- ih %>%
  group_by(Species, Treatment) %>%
  t_test(bray_abund_null_dev_fun ~ Period) %>%
  adjust_pvalue(method = "bonferroni") %>%
  add_significance()
stat.test 
stat.test <- stat.test %>% add_xy_position(x = "Period") 
ih_plot <- ih_plot + stat_pvalue_manual(stat.test)
ih_plot <- annotate_figure(ih_plot, right = text_grob("I. hostmanii", color = "black", face = "italic", size = 20))

#jaccop
jc <- data %>% filter(Species == "Jac.cop") 
 jc_plot <- ggboxplot(
   jc, x= "Period" , y = "bray_abund_null_dev_fun", fill = "Period", facet.by = "Treatment", alpha =0.7) + 
   geom_jitter(shape = "circle", alpha = 0.5) +
   scale_fill_manual(values = c(Drought = "#ff9e00", Recovery = "#219ebc"))+
  theme_classic() +
  theme(axis.text.x = element_blank(), axis.title.x = element_blank())+
  ylab("NDV")+
  theme(legend.position = "none")
jc_plot
stat.test <- jc %>%
  group_by(Species, Treatment) %>%
  t_test(bray_abund_null_dev_fun ~ Period) %>%
  adjust_pvalue(method = "bonferroni") %>%
  add_significance()
stat.test 
stat.test <- stat.test %>% add_xy_position(x = "Period")
jc_plot <- jc_plot + stat_pvalue_manual(stat.test)
jc_plot <- annotate_figure(jc_plot, left = text_grob("J. copaia        ", color = "black", face = "italic", size = 20))

#pteoff
po <- data %>% filter(Species == "Pte.off") 
 po_plot <- ggboxplot(
   po, x= "Period" , y = "bray_abund_null_dev_fun", fill = "Period", facet.by = "Treatment", alpha =0.7) + 
  geom_jitter(shape = "circle", alpha = 0.5) +
  scale_fill_manual(values = c(Drought = "#ff9e00", Recovery = "#219ebc"))+
  theme_classic() +
  theme(axis.text.x = element_blank(), axis.title.x = element_blank())+
  ylab("NDV")+
  theme(legend.position = "none")
po_plot
stat.test <- po %>%
  group_by(Species, Treatment) %>%
  t_test(bray_abund_null_dev_fun ~ Period) %>%
  adjust_pvalue(method = "bonferroni") %>%
  add_significance()
stat.test 
stat.test <- stat.test %>% add_xy_position(x = "Period")
po_plot <- po_plot + stat_pvalue_manual(stat.test)
po_plot <- annotate_figure(po_plot, right = text_grob("P. officinalis", color = "black", face = "italic", size = 20))

#symglo
sg <- data %>% filter(Species == "Sym.off") 
 sg_plot <- ggboxplot(
   sg, x= "Period" , y = "bray_abund_null_dev_fun", fill = "Period", facet.by = "Treatment", alpha =0.7) + 
  geom_jitter(shape = "circle", alpha = 0.5) +
  scale_fill_manual(values = c(Drought = "#ff9e00", Recovery = "#219ebc"))+
  theme_classic() +
  theme(axis.text.x = element_blank(), axis.title.x = element_blank())+
  ylab("NDV")+
  theme(legend.position = "none")
sg_plot
stat.test <- sg %>%
  group_by(Species, Treatment) %>%
  t_test(bray_abund_null_dev_fun ~ Period) %>%
  adjust_pvalue(method = "bonferroni") %>%
  add_significance()
stat.test 
stat.test <- stat.test %>% add_xy_position(x = "Period")
sg_plot <- sg_plot + stat_pvalue_manual(stat.test)
sg_plot <- annotate_figure(sg_plot, left = text_grob("S. globulifera  ", color = "black", face = "italic", size = 20))

#tacmel
tm <- data %>% filter(Species == "Tac.mel") 
tm_plot <- ggboxplot(
  tm, x= "Period" , y = "bray_abund_null_dev_fun", fill = "Period", facet.by = "Treatment", alpha =0.7) + 
  geom_jitter(shape = "circle", alpha = 0.5) +
  scale_fill_manual(values = c(Drought = "#ff9e00", Recovery = "#219ebc"))+
  theme_classic() +
  theme(axis.text.x = element_blank(), axis.title.x = element_blank())+
  ylab("NDV")+
  theme(legend.position = "none")
tm_plot
stat.test <- tm %>%
  group_by(Species, Treatment) %>%
  t_test(bray_abund_null_dev_fun ~ Period) %>%
  adjust_pvalue(method = "bonferroni") %>%
  add_significance()
stat.test 
stat.test <- stat.test %>% add_xy_position(x = "Period")
tm_plot <- tm_plot + stat_pvalue_manual(stat.test)
tm_plot <- annotate_figure(tm_plot, right = text_grob("T. melinonii", color = "black", face = "italic", size = 20))

#virsur
vs <- data %>% filter(Species == "Vir.sur") 
vs_plot <- ggboxplot(
  vs, x= "Period" , y = "bray_abund_null_dev_fun", fill = "Period", facet.by = "Treatment", alpha =0.7) + 
  geom_jitter(shape = "circle", alpha = 0.5) +
  scale_fill_manual(values = c(Drought = "#ff9e00", Recovery = "#219ebc"))+
  theme_classic() +
  theme(axis.text.x = element_blank(), axis.title.x = element_blank())+
  ylab("NDV ")+
  theme(legend.position = "none", legend.text = element_text(size = 16),
    legend.title = element_text(size = 20))
vs_plot
stat.test <- vs %>%
  filter(!(Treatment == "D3")) %>%
  group_by(Species, Treatment) %>%
  t_test(bray_abund_null_dev_fun ~ Period) %>%
  adjust_pvalue(method = "bonferroni") %>%
  add_significance()
stat.test 
stat.test <- stat.test %>% add_xy_position(x = "Period") 
vs_plot <- vs_plot + stat_pvalue_manual(stat.test)
vs_plot <- annotate_figure(vs_plot, left = text_grob("V. surinamensis", color = "black", face = "italic", size = 20))

#get legend
legend <- get_legend(vs_plot) #get legend work with no complicated plots fso just grab the legend when plot is simpler
as_ggplot(legend)

#plot all
ndv_recov_bac <- ggarrange(ef_plot, ih_plot, jc_plot, po_plot, sg_plot, tm_plot, vs_plot, as_ggplot(legend), ncol=2, nrow = 4)
ndv_recov_bac
ndv_recov_bac <- annotate_figure(ndv_recov_bac, top = text_grob("Stochastic and deterministic processes of bacterial communities during and after drought", color = "black", face = "bold", size = 20))
ggsave(filename = "./results/microbio/ndv_bac_recov.jpeg", plot = ndv_recov_bac, width = 12, height = 10)

```


#Resistance all together
```{r}
#first run separate codes for fungi and bacteria, then merge tables for only plot

#for fungi
fungi <- fungi %>% 
  mutate(Kingdom = "fungi")

#for bacteria
bacteria <- bacteria %>%
  mutate(Kingdom = "bacteria")

#both
resistance <- rbind(fungi, bacteria)

resistance <- read.csv("resistance_kingdom.csv")

#plot
ef <- resistance %>%
  filter(Species == "Epe.fal") %>%
  ggplot() +
 aes(x = Treatment, y = Resistance,  colour = Kingdom) +
 ylab("Resistance")+
  geom_jitter(shape = "circle", size = 5,  alpha = 0.5)+
 scale_color_manual(values = c( 
      bacteria = "#BB17A0", fungi = "#50DABD")) +
 labs(title = "E. falcata") +
 xlab(" ")+
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
 theme_minimal(base_size =20)+
 theme(title = element_text(face = "italic"), plot.title =   element_text(hjust = 0.5))+
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
 theme(legend.position = "none")
ef
ih <- resistance %>%
  filter(Species == "Iry.hos") %>%
  ggplot() +
 aes(x = Treatment, y = Resistance,  colour = Kingdom) +
 ylab("Resistance")+
  geom_jitter(shape = "circle", size = 5,  alpha = 0.5)+
 scale_color_manual(values = c( 
      bacteria = "#BB17A0", fungi = "#50DABD")) +
 labs(title = "I. hostmanii") +
 xlab(" ")+
  ylab(" ")+
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
 theme_minimal(base_size =20)+
 theme(title = element_text(face = "italic"), plot.title =   element_text(hjust = 0.5))+
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
 theme(legend.position = "none")
ih

jc <- resistance %>%
  filter(Species == "Jac.cop") %>%
 ggplot() +
 aes(x = Treatment, y = Resistance,  colour = Kingdom) +
 ylab("Resistance")+
  geom_jitter(shape = "circle", size = 5,  alpha = 0.5)+
 scale_color_manual(values = c( 
      bacteria = "#BB17A0", fungi = "#50DABD")) +
 labs(title = "J. copaia") +
 xlab(" ")+
  ylab(" ")+
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
 theme_minimal(base_size =20)+
 theme(title = element_text(face = "italic"), plot.title =   element_text(hjust = 0.5))+
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
 theme(legend.position = "none")
jc

po <- resistance %>%
  filter(Species == "Pte.off") %>%
  ggplot() +
 aes(x = Treatment, y = Resistance,  colour = Kingdom) +
 ylab("Resistance")+
geom_jitter(shape = "circle", size = 5,  alpha = 0.5)+
 scale_color_manual(values = c( 
      bacteria = "#BB17A0", fungi = "#50DABD")) +
 labs(title = "P. officinalis") +
 xlab(" ")+
  ylab(" ")+
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
 theme_minimal(base_size =20)+
 theme(title = element_text(face = "italic"), plot.title =   element_text(hjust = 0.5))+
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
   theme(legend.position = "none")
po

sg <- resistance %>%
  filter(Species == "Sym.off") %>%
  ggplot() +
 aes(x = Treatment, y = Resistance,  colour = Kingdom) +
 ylab("Resistance")+
 geom_jitter(shape = "circle", size = 5,  alpha = 0.5)+
 scale_color_manual(values = c( 
      bacteria = "#BB17A0", fungi = "#50DABD")) +
 labs(title = "S. globulifera") +
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
 theme_minimal(base_size =20)+
 theme(title = element_text(face = "italic"), plot.title =   element_text(hjust = 0.5))+
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
   theme(legend.position = "none")
sg

tm <- resistance %>%
  filter(Species == "Tac.mel") %>%
  ggplot() +
 aes(x = Treatment, y = Resistance,  colour = Kingdom) +
 ylab("Resistance")+
 geom_jitter(shape = "circle", size = 5,  alpha = 0.5)+
 scale_color_manual(values = c( 
      bacteria = "#BB17A0", fungi = "#50DABD")) +
 labs(title = "T. melinonii") +
  ylab(" ")+
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
 theme_minimal(base_size =20)+
 theme(title = element_text(face = "italic"), plot.title =   element_text(hjust = 0.5))+
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
   theme(legend.position = "none")
tm

vs <- resistance %>%
  filter(Species == "Vir.sur") %>%
  ggplot() +
 aes(x = Treatment, y = Resistance,  colour = Kingdom) +
 ylab("Resistance")+
 geom_jitter(shape = "circle", size = 5,  alpha = 0.5) +
scale_color_manual(values = c( 
      bacteria = "#BB17A0", fungi = "#50DABD")) +
 labs(title = "V. surinamensis") +
  ylab(" ")+
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
 theme_minimal(base_size =20)+
 theme(title = element_text(face = "italic"), plot.title =   element_text(hjust = 0.5))+
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
   theme(legend.position = "none", legend.direction = "vertical")
vs

#get legend
legend <- get_legend(vs)
legend <- as_ggplot(legend)


resistance_kingdom <- ggarrange(ef, ih, jc, po, sg, tm, vs, legend, nrow = 2, ncol = 4)
resistance_kingdom
ggsave(filename = "./results/microbio/resistance_kingdom.jpeg", plot = resistance_kingdom, width = 12, height = 10)


#write csv
write.csv(resistance, "resistance_kingdom.csv")
```

#Recovery all together
```{r}
#first run separate codes for fungi and bacteria, then merge tables for only plot

#for fungi
fungi <- fungi %>% 
  mutate(Kingdom = "fungi")

#for bacteria
bacteria <- bacteria %>%
  mutate(Kingdom = "bacteria")

#both
Recovery <- rbind(fungi, bacteria)

#plot
ef <- Recovery %>%
  filter(Species == "Epe.fal") %>%
  ggplot() +
 aes(x = Treatment, y = Recovery,  colour = Kingdom) +
 ylab("Recovery")+
  geom_jitter(shape = "circle", size = 5,  alpha = 0.5)+
 scale_color_manual(values = c( 
      bacteria = "#BB17A0", fungi = "#50DABD")) +
 labs(title = "E. falcata") +
 xlab(" ")+
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
 theme_minimal(base_size =20)+
 theme(title = element_text(face = "italic"), plot.title =   element_text(hjust = 0.5))+
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
 theme(legend.position = "none")
ef
ih <- Recovery %>%
  filter(Species == "Iry.hos") %>%
  ggplot() +
 aes(x = Treatment, y = Recovery,  colour = Kingdom) +
 ylab("Recovery")+
  geom_jitter(shape = "circle", size = 5,  alpha = 0.5)+
 scale_color_manual(values = c( 
      bacteria = "#BB17A0", fungi = "#50DABD")) +
 labs(title = "I. hostmanii") +
 xlab(" ")+
  ylab(" ")+
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
 theme_minimal(base_size =20)+
 theme(title = element_text(face = "italic"), plot.title =   element_text(hjust = 0.5))+
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
 theme(legend.position = "none")
ih

jc <- Recovery %>%
  filter(Species == "Jac.cop") %>%
 ggplot() +
 aes(x = Treatment, y = Recovery,  colour = Kingdom) +
 ylab("Recovery")+
  geom_jitter(shape = "circle", size = 5,  alpha = 0.5)+
 scale_color_manual(values = c( 
      bacteria = "#BB17A0", fungi = "#50DABD")) +
 labs(title = "J. copaia") +
 xlab(" ")+
  ylab(" ")+
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
 theme_minimal(base_size =20)+
 theme(title = element_text(face = "italic"), plot.title =   element_text(hjust = 0.5))+
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
 theme(legend.position = "none")
jc

po <- Recovery %>%
  filter(Species == "Pte.off") %>%
  ggplot() +
 aes(x = Treatment, y = Recovery,  colour = Kingdom) +
 ylab("Recovery")+
geom_jitter(shape = "circle", size = 5,  alpha = 0.5)+
 scale_color_manual(values = c( 
      bacteria = "#BB17A0", fungi = "#50DABD")) +
 labs(title = "P. officinalis") +
 xlab(" ")+
  ylab(" ")+
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
 theme_minimal(base_size =20)+
 theme(title = element_text(face = "italic"), plot.title =   element_text(hjust = 0.5))+
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
   theme(legend.position = "none")
po

sg <- Recovery %>%
  filter(Species == "Sym.off") %>%
  ggplot() +
 aes(x = Treatment, y = Recovery,  colour = Kingdom) +
 ylab("Recovery")+
 geom_jitter(shape = "circle", size = 5,  alpha = 0.5)+
 scale_color_manual(values = c( 
      bacteria = "#BB17A0", fungi = "#50DABD")) +
 labs(title = "S. globulifera") +
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
 theme_minimal(base_size =20)+
 theme(title = element_text(face = "italic"), plot.title =   element_text(hjust = 0.5))+
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
   theme(legend.position = "none")
sg

tm <- Recovery %>%
  filter(Species == "Tac.mel") %>%
  ggplot() +
 aes(x = Treatment, y = Recovery,  colour = Kingdom) +
 ylab("Recovery")+
 geom_jitter(shape = "circle", size = 5,  alpha = 0.5)+
 scale_color_manual(values = c( 
      bacteria = "#BB17A0", fungi = "#50DABD")) +
 labs(title = "T. melinonii") +
  ylab(" ")+
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
 theme_minimal(base_size =20)+
 theme(title = element_text(face = "italic"), plot.title =   element_text(hjust = 0.5))+
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
   theme(legend.position = "none")
tm

vs <- Recovery %>%
  filter(Species == "Vir.sur") %>%
  ggplot() +
 aes(x = Treatment, y = Recovery,  colour = Kingdom) +
 ylab("Recovery")+
 geom_jitter(shape = "circle", size = 5,  alpha = 0.5) +
scale_color_manual(values = c( 
      bacteria = "#BB17A0", fungi = "#50DABD")) +
 labs(title = "V. surinamensis") +
  ylab(" ")+
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
 theme_minimal(base_size =20)+
 theme(title = element_text(face = "italic"), plot.title =   element_text(hjust = 0.5))+
 geom_hline(yintercept = 0, linetype = "dashed", color = "red")+
   theme(legend.position = "none", legend.direction = "vertical")
vs

#get legend
legend <- get_legend(vs)
legend <- as_ggplot(legend)


Recovery_kingdom <- ggarrange(ef, ih, jc, po, sg, tm, vs, legend, nrow = 2, ncol = 4)
Recovery_kingdom
ggsave(filename = "./results/microbio/Recovery_kingdom.jpeg", plot = Recovery_kingdom, width = 12, height = 10)
```


#RESILIENCE
```{r}

resilience_fun_bac <- left_join(resistance, Recovery)

resilience_fun_bac$facet_label <- NA
resilience_fun_bac$facet_label[resilience_fun_bac$Species == "Epe.fal"] <- "E. falcata"
resilience_fun_bac$facet_label[resilience_fun_bac$Species == "Iry.hos"] <- "I. hostmannii"
resilience_fun_bac$facet_label[resilience_fun_bac$Species == "Jac.cop"] <- "J. copaia"
resilience_fun_bac$facet_label[resilience_fun_bac$Species == "Pte.off"] <- "P. officinalis"
resilience_fun_bac$facet_label[resilience_fun_bac$Species == "Sym.off"] <- "S. globulifera"
resilience_fun_bac$facet_label[resilience_fun_bac$Species == "Tac.mel"] <- "T. melinonii"
resilience_fun_bac$facet_label[resilience_fun_bac$Species == "Vir.sur"] <- "V. surinamensis"

#at D1
Resilience_D1 <- resilience_fun_bac %>% dplyr::filter(Treatment == "D1") %>% dplyr::select(-Species) %>% rename( Species =facet_label )

Resilience_D2 <- resilience_fun_bac %>% filter(Treatment == "D2")  %>% dplyr::select(-Species) %>% rename( Species =facet_label )

Resilience_D3 <- resilience_fun_bac %>% filter(Treatment == "D3")  %>% dplyr::select(-Species) %>% rename( Species =facet_label )


# Create the biplot D1 
bac_D1 <- Resilience_D1 %>% dplyr::filter(Kingdom == "bacteria") %>%
ggplot(aes(x = Resistance, y =Recovery)) +
  geom_point(size= 4,aes(color = Species)) +
#geom_text_repel(aes(label = facet_label), box.padding = 0.5, point.padding = 0.3, size = 7)+
  labs(
       x = "",
       y = "Recovery Score")+
   geom_hline(yintercept = 0, linetype = "dashed", color = "red") +
   geom_vline(xintercept = 0, linetype = "dashed", color = "red") +
   geom_segment( aes(x=0, xend=1,y=1, yend=0),linewidth = 1,color = "#34a0a4")+
  # ylim(c(-0.8,1))+
  # xlim(c(0,1))+
  theme_minimal(base_size = 22)+
  guides(color= "none")+ggtitle ("Bacteria")+ theme(title = element_text(face= "italic"))

fun_D1 <- Resilience_D1 %>% dplyr::filter(Kingdom == "fungi") %>%
ggplot(aes(x = Resistance, y =Recovery)) +
  geom_point(size= 4,aes(color = Species)) +
#geom_text_repel(aes(label = facet_label), box.padding = 0.5, point.padding = 0.3, size = 7)+
  labs(
       x = "",
       y = "")+
   geom_hline(yintercept = 0, linetype = "dashed", color = "red") +
   geom_vline(xintercept = 0, linetype = "dashed", color = "red") +
   geom_segment( aes(x=0, xend=1,y=1, yend=0),linewidth = 1,color = "#34a0a4")+
  # ylim(c(-0.8,1))+
  # xlim(c(0,1))+
  theme_minimal(base_size = 22)+
  guides(color= "none")+ggtitle ("Fungi")+ theme(title = element_text(face= "italic"))

# Create a text grob for the title
title_D1 <- "Current drought scenario"
tgrob <- ggpubr::text_grob(title_D1,size = 24, color = "#8CB425")
# Draw the text
plot_0 <- as_ggplot(tgrob) + theme(plot.margin = margin(0,0,0,0, "cm"))

bac_fun_D1 <- ggarrange(bac_D1,fun_D1, nrow=1)

Resilience_plot_D1 <- ggarrange(plot_0, bac_fun_D1, nrow = 2, heights = c(0.5,2)) 
Resilience_plot_D1


#######################################################
#at D2

# Create the biplot D2 
bac_D2 <- Resilience_D2 %>% dplyr::filter(Kingdom == "bacteria") %>%
ggplot(aes(x = Resistance, y =Recovery)) +
  geom_point(size= 4,aes(color = Species)) +
#geom_text_repel(aes(label = facet_label), box.padding = 0.5, point.padding = 0.3, size = 7)+
  labs(
       x = "",
       y = "Recovery Score")+
   geom_hline(yintercept = 0, linetype = "dashed", color = "red") +
   geom_vline(xintercept = 0, linetype = "dashed", color = "red") +
   geom_segment( aes(x=0, xend=1,y=1, yend=0),linewidth = 1,color = "#34a0a4")+
  #  ylim(c(-0.8,1))+
  # xlim(c(0,1))+
  theme_minimal(base_size = 22)+
  guides(color = "none")

fun_D2 <- Resilience_D2 %>% dplyr::filter(Kingdom == "fungi") %>%
ggplot(aes(x = Resistance, y =Recovery)) +
  geom_point(size= 4,aes(color = Species)) +
#geom_text_repel(aes(label = facet_label), box.padding = 0.5, point.padding = 0.3, size = 7)+
  labs(
       x = "",
       y = "")+
   geom_hline(yintercept = 0, linetype = "dashed", color = "red") +
   geom_vline(xintercept = 0, linetype = "dashed", color = "red") +
   geom_segment( aes(x=0, xend=1,y=1, yend=0),linewidth = 1,color = "#34a0a4")+
  # ylim(c(-0.8,1))+
  # xlim(c(0,1))+
  theme_minimal(base_size = 22)+
  guides(color= "none")

# Create a text grob for the title
title_D2 <- "Projected drought scenario"
tgrob <- ggpubr::text_grob(title_D2,size = 24, color = "#eeba0b")
# Draw the text
plot_0 <- as_ggplot(tgrob) + theme(plot.margin = margin(0,0,0,0, "cm"))
bac_fun_D2 <- ggarrange(bac_D2,fun_D2, nrow=1)
Resilience_plot_D2 <- ggarrange(plot_0, bac_fun_D2, nrow = 2, heights = c(0.5,2,2)) 
Resilience_plot_D2


#######################################################


#at D3

# Create the biplot D3 
bac_D3 <- Resilience_D3 %>% filter(Kingdom == "bacteria") %>%
ggplot(aes(x = Resistance, y = Recovery)) +
  geom_point(size= 4,aes(color = Species)) +
#geom_text(aes(label = facet_label), vjust = 1.5) +
 #geom_text_repel(aes(label = facet_label), box.padding = 0.5, point.padding = 0.3, size = 7)+  # Use geom_text_repel
    labs(
       x = "Resistance Score",
       y = "Recovery score")+
   geom_hline(yintercept = 0, linetype = "dashed", color = "red") +
   geom_vline(xintercept = 0, linetype = "dashed", color = "red") +
   geom_segment( aes(x=0, xend=1,y=1, yend=0),linewidth = 1,color = "#34a0a4")+
  # ylim(c(-0.8,1))+
  # xlim(c(0,1))+
 guides(color = guide_legend(title = "Species"))+
  theme_minimal(base_size = 22)+
  theme(legend.text = element_text(face= "italic"))+
  guides(color= "none")

fun_D3 <- Resilience_D3 %>% filter(Kingdom == "fungi") %>%
ggplot(aes(x = Resistance, y = Recovery)) +
  geom_point(size= 4,aes(color = Species)) +
#geom_text(aes(label = facet_label), vjust = 1.5) +
 #geom_text_repel(aes(label = facet_label), box.padding = 0.5, point.padding = 0.3, size = 7)+  # Use geom_text_repel
    labs(
       x = "Resistance Score",
       y = "")+
   geom_hline(yintercept = 0, linetype = "dashed", color = "red") +
   geom_vline(xintercept = 0, linetype = "dashed", color = "red") +
   geom_segment( aes(x=0, xend=1,y=1, yend=0),linewidth = 1,color = "#34a0a4")+
  # ylim(c(-0.8,1))+
  # xlim(c(0,1))+
 guides(color = guide_legend(title = "Species"))+
  theme_minimal(base_size = 22)+
  theme(legend.text = element_text(face= "italic"))+
  guides(color= "none")

# Create a text grob for the title
title_D3 <- "Extreme drought scenario"
tgrob <- ggpubr::text_grob(title_D3,size = 24, color = "#DF5220")
# Draw the text
plot_1 <- as_ggplot(tgrob) + theme(plot.margin = margin(0,0,0,0, "cm"))
bac_fun_D3 <- ggarrange(bac_D3,fun_D3, nrow=1)
Resilience_plot_D3 <- ggarrange(plot_1, bac_fun_D3,  nrow = 2, legend = "bottom", common.legend = TRUE, heights = c(0.5,2) ) 
Resilience_plot_D3

Resilience_plot <- ggarrange(Resilience_plot_D1, Resilience_plot_D2, Resilience_plot_D3, ncol=1, nrow= 3, common.legend = TRUE, legend = "bottom")
Resilience_plot 

ggsave(filename = "Resilience_plot.jpeg", plot = Resilience_plot, width = 10, height = 12)
```


#BETA PCOA
```{r}
#bacteria
load("/DRYER_greenhouse/results/Metabarlist_serre_16S_AMVG51-230505_clean.Rdata")
bacteria <- AMVG51_230505_16S_clean
rownames(bacteria$samples) <- bacteria$samples$sample_id

#fungi
load("./results/Metabarlist_serre_ITS_AMVG42-230505_clean.Rdata")
fungi <- AMVG42_230505_ITS_clean
rownames(fungi$samples) <- fungi$samples$sample_id
```

##phyloseq drought
```{r}
bacteria$samples <-bacteria$samples %>%
   mutate(Period = case_when(
     Treatment == "D1" &  Time == "t21" ~ "Drought D1",
      Treatment == "D1" &  Time == "t51" ~ "Recovery D1",
      Treatment == "D2" &  Time == "t27" ~ "Drought D2",
      Treatment == "D2" &  Time == "t57" ~ "Recovery D2",
      Treatment == "D3" &  Time == "t71" ~ "Drought D3",
      Treatment == "D3" &  Time == "t101" ~ "Recovery D3",
      TRUE                      ~ "Control"
    )
  )
bacteria$samples$facet_label <- NA
bacteria$samples$facet_label[bacteria$samples$Species == "Epe.fal"] <- "E. falcata"
bacteria$samples$facet_label[bacteria$samples$Species == "Iry.hos"] <- "I. hostmanii"
bacteria$samples$facet_label[bacteria$samples$Species == "Jac.cop"] <- "J. copaia"
bacteria$samples$facet_label[bacteria$samples$Species == "Pte.off"] <- "P. officinalis"
bacteria$samples$facet_label[bacteria$samples$Species == "Sym.off"] <- "S. globulifera"
bacteria$samples$facet_label[bacteria$samples$Species == "Tac.mel"] <- "T. melinonii"
bacteria$samples$facet_label[bacteria$samples$Species == "Vir.sur"] <- "V. surinamensis"
bacteria <- subset_metabarlist(metabarlist = bacteria, table = "samples", indices = bacteria$samples$Time %in% c("t21", "t27", "t71"))
otu_mat<- as.data.frame(t(bacteria$reads))
tax_mat<- bacteria$motus %>% dplyr::select(Phylum, Class, Order, Family, Genus)
samples_df <-  bacteria$samples %>% dplyr::select(Treatment, Time, Species, Period, facet_label)
otu_mat <- as.matrix(otu_mat)
tax_mat <- as.matrix(tax_mat)
OTU = otu_table(otu_mat, taxa_are_rows = TRUE)
TAX = tax_table(tax_mat)
samples = sample_data(samples_df)
drought_bacteria <- phyloseq(OTU, TAX, samples)
drought_bacteria
  
#fungi
fungi$samples <-fungi$samples %>%
   mutate(Period = case_when(
     Treatment == "D1" &  Time == "t21" ~ "Drought D1",
      Treatment == "D1" &  Time == "t51" ~ "Recovery from D1",
      Treatment == "D2" &  Time == "t27" ~ "Drought D2",
      Treatment == "D2" &  Time == "t57" ~ "Recoveryfrom  D2",
      Treatment == "D3" &  Time == "t71" ~ "Drought D3",
      Treatment == "D3" &  Time == "t101" ~ "Recovery from D3",
      TRUE                      ~ "Control"
    )
  )
fungi$samples$facet_label <- NA
fungi$samples$facet_label[fungi$samples$Species == "Epe.fal"] <- "E. falcata"
fungi$samples$facet_label[fungi$samples$Species == "Iry.hos"] <- "I. hostmanii"
fungi$samples$facet_label[fungi$samples$Species == "Jac.cop"] <- "J. copaia"
fungi$samples$facet_label[fungi$samples$Species == "Pte.off"] <- "P. officinalis"
fungi$samples$facet_label[fungi$samples$Species == "Sym.off"] <- "S. globulifera"
fungi$samples$facet_label[fungi$samples$Species == "Tac.mel"] <- "T. melinonii"
fungi$samples$facet_label[fungi$samples$Species == "Vir.sur"] <- "V. surinamensis"
fungi <- subset_metabarlist(metabarlist = fungi, table = "samples", indices = fungi$samples$Time %in% c("t21", "t27", "t71"))
otu_mat<- as.data.frame(t(fungi$reads))
tax_mat<- fungi$motus %>% dplyr::select(Phylum, Class, Order, Family, Genus)
samples_df <-  fungi$samples %>% dplyr::select(Treatment, Time, Species, Period, facet_label)
otu_mat <- as.matrix(otu_mat)
tax_mat <- as.matrix(tax_mat)
OTU = otu_table(otu_mat, taxa_are_rows = TRUE)
TAX = tax_table(tax_mat)
samples = sample_data(samples_df)
drought_fungi <- phyloseq(OTU, TAX, samples)
drought_fungi
```


##prepare data - transformation
```{r}
#Normalize number of reads in each sample using median sequencing depth

#bacteria
total = median(sample_sums(drought_bacteria))#3007
standf = function(x, t=total) round(t * (x / sum(x)))
drought_bacteria = transform_sample_counts(drought_bacteria, standf)

#fungi
total = median(sample_sums(drought_fungi))#8589
standf = function(x, t=total) round(t * (x / sum(x)))
drought_fungi = transform_sample_counts(drought_fungi, standf)
```

## alpha diversity
```{r}
R_bacteria <- estimate_richness(drought_bacteria, split= TRUE, measures = NULL)
plot_richness(drought_bacteria, x="Period", measures="Shannon", color = "Period")+
  geom_boxplot(alpha=0.6)+ 
  theme(legend.position="none", axis.text.x=element_text(angle=45,hjust=1,vjust=1,size=12))+ facet_wrap(~Species)
hist(R_bacteria$Shannon, main="Shannon index", xlab="") #yes normally distributed
anova.sh = aov(R_bacteria$Shannon ~ sample_data(drought_bacteria)$Period)
summary(anova.sh) #p-value; 0.446

#with hill
Shannon_bac <-hill_taxa(t(drought_bacteria@otu_table),q=1)
df <- Shannon_bac %>% 
  enframe() %>%
  rename(Shannon_bac = value, 
         sample_id = name)
drought_bacteria@sam_data$sample_id <- rownames(drought_bacteria@sam_data)
drought_bacteria@sam_data<- left_join(drought_bacteria@sam_data, df)
meta_bac<- left_join(df, bacteria$samples)
meta_bac%>%
ggplot() +
  aes(x = Period, y = Shannon_bac) +
  geom_boxplot(fill = "#112446") +
  theme_minimal() +
  facet_wrap(vars(Species))
hist(Shannon_bac, main="Shannon index", xlab="") #yes normally distributed
anova.sh = aov(Shannon_bac ~ sample_data(drought_bacteria)$Period)
summary(anova.sh) #p-value;  0.252

#fungi
R_fungi <- estimate_richness(drought_fungi, split= TRUE, measures = NULL)
plot_richness(drought_fungi, x="Period", measures="Shannon", color = "Period")+
  geom_boxplot(alpha=0.6)+ 
  theme(legend.position="none", axis.text.x=element_text(angle=45,hjust=1,vjust=1,size=12))+ facet_wrap(~Species)
hist(R_fungi$Shannon, main="Shannon index", xlab="") #not normally distributed
kruskal.test(R_fungi$Shannon~ sample_data(drought_fungi)$Period)#0.4604

#with hill
Shannon_fun <-hill_taxa(t(drought_fungi@otu_table),q=1)
df <- Shannon_fun %>% 
  enframe() %>%
  rename(Shannon_fun = value, 
         sample_id = name)
drought_fungi@sam_data$sample_id <- rownames(drought_fungi@sam_data)
drought_fungi@sam_data<- left_join(drought_fungi@sam_data, df)
meta_fun<- left_join(df, fungi$samples)
meta_fun%>%
ggplot() +
  aes(x = Period, y = Shannon_fun) +
  geom_boxplot(fill = "#112446") +
  theme_minimal() +
  facet_wrap(vars(Species))
hist(Shannon_fun, main="Shannon index", xlab="") #not normally distributed
kruskal.test(Shannon_fun~ sample_data(drought_fungi)$Period)#0.4604 as well
```

##beta diversity bacteria
```{r}
#bacteria
relab_genera = transform_sample_counts(drought_bacteria, function(x) x / sum(x) * 100) 
#or for relative abundance normalization microbiome::transform(asv.whole, transform = "compositional")
abrel_bray <- phyloseq::distance(relab_genera, method = "bray")
ord = ordinate(relab_genera, method="PCoA", distance = "bray")

bacteria_drought_ordi <- plot_ordination(relab_genera, ord, color= "Period", shape="facet_label") + 
  geom_point(size=4) + scale_shape_manual("Host species", values = c(0 ,2, 3, 7 ,8, 19, 15)) +  stat_ellipse(aes(group=Period)) +
 scale_color_manual("Treatment", values = c("#6DA0F8", "#8CB425",  "#ECE320", "#DF5220")) + theme_minimal()+ ggtitle("A. Bacteria") +theme(legend.position = "bottom", legend.direction = "vertical")+ guides(fill=guide_legend(nrow=4))
bacteria_drought_ordi

samples <- data.frame(sample_data(relab_genera))
adonis2(abrel_bray ~ Period, data = samples)#0.002 ** This result means that the adonis test is significant, therefore the null hypothesis in which the Control/Treatments have the same centroid was reject.
adonis2(abrel_bray ~ Species, data = samples)
set.seed(123); pairwise.adonis2(abrel_bray ~ Period, data = samples)

#species
bacteria_species_ordi <- plot_ordination(relab_genera, ord, color= "facet_label", shape="facet_label") + 
  geom_point(size=4) +  stat_ellipse(aes(group=facet_label)) + theme_minimal()+ ggtitle("C. Bacteria") +theme(legend.position = "bottom", legend.direction = "vertical")+ guides(fill=guide_legend(nrow=4))+theme(legend.title = element_blank())
bacteria_species_ordi
```

$parent_call
[1] "abrel_bray ~ Period , strata = Null , permutations 999"

$`Drought D1_vs_Control`
         Df SumOfSqs      R2      F Pr(>F)  
Period    1   0.1705 0.01817 1.4988   0.09 .
Residual 81   9.2159 0.98183                
Total    82   9.3864 1.00000                

Signif. codes:  0 *** 0.001 ** 0.01 * 0.05 . 0.1   1

$`Drought D1_vs_Drought D2`
         Df SumOfSqs      R2     F Pr(>F)  
Period    1   0.2069 0.03256 2.053  0.011 *
Residual 61   6.1465 0.96744               
Total    62   6.3534 1.00000               

Signif. codes:  0 *** 0.001 ** 0.01 * 0.05 . 0.1   1

$`Drought D1_vs_Drought D3`
         Df SumOfSqs      R2      F Pr(>F)    
Period    1   0.4200 0.06394 3.4155  0.001 ***
Residual 50   6.1489 0.93606                  
Total    51   6.5689 1.00000                  

Signif. codes:  0 *** 0.001 ** 0.01 * 0.05 . 0.1   1

$`Control_vs_Drought D2`
         Df SumOfSqs      R2      F Pr(>F)
Period    1   0.1229 0.01387 1.0971  0.328
Residual 78   8.7363 0.98613              
Total    79   8.8591 1.00000              

$`Control_vs_Drought D3`
         Df SumOfSqs     R2     F Pr(>F)  
Period    1   0.2684 0.0298 2.058   0.03 *
Residual 67   8.7386 0.9702               
Total    68   9.0070 1.0000               

Signif. codes:  0 *** 0.001 ** 0.01 * 0.05 . 0.1   1

$`Drought D2_vs_Drought D3`
         Df SumOfSqs     R2      F Pr(>F)  
Period    1   0.2154 0.0366 1.7856  0.038 *
Residual 47   5.6692 0.9634                
Total    48   5.8846 1.0000                

Signif. codes:  0 *** 0.001 ** 0.01 * 0.05 . 0.1   1

attr(,"class")
[1] "pwadstrata" "list" 

##beta diveristy fungi
```{r}
#fungi
relab_genera_fungi = transform_sample_counts(drought_fungi, function(x) x / sum(x) * 100) 
#or for relative abundance normalization microbiome::transform(asv.whole, transform = "compositional")
abrel_bray <- phyloseq::distance(relab_genera_fungi, method = "bray")
ord = ordinate(relab_genera_fungi, method="PCoA", distance = "bray")

fungi_drought_ordi <-plot_ordination(relab_genera_fungi, ord, color= "Period", shape="facet_label") + 
  geom_point(size=4) + scale_shape_manual("Host species", values = c(0 ,2, 3, 7 ,8, 19, 15)) +  stat_ellipse(aes(group=Period)) +
 scale_color_manual("Treatment", values = c("#6DA0F8", "#8CB425",  "#ECE320", "#DF5220")) + theme_minimal()+ ggtitle("B. Fungi") +theme(legend.position = "bottom", legend.direction = "vertical")+ guides(fill=guide_legend(nrow=4))
fungi_drought_ordi 

samples <- data.frame(sample_data(relab_genera_fungi))
adonis2(abrel_bray ~ Period, data = samples)#0.009 ** This result means that the adonis test is significant, therefore the null hypothesis in which the Control/Treatments have the same centroid was reject.
adonis2(abrel_bray ~ Species, data = samples)
set.seed(123);pairwise.adonis2(abrel_bray ~ Period, data = samples)


#species
fungi_species_ordi <- plot_ordination(relab_genera_fungi, ord, color= "facet_label", shape="facet_label") + 
  geom_point(size=4) +  stat_ellipse(aes(group=facet_label)) + theme_minimal()+ ggtitle("D. fungi") +theme(legend.position = "bottom", legend.direction = "vertical")+ guides(fill=guide_legend(nrow=4))+theme(legend.title = element_blank())
fungi_species_ordi
```
$parent_call
[1] "fungi_bray ~ Period , strata = Null , permutations 999"

$`Drought D1_vs_Control`
         Df SumOfSqs      R2      F Pr(>F)  
Period    1    0.505 0.01736 1.4135  0.088 .
Residual 80   28.584 0.98264                
Total    81   29.090 1.00000                

Signif. codes:  0 *** 0.001 ** 0.01 * 0.05 . 0.1   1

$`Drought D1_vs_Drought D2`
         Df SumOfSqs    R2      F Pr(>F)  
Period    1   0.6265 0.028 1.7283  0.019 *
Residual 60  21.7505 0.972                
Total    61  22.3770 1.000                

Signif. codes:  0 *** 0.001 ** 0.01 * 0.05 . 0.1   1

$`Drought D1_vs_Drought D3`
         Df SumOfSqs      R2      F Pr(>F)  
Period    1   0.6026 0.02973 1.7769  0.038 *
Residual 58  19.6691 0.97027                
Total    59  20.2717 1.00000                

Signif. codes:  0 *** 0.001 ** 0.01 * 0.05 . 0.1   1

$`Control_vs_Drought D2`
         Df SumOfSqs      R2      F Pr(>F)
Period    1    0.371 0.01294 1.0223  0.379
Residual 78   28.309 0.98706              
Total    79   28.680 1.00000              

$`Control_vs_Drought D3`
         Df SumOfSqs      R2      F Pr(>F)  
Period    1    0.516 0.01929 1.4953  0.066 .
Residual 76   26.228 0.98071                
Total    77   26.744 1.00000                

Signif. codes:  0 *** 0.001 ** 0.01 * 0.05 . 0.1   1

$`Drought D2_vs_Drought D3`
         Df SumOfSqs      R2      F Pr(>F)  
Period    1   0.5811 0.02909 1.6779  0.024 *
Residual 56  19.3940 0.97091                
Total    57  19.9751 1.00000                

Signif. codes:  0 *** 0.001 ** 0.01 * 0.05 . 0.1   1

attr(,"class")
[1] "pwadstrata" "list"   
##beta plot
```{r}
beta_drought <- ggarrange(bacteria_drought_ordi, fungi_drought_ordi, nrow = 1, ncol = 2, common.legend = T, legend = "right")

beta_drought_species<- ggarrange(bacteria_species_ordi, fungi_species_ordi, nrow = 1, ncol = 2, common.legend = T, legend = "right")

beta_drought_all <- ggarrange(beta_drought, beta_drought_species, nrow=2)

ggsave(filename = "./results/microbio/beta_drought_all.jpeg", plot = beta_drought_all, width = 8,height = 8)
```

##phyloseq recovery
```{r}
bacteria$samples <-bacteria$samples %>%
   mutate(Period = case_when(
     Treatment == "D1" &  Time == "t21" ~ "Drought D1",
      Treatment == "D1" &  Time == "t51" ~ "Recovery from D1",
      Treatment == "D2" &  Time == "t27" ~ "Drought D2",
      Treatment == "D2" &  Time == "t57" ~ "Recovery from D2",
      Treatment == "D3" &  Time == "t71" ~ "Drought D3",
      Treatment == "D3" &  Time == "t101" ~ "Recovery from D3",
      TRUE                      ~ "Control"
    )
  )
bacteria$samples$facet_label <- NA
bacteria$samples$facet_label[bacteria$samples$Species == "Epe.fal"] <- "E. falcata"
bacteria$samples$facet_label[bacteria$samples$Species == "Iry.hos"] <- "I. hostmanii"
bacteria$samples$facet_label[bacteria$samples$Species == "Jac.cop"] <- "J. copaia"
bacteria$samples$facet_label[bacteria$samples$Species == "Pte.off"] <- "P. officinalis"
bacteria$samples$facet_label[bacteria$samples$Species == "Sym.off"] <- "S. globulifera"
bacteria$samples$facet_label[bacteria$samples$Species == "Tac.mel"] <- "T. melinonii"
bacteria$samples$facet_label[bacteria$samples$Species == "Vir.sur"] <- "V. surinamensis"
# bacteria <- subset_metabarlist(metabarlist = bacteria, table = "samples", indices = bacteria$samples$Time %in% c("t21", "t27", "t71", "t51"))
otu_mat<- as.data.frame(t(bacteria$reads))
tax_mat<- bacteria$motus %>% dplyr::select(Phylum, Class, Order, Family, Genus)
samples_df <-  bacteria$samples %>% dplyr::select(Treatment, Time, Species, Period, facet_label)
otu_mat <- as.matrix(otu_mat)
tax_mat <- as.matrix(tax_mat)
OTU = otu_table(otu_mat, taxa_are_rows = TRUE)
TAX = tax_table(tax_mat)
samples = sample_data(samples_df)
recovery_bacteria <- phyloseq(OTU, TAX, samples)
recovery_bacteria
  
#fungi
fungi$samples <-fungi$samples %>%
   mutate(Period = case_when(
     Treatment == "D1" &  Time == "t21" ~ "Drought D1",
      Treatment == "D1" &  Time == "t51" ~ "Recovery from D1",
      Treatment == "D2" &  Time == "t27" ~ "Drought D2",
      Treatment == "D2" &  Time == "t57" ~ "Recovery from D2",
      Treatment == "D3" &  Time == "t71" ~ "Drought D3",
      Treatment == "D3" &  Time == "t101" ~ "Recovery from D3",
      TRUE                      ~ "Control"
    )
  )
fungi$samples$facet_label <- NA
fungi$samples$facet_label[fungi$samples$Species == "Epe.fal"] <- "E. falcata"
fungi$samples$facet_label[fungi$samples$Species == "Iry.hos"] <- "I. hostmanii"
fungi$samples$facet_label[fungi$samples$Species == "Jac.cop"] <- "J. copaia"
fungi$samples$facet_label[fungi$samples$Species == "Pte.off"] <- "P. officinalis"
fungi$samples$facet_label[fungi$samples$Species == "Sym.off"] <- "S. globulifera"
fungi$samples$facet_label[fungi$samples$Species == "Tac.mel"] <- "T. melinonii"
fungi$samples$facet_label[fungi$samples$Species == "Vir.sur"] <- "V. surinamensis"
# fungi <- subset_metabarlist(metabarlist = fungi, table = "samples", indices = fungi$samples$Time %in% c("t21", "t27", "t71"))
otu_mat<- as.data.frame(t(fungi$reads))
tax_mat<- fungi$motus %>% dplyr::select(Phylum, Class, Order, Family, Genus)
samples_df <-  fungi$samples %>% dplyr::select(Treatment, Time, Species, Period, facet_label)
otu_mat <- as.matrix(otu_mat)
tax_mat <- as.matrix(tax_mat)
OTU = otu_table(otu_mat, taxa_are_rows = TRUE)
TAX = tax_table(tax_mat)
samples = sample_data(samples_df)
recovery_fungi <- phyloseq(OTU, TAX, samples)
recovery_fungi
```

####prepare data - transformation
```{r}
#Normalize number of reads in each sample using median sequencing depth

#bacteria
total = median(sample_sums(recovery_bacteria))#2911
standf = function(x, t=total) round(t * (x / sum(x)))
recovery_bacteria = transform_sample_counts(recovery_bacteria, standf)

#fungi
total = median(sample_sums(recovery_fungi))#8390
standf = function(x, t=total) round(t * (x / sum(x)))
recovery_fungi = transform_sample_counts(recovery_fungi, standf)
```

### alpha diversity
```{r}
R_bacteria <- estimate_richness(recovery_bacteria, split= TRUE, measures = NULL)
plot_richness(recovery_bacteria, x="Period", measures="Shannon", color = "Period")+  geom_boxplot(alpha=0.6)+ 
  theme(legend.position="none", axis.text.x=element_text(angle=60,hjust=1,vjust=1,size=12))+ ggtitle("A. Bacteria")+ theme_minimal()+theme(axis.text.x = element_text(angle= 45), axis.title.x = element_blank(), legend.position = "none")#+ facet_wrap(~Species)

#my comparison
my_comparisons <- list( c("Control", "Drought D1"), c("Control", "Drought D2"), c("Control", "Drought D3"), c("Recovery D1", "Control"), c("Recovery D2", "Control"), c("Recovery D3", "Control"), c("Recovery D1", "Drought D1"), c("Recovery D2", "Drought D2"), c("Recovery D3", "Drought D3"))
my_comparisons_ajusted <- list( c("Control", "Drought D2"), c("Recovery D1", "Control"), c("Recovery D2", "Control"), c("Recovery D1", "Drought D1"))
symnum.args = list(cutpoints = c(0, 0.0001, 0.001, 0.01, 0.05, 1), symbols = c("****", "***", "**", "*", "ns"))



Alpha_bacteria_stat <- Alpha_bacteria + stat_compare_means(method = "wilcox.test", comparisons = my_comparisons_ajusted, label = "p.signif", symnum.args = symnum.args, hide.ns = T)

hist(R_bacteria$Shannon, main="Shannon index", xlab="") #yes normally distributed
anova.sh = aov(R_bacteria$Shannon ~ sample_data(recovery_bacteria)$Period)
summary(anova.sh) #p-value; 0.00441 **

#with hill
Shannon_bac <-hill_taxa(t(recovery_bacteria@otu_table),q=1)
df <- Shannon_bac %>% 
  enframe() %>%
  rename(Shannon_bac = value, 
         sample_id = name)
recovery_bacteria@sam_data$sample_id <- rownames(recovery_bacteria@sam_data)
recovery_bacteria@sam_data<- left_join(recovery_bacteria@sam_data, df)
meta_bac<- left_join(df, bacteria$samples)

#plot
Alpha_bacteria <- meta_bac%>%
  rename(Shannon = Shannon_bac) %>%
ggplot() +
  aes(x = Period, y = Shannon, fill= Period) +
  theme_minimal() +
   scale_fill_manual(values = c( 
      `Drought D1` = "#8CB425", `Drought D2` = "#ECE320", `Drought D3` = "#DF5220", Control = "#219ebc")) +
  geom_boxplot(alpha=0.6)+ 
  geom_jitter(alpha=0.4)+
  theme(legend.position="none", axis.text.x=element_text(angle=60,hjust=1,vjust=1,size=12))+ ggtitle("A. Bacteria")+ theme_minimal(base_size = 14)+theme(axis.text.x = element_text(angle= 45), axis.title.x = element_blank(), legend.position = "none")#+

#my comparison
my_comparisons <- list( c("Control", "Drought D1"), c("Control", "Drought D2"), c("Control", "Drought D3"), c("Recovery D1", "Control"), c("Recovery D2", "Control"), c("Recovery D3", "Control"), c("Recovery D1", "Drought D1"), c("Recovery D2", "Drought D2"), c("Recovery D3", "Drought D3"))
my_comparisons_ajusted <- list( c("Control", "Drought D2"), c("Recovery D1", "Control"), c("Recovery D2", "Control"), c("Recovery D1", "Drought D1"))
symnum.args = list(cutpoints = c(0, 0.0001, 0.001, 0.01, 0.05, 1), symbols = c("****", "***", "**", "*", "ns"))


Alpha_bacteria_stat <- Alpha_bacteria + stat_compare_means(method = "wilcox.test", comparisons = my_comparisons_ajusted, label = "p.signif", symnum.args = symnum.args, hide.ns = T)

hist(Shannon_bac, main="Shannon index", xlab="") #yes normally distributed
anova.sh = aov(Shannon_bac ~ sample_data(recovery_bacteria)$Period)
summary(anova.sh) #p-value;  0.000558 ***

#fungi
R_fungi <- estimate_richness(recovery_fungi, split= TRUE, measures = NULL)
Alpha_fungi <- plot_richness(recovery_fungi, x="Period", measures="Shannon", color = "Period")+  geom_boxplot(alpha=0.6)+ 
  theme(legend.position="none", axis.text.x=element_text(angle=45,hjust=1,vjust=1,size=12))+ ggtitle("B. fungi")+ theme_minimal()+theme(axis.text.x = element_text(angle= 45), axis.title.x = element_blank(), legend.position = "none")

#my comparison
my_comparisons <- list( c("Control", "Drought D1"), c("Control", "Drought D2"), c("Control", "Drought D3"), c("Recovery D1", "Control"), c("Recovery D2", "Control"), c("Recovery D3", "Control"), c("Recovery D1", "Drought D1"), c("Recovery D2", "Drought D2"), c("Recovery D3", "Drought D3"))
my_comparisons_ajusted <- list( c("Control", "Drought D2"), c("Recovery D1", "Control"), c("Recovery D2", "Control"), c("Recovery D1", "Drought D1"))
symnum.args = list(cutpoints = c(0, 0.0001, 0.001, 0.01, 0.05, 1), symbols = c("****", "***", "**", "*", "ns"))

Alpha_fungi_stat <- Alpha_fungi+ stat_compare_means(method = "kruskal", hide.ns = T)  #nothing is significant

hist(R_fungi$Shannon, main="Shannon index", xlab="") #not normally distributed
kruskal.test(R_fungi$Shannon~ sample_data(recovery_fungi)$Period)#0.4604

#with hill
Shannon_fun <-hill_taxa(t(recovery_fungi@otu_table),q=1)
df <- Shannon_fun %>% 
  enframe() %>%
  rename(Shannon_fun = value, 
         sample_id = name)
recovery_fungi@sam_data$sample_id <- rownames(recovery_fungi@sam_data)
recovery_fungi@sam_data<- left_join(recovery_fungi@sam_data, df)
meta_fun<- left_join(df, fungi$samples)

Alpha_fungi <- meta_fun%>%
  rename(Shannon = Shannon_fun) %>%
ggplot() +
  aes(x = Period, y = Shannon, fill= Period) +
  theme_minimal() +
   scale_fill_manual(values = c( 
      `Drought D1` = "#8CB425", `Drought D2` = "#ECE320", `Drought D3` = "#DF5220", Control = "#219ebc")) +
  geom_boxplot(alpha=0.6)+ 
  geom_jitter(alpha=0.4)+
  theme(legend.position="none", axis.text.x=element_text(angle=60,hjust=1,vjust=1,size=12))+ ggtitle("B. Fungi")+ theme_minimal(base_size = 14)+theme(axis.text.x = element_text(angle= 45), axis.title.x = element_blank(), legend.position = "none")#+

Alpha_fungi_stat <- Alpha_fungi+ stat_compare_means(method = "kruskal", hide.ns = T)  #nothing is significant

hist(Shannon_fun, main="Shannon index", xlab="") #not normally distributed
kruskal.test(Shannon_fun~ sample_data(recovery_fungi)$Period)#0.4604 as well

ggarrange(Alpha_bacteria_stat, Alpha_fungi_stat)
```

####beta diversity bacteria
option 1

```{r}
#bacteria Recovery D1
bac<- subset_samples(recovery_bacteria, Time %in% c("t51","t57", "t101"))
relab_genera = transform_sample_counts(bac, function(x) x / sum(x) * 100) 
#or for relative abundance normalization microbiome::transform(asv.whole, transform = "compositional")
abrel_bray <- phyloseq::distance(relab_genera, method = "bray")
ord = ordinate(relab_genera, method="PCoA", distance = "bray")

bacteria_recovery_ordi <- plot_ordination(relab_genera, ord, color= "Period", shape="facet_label") + 
  geom_point(size=4) + scale_shape_manual("Species", values = c(0 ,2, 3, 7 ,8, 19, 15)) +  stat_ellipse(aes(group=Period)) +
 scale_color_manual("Treatment", values = c("#6DA0F8", "#8CB425",  "#ECE320", "#DF5220"))+
  theme_minimal()+ ggtitle("A. Bacteria") +theme(legend.position = "none")
  
bacteria_recovery_ordi

samples <- data.frame(sample_data(relab_genera))
adonis2(abrel_bray ~ Period, data = samples)#0.002 ** This result means that the adonis test is significant, therefore the null hypothesis in which the Control/Treatments have the same centroid was reject.
adonis2(abrel_bray ~ Species, data = samples)
pairwise.adonis2(abrel_bray ~ Period, data = samples)

#species
bacteria_recovery_ordi_species <-  plot_ordination(relab_genera, ord, color= "facet_label", shape="facet_label") + 
  geom_point(size=4) +  stat_ellipse(aes(group=facet_label)) + theme_minimal()+ ggtitle("C. Bacteria") +theme(legend.position = "bottom", legend.direction = "vertical")+ guides(fill=guide_legend(nrow=4))+theme(legend.title = element_blank())
bacteria_recovery_ordi_species

```


option 2
```{r}
#bacteria Recovery D1
bac_D1<- subset_samples(recovery_bacteria, Time %in% c("t21","t51"))
relab_genera = transform_sample_counts(bac_D1, function(x) x / sum(x) * 100) 
#or for relative abundance normalization microbiome::transform(asv.whole, transform = "compositional")
abrel_bray <- phyloseq::distance(relab_genera, method = "bray")
ord = ordinate(relab_genera, method="PCoA", distance = "bray")

bacteria_recovery_ordi_D1 <- plot_ordination(relab_genera, ord, color= "Period", shape="facet_label") + 
  geom_point(size=4) + scale_shape_manual("Species", values = c(0 ,2, 3, 7 ,8, 19, 15)) +  stat_ellipse(aes(group=Period)) +
 scale_color_manual("Period", values = c("#6DA0F8", "#8CB425",  "#343a40")) +
  theme_minimal()+ ggtitle("A. Bacteria") +theme(legend.position = "none")
  
bacteria_recovery_ordi_D1

samples <- data.frame(sample_data(relab_genera))
adonis2(abrel_bray ~ Period, data = samples)#0.002 ** This result means that the adonis test is significant, therefore the null hypothesis in which the Control/Treatments have the same centroid was reject.
adonis2(abrel_bray ~ Species, data = samples)
pairwise.adonis2(abrel_bray ~ Period, data = samples)

# $parent_call
# [1] "abrel_bray ~ Period , strata = Null , permutations 999"
# 
# $`Drought D1_vs_Control`
#          Df SumOfSqs      R2     F Pr(>F)
# Period    1   0.1230 0.01929 1.239  0.186
# Residual 63   6.2547 0.98071             
# Total    64   6.3778 1.00000             
# 
# $`Drought D1_vs_Recovery D1`
#          Df SumOfSqs      R2      F Pr(>F)   
# Period    1   0.3085 0.04702 2.9108  0.002 **
# Residual 59   6.2531 0.95298                 
# Total    60   6.5617 1.00000                 
# ---
# Signif. codes:  0 *** 0.001 ** 0.01 * 0.05 . 0.1   1
# 
# $`Control_vs_Recovery D1`
#          Df SumOfSqs      R2      F Pr(>F)  
# Period    1   0.2022 0.03316 1.9895  0.033 *
# Residual 58   5.8942 0.96684                
# Total    59   6.0964 1.00000                
# ---
# Signif. codes:  0 *** 0.001 ** 0.01 * 0.05 . 0.1   1
# 
# attr(,"class")
# [1] "pwadstrata" "list"   

#bacteria Recovery D2
bac_D2<- subset_samples(recovery_bacteria, Time %in% c("t27","t57"))
relab_genera = transform_sample_counts(bac_D2, function(x) x / sum(x) * 100) 
#or for relative abundance normalization microbiome::transform(asv.whole, transform = "compositional")
abrel_bray <- phyloseq::distance(relab_genera, method = "bray")
ord = ordinate(relab_genera, method="PCoA", distance = "bray")

bacteria_recovery_ordi_D2 <- plot_ordination(relab_genera, ord, color= "Period", shape="facet_label") + 
  geom_point(size=4) + scale_shape_manual("Species", values = c(0 ,2, 3, 7 ,8, 19, 15)) +  stat_ellipse(aes(group=Period)) +
 scale_color_manual("Period", values = c("#6DA0F8", "#ECE320",  "#343a40")) +
  theme_minimal()+ ggtitle("C.") +theme(legend.position = "none")
  
bacteria_recovery_ordi_D2

samples <- data.frame(sample_data(relab_genera))
adonis2(abrel_bray ~ Period, data = samples)#0.002 ** This result means that the adonis test is significant, therefore the null hypothesis in which the Control/Treatments have the same centroid was reject.
adonis2(abrel_bray ~ Species, data = samples)
pairwise.adonis2(abrel_bray ~ Period, data = samples)
# 
# $parent_call
# [1] "abrel_bray ~ Period , strata = Null , permutations 999"
# 
# $`Drought D2_vs_Control`
#          Df SumOfSqs      R2      F Pr(>F)  
# Period    1   0.1859 0.02805 1.9333  0.037 *
# Residual 67   6.4421 0.97195                
# Total    68   6.6280 1.00000                
# ---
# Signif. codes:  0 *** 0.001 ** 0.01 * 0.05 . 0.1   1
# 
# $`Drought D2_vs_Recovery D2`
#          Df SumOfSqs     R2      F Pr(>F)    
# Period    1   0.4456 0.0584 3.7834  0.001 ***
# Residual 61   7.1849 0.9416                  
# Total    62   7.6306 1.0000                  
# ---
# Signif. codes:  0 *** 0.001 ** 0.01 * 0.05 . 0.1   1
# 
# $`Control_vs_Recovery D2`
#          Df SumOfSqs      R2      F Pr(>F)  
# Period    1   0.1878 0.02297 1.6456  0.087 .
# Residual 70   7.9887 0.97703                
# Total    71   8.1765 1.00000                
# ---
# Signif. codes:  0 *** 0.001 ** 0.01 * 0.05 . 0.1   1
# 
# attr(,"class")
# [1] "pwadstrata" "list"      

#bacteria Recovery D3
bac_D3<- subset_samples(recovery_bacteria, Time %in% c("t71","t101"))
relab_genera = transform_sample_counts(bac_D3, function(x) x / sum(x) * 100) 
#or for relative abundance normalization microbiome::transform(asv.whole, transform = "compositional")
abrel_bray <- phyloseq::distance(relab_genera, method = "bray")
ord = ordinate(relab_genera, method="PCoA", distance = "bray")

bacteria_recovery_ordi_D3 <- plot_ordination(relab_genera, ord, color= "Period", shape="facet_label") + 
  geom_point(size=4) + scale_shape_manual("Species", values = c(0 ,2, 3, 7 ,8, 19, 15)) +  stat_ellipse(aes(group=Period)) +
 scale_color_manual("Period", values = c("#6DA0F8", "#DF5220",  "#343a40")) +
  theme_minimal()+ ggtitle("E.") +theme(legend.position = "none")
  
bacteria_recovery_ordi_D3

samples <- data.frame(sample_data(relab_genera))
adonis2(abrel_bray ~ Period, data = samples)#0.002 ** This result means that the adonis test is significant, therefore the null hypothesis in which the Control/Treatments have the same centroid was reject.
adonis2(abrel_bray ~ Species, data = samples)
pairwise.adonis2(abrel_bray ~ Period, data = samples)

$parent_call
[1] "abrel_bray ~ Period , strata = Null , permutations 999"

# $`Recovery D3_vs_Control`
#          Df SumOfSqs      R2      F Pr(>F)  
# Period    1   0.1955 0.03857 1.9657  0.053 .
# Residual 49   4.8745 0.96143                
# Total    50   5.0700 1.00000                
# ---
# Signif. codes:  0 *** 0.001 ** 0.01 * 0.05 . 0.1   1
# 
# $`Recovery D3_vs_Drought D3`
#          Df SumOfSqs      R2      F Pr(>F)    
# Period    1   0.6916 0.13324 6.3026  0.001 ***
# Residual 41   4.4988 0.86676                  
# Total    42   5.1903 1.00000                  
# ---
# Signif. codes:  0 *** 0.001 ** 0.01 * 0.05 . 0.1   1
# 
# $`Control_vs_Drought D3`
#          Df SumOfSqs      R2      F Pr(>F)  
# Period    1   0.3675 0.05728 2.6734  0.024 *
# Residual 44   6.0485 0.94272                
# Total    45   6.4160 1.00000                
# ---
# Signif. codes:  0 *** 0.001 ** 0.01 * 0.05 . 0.1   1
# 
# attr(,"class")
# [1] "pwadstrata" "list"  
```

###beta diverisity fungi
option 1

```{r}

fun_recov<- subset_samples(recovery_fungi, Time %in% c("t51","t57", "t101"))
relab_genera = transform_sample_counts(fun_recov, function(x) x / sum(x) * 100) 
#or for relative abundance normalization microbiome::transform(asv.whole, transform = "compositional")
abrel_bray <- phyloseq::distance(relab_genera, method = "bray")
ord = ordinate(relab_genera, method="PCoA", distance = "bray")

fungi_recovery_ordi <- plot_ordination(relab_genera, ord, color= "Period", shape="facet_label") + 
  geom_point(size=4) + scale_shape_manual("Species", values = c(0 ,2, 3, 7 ,8, 19, 15)) +  stat_ellipse(aes(group=Period)) +
 scale_color_manual("Treatment", values = c("#6DA0F8", "#8CB425",  "#ECE320", "#DF5220"))+
  theme_minimal()+ ggtitle("B. Fungi") +theme(legend.position = "none")
  
fungi_recovery_ordi

samples <- data.frame(sample_data(relab_genera))
adonis2(abrel_bray ~ Period, data = samples)#0.002 ** This result means that the adonis test is significant, therefore the null hypothesis in which the Control/Treatments have the same centroid was reject.
adonis2(abrel_bray ~ Species, data = samples)
pairwise.adonis2(abrel_bray ~ Period, data = samples)

#species
fungi_recovery_ordi_species <-  plot_ordination(relab_genera, ord, color= "facet_label", shape="facet_label") + 
  geom_point(size=4) +  stat_ellipse(aes(group=facet_label)) + theme_minimal()+ ggtitle("D. fungi") +theme(legend.position = "bottom", legend.direction = "vertical")+ guides(fill=guide_legend(nrow=4))+theme(legend.title = element_blank())
fungi_recovery_ordi_species
```

##beta plot recovery
```{r}
beta_recovery <- ggarrange(bacteria_recovery_ordi, fungi_recovery_ordi, nrow = 1, ncol = 2, common.legend = T, legend = "right")

beta_recovery_species <- ggarrange(bacteria_recovery_ordi_species, fungi_recovery_ordi_species,  nrow = 1, ncol = 2, common.legend = T, legend = "right")

beta_all_recov <- ggarrange(beta_recovery, beta_recovery_species, nrow= 2)

ggsave(filename = "./results/microbio/beta_recovery_all.jpeg", plot = beta_all_recov, width = 8,height = 8)
```

option2
```{r}
#fungi Recovery D1
fun_D1<- subset_samples(recovery_fungi, Time %in% c("t21","t51"))
relab_genera = transform_sample_counts(fun_D1, function(x) x / sum(x) * 100) 
#or for relative abundance normalization microbiome::transform(asv.whole, transform = "compositional")
abrel_bray <- phyloseq::distance(relab_genera, method = "bray")
ord = ordinate(relab_genera, method="PCoA", distance = "bray")

fungi_recovery_ordi_D1 <- plot_ordination(relab_genera, ord, color= "Period", shape="facet_label") + 
  geom_point(size=4) + scale_shape_manual("Species", values = c(0 ,2, 3, 7 ,8, 19, 15)) +  stat_ellipse(aes(group=Period)) +
 scale_color_manual("Treatment", values = c("#6DA0F8", "#8CB425",  "#343a40")) +
  theme_minimal()+ ggtitle("B. fungi") +theme(legend.direction = "vertical")+ guides(fill=guide_legend(nrow=4))+
   guides(shape="none")
  
fungi_recovery_ordi_D1

samples <- data.frame(sample_data(relab_genera))
adonis2(abrel_bray ~ Period, data = samples)#0.002 ** This result means that the adonis test is significant, therefore the null hypothesis in which the Control/Treatments have the same centroid was reject.
adonis2(abrel_bray ~ Species, data = samples)
pairwise.adonis2(abrel_bray ~ Period, data = samples)

#fungi Recovery D2
fun_D2<- subset_samples(recovery_fungi, Time %in% c("t27","t57"))
relab_genera = transform_sample_counts(fun_D2, function(x) x / sum(x) * 100) 
#or for relative abundance normalization microbiome::transform(asv.whole, transform = "compositional")
abrel_bray <- phyloseq::distance(relab_genera, method = "bray")
ord = ordinate(relab_genera, method="PCoA", distance = "bray")

fungi_recovery_ordi_D2 <- plot_ordination(relab_genera, ord, color= "Period", shape="facet_label") + 
  geom_point(size=4) + scale_shape_manual("Species", values = c(0 ,2, 3, 7 ,8, 19, 15)) +  stat_ellipse(aes(group=Period)) +
 scale_color_manual("Treatment", values = c("#6DA0F8", "#ECE320",  "#343a40")) +
  theme_minimal()+ ggtitle("D.") +theme(legend.direction = "vertical")+ guides(fill=guide_legend(nrow=4))+
   guides(shape="none")
  
fungi_recovery_ordi_D2

samples <- data.frame(sample_data(relab_genera))
adonis2(abrel_bray ~ Period, data = samples)#0.002 ** This result means that the adonis test is significant, therefore the null hypothesis in which the Control/Treatments have the same centroid was reject.
adonis2(abrel_bray ~ Species, data = samples)
pairwise.adonis2(abrel_bray ~ Period, data = samples)
    

#fungi Recovery D3
fun_D3<- subset_samples(recovery_fungi, Time %in% c("t71","t101"))
relab_genera = transform_sample_counts(fun_D3, function(x) x / sum(x) * 100) 
#or for relative abundance normalization microbiome::transform(asv.whole, transform = "compositional")
abrel_bray <- phyloseq::distance(relab_genera, method = "bray")
ord = ordinate(relab_genera, method="PCoA", distance = "bray")

fungi_recovery_ordi_D3 <- plot_ordination(relab_genera, ord, color= "Period", shape="facet_label") + 
  geom_point(size=4) + scale_shape_manual("Species", values = c(0 ,2, 3, 7 ,8, 19, 15)) +  stat_ellipse(aes(group=Period)) +
 scale_color_manual("Treatment", values = c("#6DA0F8", "#DF5220",  "#343a40")) +
  theme_minimal()+ ggtitle("F.")  +theme(legend.direction = "vertical")+ guides(fill=guide_legend(nrow=4))+
   guides(shape="none")

fungi_recovery_ordi_D3

samples <- data.frame(sample_data(relab_genera))
adonis2(abrel_bray ~ Period, data = samples)#0.002 ** This result means that the adonis test is significant, therefore the null hypothesis in which the Control/Treatments have the same centroid was reject.
adonis2(abrel_bray ~ Species, data = samples)
pairwise.adonis2(abrel_bray ~ Period, data = samples)


```

```{r}
#get legend
legend_getting <- plot_ordination(relab_genera, ord, shape="facet_label") + 
  geom_point(size=4) + scale_shape_manual("Host species", values = c(0 ,2, 3, 7 ,8, 19, 15)) +theme_minimal()+theme(legend.direction = "horizontal")+ guides(shape=guide_legend(nrow=3))
  
legend_shape<-get_legend(legend_getting)
as_ggplot(legend_shape)

beta_recovery <- ggarrange(bacteria_recovery_ordi_D1, fungi_recovery_ordi_D1, bacteria_recovery_ordi_D2, fungi_recovery_ordi_D2, bacteria_recovery_ordi_D3, fungi_recovery_ordi_D3, ncol=2, nrow=3, widths = c(0.8,1,0.8,1,0.8,1), heights = c(1,1,1,1,1,1))

beta_recovery_final <-ggarrange(beta_recovery, as_ggplot(legend_shape), nrow=2, ncol=1, heights = c(1, 0.1))

ggsave(filename = "./results/microbio/beta_recovery_pcoa.jpeg", plot = beta_recovery_final, width = 6,height = 8)
```

#brouillon

```{r}
#bacteria drought

#previsous steps 
drought_bacteria = transform_sample_counts(drought_bacteria, standf)

s_phylum <- tax_glom(drought_bacteria, taxrank = "Phylum", NArm = FALSE)

genusabundance <- s_phylum %>%
  tax_glom(taxrank = "Phylum") %>%
  psmelt() 
```

```{r}
library("phyloseq")
library("ggplot2")      # graphics
library("readxl")       # necessary to import the data from Excel file
library("dplyr")        # filter and reformat data frames
library("tibble")       # Needed for converting column to row names

bacteria$samples <-bacteria$samples %>%
   mutate(Period = case_when(
     Treatment == "D1" &  Time == "t21" ~ "Drought D1",
      Treatment == "D1" &  Time == "t51" ~ "Recovery D1",
      Treatment == "D2" &  Time == "t27" ~ "Drought D2",
      Treatment == "D2" &  Time == "t57" ~ "Recovery D2",
      Treatment == "D3" &  Time == "t71" ~ "Drought D3",
      Treatment == "D3" &  Time == "t101" ~ "Recovery D3",
      TRUE                      ~ "Control"
    )
  )
bacteria <- subset_metabarlist(metabarlist = bacteria, table = "samples", indices = bacteria$samples$Time %in% c("t21", "t27", "t71"))

otu_mat<- as.data.frame(t(bacteria$reads))

tax_mat<- bacteria$motus %>% dplyr::select(Phylum, Class, Order, Family, Genus)

samples_df <-  bacteria$samples %>% dplyr::select(Treatment, Time, Species, Period)

otu_mat <- as.matrix(otu_mat)
tax_mat <- as.matrix(tax_mat)

OTU = otu_table(otu_mat, taxa_are_rows = TRUE)
  TAX = tax_table(tax_mat)
  samples = sample_data(samples_df)
  
  drought <- phyloseq(OTU, TAX, samples)
  drought

#Normalize number of reads in each sample using median sequencing depth :2912
total = median(sample_sums(drought))
standf = function(x, t=total) round(t * (x / sum(x)))
drought = transform_sample_counts(drought, standf) 

# #
# set.seed(1)
# ps.rarefied = rarefy_even_depth(drought, rngseed=1, sample.size=0.9*min(sample_sums(drought)), replace=F)

#calculta hill
OTU_norm <-drought@otu_table

Shannon_bac <-hill_taxa(t(OTU_norm),q=1)

df <- Shannon_bac %>% 
  enframe() %>%
  rename(Shannon_bac = value, 
         sample_id = name)

data <- drought@sam_data
data$sample_id <- rownames(data)

data<- as.data.frame(data)
df <- as.data.frame(df)

bacteria <- left_join(df,data, by="sample_id")


drought_fraction <- merge_samples(drought, c("Treatment"))
plot_bar(drought_fraction, fill = "Phylum") + 
  geom_bar(aes(color=Phylum, fill=Phylum), stat="identity", position="stack")

drought_abund <- filter_taxa(drought, function(x) sum(x > total*0.20) > 0, TRUE)
  drought_abund
 plot_heatmap(drought_abund, method = "PCoA", distance = "bray",  taxa.label = "Class", taxa.order = "Class", 
               trans=NULL, low="beige", high="red", na.value="beige", sample.label = "Treatment", sample.order = "Treatment")
 
 plot_richness(ps.rarefied, measures=c("Chao1", "Shannon"), x="Period")

 dist = phyloseq::distance(ps.rarefied, method="bray", weighted=F)
 ordination = ordinate(ps.rarefied, method="NMDS", distance=dist)

plot_ordination(ps.rarefied, ordination, type="samples", color="Species", title="OTUs")
   #facet_wrap(~Species, 1) + 
  
ps_tr <- microbiome::transform(drought, "clr")

ps_dist_matrix <- phyloseq::distance(drought, method ="bray")

vegan::adonis2(ps_dist_matrix ~ phyloseq::sample_data(drought)$Period* phyloseq::sample_data(drought)$Species)

pairwise.adonis(ps_dist_matrix, phyloseq::sample_data(drought)$Period+phyloseq::sample_data(drought)$Specie)
 

    
```

```{r}
#PCOA phyloseq

library("phyloseq")
library("ggplot2")      # graphics
library("readxl")       # necessary to import the data from Excel file
library("dplyr")        # filter and reformat data frames
library("tibble")       # Needed for converting column to row names

data_ITS$samples <-data_ITS$samples %>%
   mutate(Period = case_when(
     Treatment == "D1" &  Time == "t21" ~ "Drought D1",
      Treatment == "D1" &  Time == "t51" ~ "Recovery D1",
      Treatment == "D2" &  Time == "t27" ~ "Drought D2",
      Treatment == "D2" &  Time == "t57" ~ "Recovery D2",
      Treatment == "D3" &  Time == "t71" ~ "Drought D3",
      Treatment == "D3" &  Time == "t101" ~ "Recovery D3",
      TRUE                      ~ "Control"
    )
  )

otu_mat<- as.data.frame(t(data_ITS$reads))

tax_mat<- data_ITS$motus %>% dplyr::select(Phylum, Class, Order, Family, Genus)

samples_df <-  data_ITS$samples %>% dplyr::select(Treatment, Time, Species, Period)

otu_mat <- as.matrix(otu_mat)
tax_mat <- as.matrix(tax_mat)

OTU = otu_table(otu_mat, taxa_are_rows = TRUE)
  TAX = tax_table(tax_mat)
  samples = sample_data(samples_df)
  
  drought <- phyloseq(OTU, TAX, samples)
  drought

#Normalize number of reads in each sample using median sequencing depth :2912
total = median(sample_sums(drought))
standf = function(x, t=total) round(t * (x / sum(x)))
drought = transform_sample_counts(drought, standf) 

 plot_richness(drought, measures=c("Chao1", "Shannon"), x="Period") + geom_boxplot()

 dist = phyloseq::distance(drought, method="bray", weighted=F)
 ordination = ordinate(drought, method="PCoA", distance=dist)

 
plot_ordination(drought, ordination, type="samples", color="Period", title="OTUs") +  facet_wrap(~Species, 1) +  stat_ellipse(aes(color = Period, group = Period)) 
  
   
    
adonis2(dist ~ sample_data(ps.rarefied)$Period)


#
set.seed(1)
ps.rarefied = rarefy_even_depth(drought, rngseed=1, sample.size=0.9*min(sample_sums(drought)), replace=F)

#calculta hill
OTU_norm <-drought@otu_table

Shannon_bac <-hill_taxa(t(OTU_norm),q=1)

df <- Shannon_bac %>% 
  enframe() %>%
  rename(Shannon_bac = value, 
         sample_id = name)

data <- drought@sam_data
data$sample_id <- rownames(data)

data<- as.data.frame(data)
df <- as.data.frame(df)

data_ITS <- left_join(df,data, by="sample_id")


drought_fraction <- merge_samples(drought, c("Treatment"))
plot_bar(drought_fraction, fill = "Phylum") + 
  geom_bar(aes(color=Phylum, fill=Phylum), stat="identity", position="stack")

drought_abund <- filter_taxa(drought, function(x) sum(x > total*0.20) > 0, TRUE)
  drought_abund
 plot_heatmap(drought_abund, method = "PCoA", distance = "bray",  taxa.label = "Class", taxa.order = "Class", 
               trans=NULL, low="beige", high="red", na.value="beige", sample.label = "Treatment", sample.order = "Treatment")
 
 plot_richness(ps.rarefied, measures=c("Chao1", "Shannon"), x="Period")

 dist = phyloseq::distance(ps.rarefied, method="bray", weighted=F)
 ordination = ordinate(ps.rarefied, method="NMDS", distance=dist)

plot_ordination(ps.rarefied, ordination, type="samples", color="Species", title="OTUs")
   #facet_wrap(~Species, 1) + 
  
   
    
adonis2(dist ~ sample_data(ps.rarefied)$Period)
 

    
```

#PCOA not phyloseq drought
```{r}
load("/DRYER_greenhouse/results/Metabarlist_serre_16S_AMVG51-230505_clean.Rdata")
bacteria <- AMVG51_230505_16S_clean
rownames(bacteria$samples) <- bacteria$samples$sample_id
bacteria$samples <-bacteria$samples %>%
   mutate(Period = case_when(
     Treatment == "D1" &  Time == "t21" ~ "Drought D1",
      Treatment == "D1" &  Time == "t51" ~ "Recovery D1",
      Treatment == "D2" &  Time == "t27" ~ "Drought D2",
      Treatment == "D2" &  Time == "t57" ~ "Recovery D2",
      Treatment == "D3" &  Time == "t71" ~ "Drought D3",
      Treatment == "D3" &  Time == "t101" ~ "Recovery D3",
      TRUE                      ~ "Control"
    )
  )

drought <- subset_metabarlist(metabarlist = bacteria, table = "samples", indices = bacteria$samples$Time %in% c("t21", "t27", "t71"))

drought$samples$Period <- as.factor(drought$samples$Period)

hill <- hill_taxa_parti_pairwise(drought$reads,q=1,output="matrix", pairs = "full") 
hill.beta  <- hill$TD_beta
hill.beta.dist <- as.dist(hill.beta)


hill.beta.dist <- hill.beta.dist %>%
  as.matrix() %>%
  as_tibble(rownames = "sample_id") %>%
  pivot_longer(-sample_id) %>%
  filter(sample_id < name) %>% #don't repeat the same compared to self
  mutate(indv_a = str_replace(sample_id, ".*__", "")) %>%
  mutate(indv_b = str_replace(name, ".*__", ""))
 
info <- drought$samples %>%
  dplyr::select(sample_id, Treatment, Time, Species, code, Period)

dim(info) #132*5
dim(hill.beta.dist) #37950     5

dist_merged <- merge(hill.beta.dist, info, by.x = "indv_a", by.y = "code", all.x = TRUE) %>%
  rename(indv_a_species = Species) %>%
  rename(indv_a_time = Time) %>%
  rename(indv_a_treatment = Treatment) %>%
  dplyr::select(-sample_id.x, -sample_id.y)

dist_merged2 <- merge(dist_merged, info, by.x = "indv_b", by.y = "code", all.x = TRUE) %>%
  rename(indv_b_species = Species) %>%
  rename(indv_b_time = Time) %>%
  rename(indv_b_treatment = Treatment) %>%
  dplyr::select(-sample_id, -name)

beta<- dist_merged2 %>% 
  dplyr::filter(indv_b_species == indv_a_species) %>% #compare within same species
  filter(indv_b_time == indv_a_time) %>%#compare with the same time period
  filter(indv_b_treatment ==indv_a_treatment )

ef_beta <- beta %>% filter(indv_a_species== "Epe.fal")
ih_beta <- beta %>% filter(indv_a_species== "Iry.hos")
jc_beta <- beta %>% filter(indv_a_species== "Jac.cop")
po_beta <- beta %>% filter(indv_a_species== "Pte.off")
sg_beta <- beta %>% filter(indv_a_species== "Sym.off")
tm_beta <- beta %>% filter(indv_a_species== "Tac.mel")
vs_beta <- beta %>% filter(indv_a_species== "Vir.sur")


##permanova per species
ef.pmanova <- adonis2(ef_beta$value ~ ef_beta$indv_a_treatment,  data = ef_beta)
ef.pmanova

pairwise.adonis(ef_beta$value ~ ef_beta$indv_a_treatment) #does not work it has to be matrix

ih.pmanova <- adonis2(ih_beta$value ~ ih_beta$indv_a_treatment,  data = ih_beta)
ih.pmanova

jc.pmanova <- adonis2(jc_beta$value ~ jc_beta$indv_a_treatment,  data = jc_beta)
jc.pmanova

po.pmanova <- adonis2(po_beta$value ~ po_beta$indv_a_treatment,  data = po_beta)
po.pmanova

sg.pmanova <- adonis2(sg_beta$value ~ sg_beta$indv_a_treatment,  data = sg_beta)
sg.pmanova

tm.pmanova <- adonis2(tm_beta$value ~ tm_beta$indv_a_treatment,  data = tm_beta)
tm.pmanova

vs.pmanova <- adonis2(vs_beta$value ~ vs_beta$indv_a_treatment,  data = vs_beta)
vs.pmanova

tukey_hsd(vs.pmanova)
```














