---
title: "metabaR_16S_serre"
author: "Marion Boisseaux"
date: "2023-04-13"
output: html_document
---


```{r setup, include=FALSE}
#Package nécessaire
# install bioconductor dependencies
#install.packages("BiocManager")
#BiocManager::install("biomformat")

# install metabaR package
#install.packages("remotes")
#remotes::install_github("metabaRfactory/metabaR")

library(ape) #read fasta files
library(ggplot2)
library(ggpubr)
library(reshape2)
library(metabaR)
library(readr)
library(dplyr)
library(tidyverse)
library(tidyr)
library(stringr)
library(dada2)
```

#MetabarList

Creation de la MetarbarList avec les 4 objets:

*Motus : Motus en ligne et leur attributs en colonne (séquence, assignation, score d’assignation, etc)*
*PCR : PCRs en ligne et leur attributs en colonne (sample_id, type, type_de_contrôle)*
*Reads : une table avec les séquences en ligne et les Motus (abondance) en colonne*
*Sample : les échantillons en ligne et leurs infos associer en colonne (position GPS, altitude, milieu de prélèvement, etc). Mais si non disponible la liste des échantillons suffit*


```{r MetabarList, include=FALSE}
#samples file preparation for the conversion
file_samples <- read.delim2("D:/DRYER_greenhouse_data/DRYER_serre_16S_ngsfilter.tab", header = FALSE)
file_samples <-as.data.frame(sapply(file_samples, function(x) gsub("\"", "", x)))
samples  <- file_samples %>%
  #select(project, organ, marker, samples_name, Family, Genus, Species) %>%
  rename(project = V1) %>%
  rename(sample_id = V2) %>%
  rename(tag = V3) %>%
  rename(primer_forward = V4) %>%
  rename(primer_reverse = V5) %>%
  #rename(info = V6) %>%
  relocate(sample_id, .before = project) 

samples$type <- ifelse(grepl("PCR|CTA|EXT", samples$sample_id), "control", "sample")


samples <- samples %>% separate(sample_id, into = c("project", "s_project", "organ", "marker"), sep = "_", remove = FALSE)

samples <- samples %>% separate(sample_id, into = c("X", "code"), sep = "__", remove = FALSE) %>% dplyr::select(-X)

samples$code <- gsub("[^0-9]", "", samples$code) #keep only numbers, remove letters
rownames(samples) <- samples$sample_id

#write.table(samples, file = "E:/DRYER_greenhouse/resources/16Sfilesamples.txt", sep ="\t")

#pcrs
ngsfilter <- read.delim2("D:/DRYER_greenhouse/resources/16Sfilesamples.txt", header = TRUE)
ngsfilter <-as.data.frame(sapply(ngsfilter, function(x) gsub("\"", "", x)))
pcrs <- ngsfilter
pcrs <- pcrs[!duplicated(pcrs$sample_id),]


pcrs <- pcrs %>% 
   mutate(control_type = ifelse(str_detect(type, 'sample'), NA,
                               ifelse(str_detect(sample_id, 'CTAB'), 'sequencing',  #i put sequencing but it's actually sterilisation. just to make sure the metabarlist works
                                      ifelse(str_detect(sample_id, 'EXT'), 'extraction', 
                                             ifelse(str_detect(sample_id, 'PCR'), 'pcr', NA))))) %>%
  dplyr::select(sample_id, type, control_type)
rownames(pcrs) <- pcrs$sample_id

#obifile (add the 2 16S runs)

#first obifile from pipeline AMVG-51_230505
file_obitab1 <- read.delim2("D:/DRYER_greenhouse_data/fasteris_16S/Post-pipeline_obitools/AMVG-51_230505/DRYER_serre_R1R2_good_demultiplexed_derepl_basicfilt_cl_agg.tab")
#colnames(file_obitab1)
colnames(file_obitab1) <- gsub("sample.", "", colnames(file_obitab1)) # remove the pattern 'sample.' from column names

file_obitab2 <- read.delim2("D:/DRYER_greenhouse_data/fasteris_16S/Post-pipeline_obitools/AMVG-51_230512/DRYER_serre_R1R2_good_demultiplexed_derepl_basicfilt_cl_agg.tab")
#colnames(file_obitab1)
colnames(file_obitab2) <- gsub("sample.", "", colnames(file_obitab2)) # remove the pattern 'sample.' from column names

#read1 -from run1
colnames(file_obitab1)
read1 <- file_obitab1 %>% dplyr::select(c(1,20:325))
colnames(read1)
colnames(read1) <- gsub(pattern = "DRYER_greenhouse_Leaf_16S__PCR.", replacement = "DRYER_greenhouse_Leaf_16S__PCR-", x = colnames(read1)) #keep only numbers, remove letters
read_t1 <- t(read1[,-1])
read_t1 <- as.data.frame(read_t1)
colnames(read_t1) <- read1$id
read_t1[is.na(read_t1)] <- 0 #Replace na values with 0 using is.na()
read_t1 <- as.matrix(read_t1)


#read2 -from run2
colnames(file_obitab2)
read2 <- file_obitab2 %>% dplyr::select(c(1,21:326))
colnames(read2)
colnames(read2) <- gsub(pattern = "DRYER_greenhouse_Leaf_16S__PCR.", replacement = "DRYER_greenhouse_Leaf_16S__PCR-", x = colnames(read2)) #keep only numbers, remove letters
read_t2 <- t(read2[,-1])
read_t2 <- as.data.frame(read_t2)
colnames(read_t2) <- read2$id
read_t2[is.na(read_t2)] <- 0 #Replace na values with 0 using is.na()
read_t2 <- as.matrix(read_t2)


#motus
motus <- file_obitab1[, c(1,19,346)] #take the sequence column
rownames(motus) <- motus$id

#first construct the metabarlist without assignation


summary(file_obitab1$seq_length) 

#dimensions
dim(read_t1) # 306 6591
dim(read_t2) #306 6495
dim(motus) # 6591    3
dim(assign_motus) #6591    9
dim(pcrs) #306   3
dim(samples)#306  10


#dimensions are not the same, so investigate missing controls
read_t_essai <- as.data.frame(read_t)
setdiff(rownames(pcrs),rownames(read_t)) #character(0)

#Metabarlist
MetabarList <- metabarlist_generator(reads = read_t1 , 
                                       motus = motus, #first metabarlist avec motus then once taxa assignation is done: with assign_motus
                                     pcrs = pcrs, 
                                     samples = samples)

#save the metabarlist for the taxa assignation to do on the cluster
save(MetabarList, file = "./results/Metabarlist_serre_16S_AMVG51-230505.Rdata")
load("./results/Metabarlist_serre_16S_AMVG51-230505.Rdata")

#rebuild metabarlist with assignation
#assign
assign <- read.csv("E:/DRYER_greenhouse_data/fasteris_16S/taxa.serre.AMVG51-230505.16S.csv", header = TRUE)
assign <- assign %>% rename(sequence= X)
assign$sequence <- tolower(assign$sequence) #convert in lowercase letters
assign_motus <- left_join(motus,assign, by= "sequence")
dim(assign_motus) #[1] 6591    9
dim(motus) #[1] 6591    3
dim(assign) #[1]3747   8
rownames(assign_motus) <- assign_motus$id

#Assigned Metabarlist
MetabarList <- metabarlist_generator(reads = read_t , 
                                       motus = assign_motus, #first metabarlist avec motus then once taxa assignation is done: with assign_motus
                                     pcrs = pcrs, 
                                     samples = samples)
```


# Diagnostic plots

Basic stats on the raw data

```{r}
# Compute the number of reads per pcr
MetabarList$pcrs$nb_reads <- rowSums(MetabarList$reads)
summary(MetabarList$pcrs$nb_reads) 
min(MetabarList$pcrs$nb_reads)
sum(MetabarList$pcrs$nb_reads) #
Nbr_reads <- MetabarList$pcrs

#total number of reads
sum(Nbr_reads$nb_reads) #

#number of reads for all samples, removing control
Nbr_samples <-Nbr_reads %>% filter(type != "control") 
sum(Nbr_samples$nb_reads) #

#removing reads below 1186 reads
Nbr_samples <- Nbr_samples %>% filter(nb_reads > 1000)
summary(Nbr_samples$nb_reads)

  
A <- ggplot(Nbr_reads, aes(x = nb_reads, fill = type))+
  geom_histogram() +
   theme_minimal() +
  theme(legend.position = "bottom")+
  theme(text = element_text(size = 12))
 

B <- ggplot(Nbr_samples, aes(x = nb_reads))+ #histogram on the samples after removing 
  geom_histogram() +
  theme_minimal() +
  scale_x_continuous(breaks = seq(0, 40000, by = 5000))+
    theme(text = element_text(size = 12))

ggarrange(A, B, labels = c("Bacteria (AMVG-51_230505) - nr_reads for all with controls" ,"nb_reads for samples"))

# Compute the number of motus per pcr
MetabarList$pcrs$nb_motus <- rowSums(MetabarList$reads>0)
summary(MetabarList$pcrs$nb_motus <10) #only 2 beneath 10.

length(MetabarList$motus$sequence)  #5361
length(unique(MetabarList$motus$sequence))
# 5361 #nombre d'OTUs unique.

# Create an input table (named check1) for ggplot of 3 columns: 
#  (i) control type 
#  (ii) a vector indicated whether it corresponds to nb_reads or nb_motus, 
#  (iii) the corresponding values.

check1 <- melt(MetabarList$pcrs[,c("control_type", "nb_reads", "nb_motus")]) 
check1$control_type[which(check1$control_type == "sequencing")] <- "sterilisation"
ggplot(data <- check1, aes(x=control_type, y=value, color=control_type)) + 
  geom_boxplot() + theme_bw() + 
  geom_jitter(alpha=0.2) + 
  scale_color_manual(values = c("brown", "red", "cyan4","pink"), na.value = "darkgrey") +
  facet_wrap(~variable, scales = "free_y") + 
  theme(axis.text.x = element_text(angle=45, h=1))+
  ggtitle("bacteria AMVG-51-230505")

# Using the nb_reads and nb_motus defined previously

ggplot(MetabarList$pcrs, aes(x=nb_reads, y=nb_motus, color = control_type)) + 
  geom_point() + theme_bw() + 
  scale_y_log10() + scale_x_log10() + 
  scale_color_manual(values = c("brown", "red", "cyan4","pink"), na.value = "darkgrey")



# Check results
summary_metabarlist(MetabarList)

```

$dataset_dimension
        n_row n_col
reads     306  6591
motus    6591     9
pcrs      306     5
samples   306    10

$dataset_statistics
        nb_reads nb_motus avg_reads sd_reads avg_motus sd_motus
pcrs     1501688     6591  4907.477 3977.974  209.8627 101.0158
samples  1490304     6584  5103.781 3966.089  216.6849  98.2849

# Rarefaction curves

```{r eval=FALSE, include=FALSE}
natura.raref = hill_rarefaction(MetabarList, nboot = 20, nsteps = 10)
head(natura.raref$hill_table)
gghill_rarefaction(natura.raref) + ggtitle("AMVG-51-230505")

# Define a vector containing the Material info for each pcrs
material <- MetabarList$samples$code[match(MetabarList$pcrs$sample_id,
                                                rownames(MetabarList$samples))]

```

# Detecting contaminants

Due to the tagjump bias, many genuine MOTUs that are abundant in samples can be detected in negative controls. Consequently, simply removing from the dataset any MOTU that occurs in negative controls is a **very bad idea.**

The function *contaslayer* relies on this assumption and detects MOTUs whose relative abundance across the whole dataset is highest in negative controls. Note however that this approach won’t be appropriate if the negative controls have been contaminated with biological samples. In this case,contaslayer should identify MOTUs that are dominants in samples.

The function contaslayer adds a new column in table motus indicating whether the MOTU is a genuine MOTU **TRUE** or a contaminant *FALSE*.

`si ca bug après, c'est parce que certains echantillons ne sont pas presents partout, bien faire setdiff et verifier les dimensions des 4 sous-objet de la metabarlist`

## sterilisation contaminants

```{r}
serre <- contaslayer(metabarlist = MetabarList,
                         control_types = "sequencing", #but sterilization
                         output_col = "not_a_sterilisation_conta")


table(serre$motus$not_a_sterilisation_conta)
# 
# FALSE  TRUE 
#      7  6584 

# Identify the most common contaminant
# get contaminant ids
conta_sterilisation <- serre$motus$id[which(serre[["motus"]][["not_a_sterilisation_conta"]] == FALSE)]
conta_sterilisation<- as.data.frame(conta_sterilisation)

conta_sterilisation <- conta_sterilisation %>%  rename(id = conta_sterilisation)

conta_sterilisation <- left_join(conta_sterilisation, serre$motus)

# Compute relative abundance of all pcr contaminants together
a <- data.frame(conta.relab.sterilisation = rowSums(serre$reads[,!serre$motus$not_a_sterilisation_conta]) /
                                    rowSums(serre$reads))
# Add information on control types
a$control_type <- serre$pcrs$control_type[match(rownames(a), rownames(serre$pcrs))]

ggplot(a, aes(x=control_type, y=conta.relab.sterilisation, color=control_type)) +
  geom_boxplot() + geom_jitter(alpha=0.5) +
  scale_color_manual(values = c("brown", "red", "cyan4","pink"), na.value = "darkgrey") +
  labs(x=NULL, y="Prop. Reads (log10)") +
  theme_bw() +
  scale_y_log10()

```


## pcrs contaminants
```{r}
serre <- contaslayer(serre,
                         control_types = "pcr",
                         output_col = "not_a_pcr_conta")

table(serre$motus$not_a_pcr_conta)
# FALSE  TRUE 
#    8  6583 

# Identify the most common contaminant
# get contaminant ids

conta_pcr <- serre$motus$id[which(serre[["motus"]][["not_a_pcr_conta"]] == FALSE)]
conta_pcr <- as.data.frame(conta_pcr)

conta_pcr <- conta_pcr %>% 
  rename(id = conta_pcr)

conta_pcr <- left_join(conta_pcr, serre$motus)

# Compute relative abundance of all pcr contaminants together
a$conta.relab.pcr <- rowSums(serre$reads[,!serre$motus$not_a_pcr_conta]) /
                                    rowSums(serre$reads)

ggplot(a, aes(x=control_type, y=conta.relab.pcr, color=control_type)) +
  geom_boxplot() + geom_jitter(alpha=0.5) +
  scale_color_manual(values = c("brown", "red", "cyan4","pink"), na.value = "darkgrey") +
  labs(x=NULL, y="Prop. Reads (log10)") +
  theme_bw() +
  scale_y_log10()

```


## extraction contaminants

```{r extraction contaminants}

serre <- contaslayer(serre,
                         control_types = "extraction",
                         output_col = "not_an_extraction_conta")

table(serre$motus$not_an_extraction_conta)
#> 
#FALSE  TRUE 
 # 1  6590  
conta_extraction <- serre$motus$id[which(serre[["motus"]][["not_an_extraction_conta"]] == FALSE)]
conta_extraction <- as.data.frame(conta_extraction)

conta_extraction <- conta_extraction %>% 
    rename(id = conta_extraction)

conta_extraction <- left_join(conta_extraction, serre$motus)

# Compute relative abundance of all contaminants together

b<- serre$reads[,!serre$motus$not_an_extraction_conta]
b<- as.data.frame(b)
b%>% mutate(btotal = b/ rowSums(serre$reads))
a$conta.relab.extraction <-b$b /rowSums(serre$reads)

ggplot(a, aes(x=control_type, y=conta.relab.extraction, color=control_type)) +
  geom_boxplot() + geom_jitter(alpha=0.5) +
  scale_color_manual(values = c("brown", "red", "cyan4","pink"), na.value = "darkgrey") +
  labs(x=NULL, y="Prop. Reads (log10)") +
  theme_bw() +
  scale_y_log10() #ca va pas trop de conta dans les echantillons

```
##flag contaminants

Overall, samples yield much less amounts of extraction contaminants than experimental negative controls. No pcrs corresponding to samples have > 10% of their reads corresponding to contaminants. We can still flag the controls:

```{r}
#flag pcrs with total contaminant relative abundance > 10% of reads
#1 for contaminants that came from sterilisation
#2 for contaminants that came from extraction
#3 for pcr

#to get the detail
serre$pcrs$low_contamination_level <-
  ifelse(a$conta.relab.sterilisation[match(rownames(serre$pcrs), rownames(a))]>1e-1,  1,
  ifelse(a$conta.relab.extraction[match(rownames(serre$pcrs), rownames(a))]>1e-1,  2,
  ifelse(a$conta.relab.pcr[match(rownames(serre$pcrs), rownames(a))]>1e-1, 3, 0)))

#to do it all at once
serre$pcrs$low_contamination_level <-
  ifelse(a$conta.relab.sterilisation[match(rownames(serre$pcrs), rownames(a))]>1e-1,  F, 
  ifelse(a$conta.relab.extraction[match(rownames(serre$pcrs), rownames(a))]>1e-1,  F,
  ifelse(a$conta.relab.pcr[match(rownames(serre$pcrs), rownames(a))]>1e-1, F, T)))

# Proportion of potentially functional (TRUE) vs. failed (FALSE) pcrs
# (controls included) based on this criterion
table(serre$pcrs$low_contamination_level) / nrow(serre$pcrs)

#all at once
#  FALSE        TRUE 
# 0.003267974 0.996732026 
```

## spurious or non-target MOTUs

Non-target sequences can be amplified if the primers are not specific enough. On the other hand, some highly degraded sequences can be produced throughout the data production process, such as primer dimers, or chimeras from multiple parents (hereafter referred to as spurious MOTUs). To detect these, one can use the information related to taxonomic assignments and associated similarity scores.

Since the dataset was obtained with primers that target 16S, other MOTUs should be excluded. At this stage of the analysis, we only flag MOTUs based on this criterion.

```{r}
#Flag MOTUs corresponding to target (TRUE) vs. non-target (FALSE) taxa
serre$motus$target_taxon <- grepl("Bacteria", serre$motus$Kingdom)

# Proportion of each of these over total number of MOTUs
table(serre$motus$target_taxon) / nrow(serre$motus)
#> 
#>    FALSE       TRUE 
# 0.01426187 0.98573813 

# Removing Ricketsiales-Mitonchondria and Chloroplasts (non-target host-derived DNA amplification)


serre$motus$target_taxon <- NA
serre$motus$target_taxon[which(serre$motus$Family == 'Mitochondria')] <- FALSE #0.03747535 
serre$motus$target_taxon[which(serre$motus$Order == 'Chloroplast')] <- FALSE #0.04582006
serre$motus$target_taxon[which(is.na(serre$motus$target_taxon) ==TRUE & serre$motus$Kingdom != 'Bacteria')] <- FALSE #0.04855105 
serre$motus$target_taxon[which(is.na(serre$motus$target_taxon) ==TRUE & serre$motus$Kingdom == 'Bacteria')] <- TRUE
serre$motus$target_taxon[which(is.na(serre$motus$target_taxon) ==TRUE & is.na(serre$motus$Kingdom) == TRUE)] <- FALSE

# Verify the updated values
table(serre$motus$target_taxon) / nrow(serre$motus)
# 
#      FALSE       TRUE 
# 0.06008193 0.93991807 

# Intersection with extraction contaminant flags (not contaminant = T)
table(serre$motus$target_taxon,
      serre$motus$not_an_extraction_conta)
#>        
  #   FALSE TRUE
  # FALSE     0   94
  # TRUE      1 6496

table(serre$motus$target_taxon,
      serre$motus$not_a_sterilisation_conta)

table(serre$motus$target_taxon,
      serre$motus$not_a_pcr_conta)
```

Next, we want to identify MOTUs whose sequence is too dissimilar from references. This filtering criterion relies on the assumption that current reference databases capture most of the diversity at broad taxonomic levels (i.e. already have for example at least one representative of each phyla). Considering this, MOTUs being too distant from reference databases are more likely to be a degraded sequence, especially if such MOTUs are relatively numerous and of low abundance. To assess this, one can use the distribution of MOTU similarity scores, weighted and unweighted by their relative abundance.

```{r}
#pipeline does not keep low scores
```

## detecting PCR outliers

A first way to identify failed PCRs is to flag them based on the pcr sequencing depth

```{r}
ggplot(serre$pcrs, aes(nb_reads, fill= type)) +
    geom_histogram(bins=40, color="grey") +
    geom_vline(xintercept = 1185, lty=2, color="orange") + # threshold
    scale_x_log10() +
    labs(x="# Reads (with all MOTUs and PCRs)",
        y="# PCRs") +
    theme_bw() +
    theme(panel.grid = element_blank())
#fortunately, samples yield high amount of reads.
# Flag pcrs with an acceptable sequencing depth (TRUE) or inacceptable one (FALSE)
serre$pcrs$seqdepth_ok <- ifelse(serre$pcrs$nb_reads < 1185, F, T)

# Proportion of each of these over total number of pcrs, control excluded
table(serre$pcrs$seqdepth_ok[serre$pcrs$type=="sample"]) /
  nrow(serre$pcrs[serre$pcrs$type=="sample",])

 

```

## lowering tag-jumps

Tag-jumps (*sequencing outputs with false combinations of used tags*). Unless these sequences can be identified and excluded from downstream analyses, tag jumps creating sequences with false, but already used tag combinations, can cause incorrect assignment of sequences to samples and artificially inflate diversity. They are frequency-dependent, i.e. abundant genuine MOTUs are more likely to be found in low abundance in samples were they are not supposed to be than rare genuine MOTUs. To reduce the amount of such false positives, the function tagjumpslayer considers each MOTU separately and corrects its abundance in pcrs (see tagjumpslayer help for more information on possible correction methods) when the MOTU relative abundance over the entire dataset is below a given threshold. Such data a curation strategy is similar to what has been proposed by Esling, Lejzerowicz, and Pawlowski (2015). Effect of this threshold can be evaluated by testing how this filtration procedure affects basic dataset characteristics (e.g. # MOTUs or reads) at different levels, as exemplified below.

```{r}
# Define a vector of thresholds to test
thresholds <- c(0,1e-4,1e-3, 1e-2, 3e-2, 5e-2)

# Run the tests and stores the results in a list
tests <- lapply(thresholds, function(x) tagjumpslayer(serre,x))
names(tests) <- paste("t_", thresholds, sep="")

# Format the data for ggplot with amount of reads at each threshold
tmp <- melt(as.matrix(do.call("rbind", lapply(tests, function(x) rowSums(x$reads)))))
colnames(tmp) <- c("threshold", "sample", "abundance")

# Add richness in MOTUs at each threshold
tmp$richness <-
  melt(as.matrix(do.call("rbind", lapply(tests, function(x) {
    rowSums(x$reads > 0)
  }))))$value

# Add control type information on pcrs and make data curation threshold numeric
tmp$controls <- serre$pcrs$control_type[match(tmp$sample, rownames(serre$pcrs))]
tmp$threshold <- as.numeric(gsub("t_", "", tmp$threshold))

# New table formatting for ggplot
tmp2 <- melt(tmp, id.vars=colnames(tmp)[-grep("abundance|richness", colnames(tmp))])

ggplot(tmp2, aes(x=as.factor(threshold), y=value)) +
  geom_boxplot(color="grey40") +
  geom_vline(xintercept = which(levels(as.factor(tmp2$threshold)) == "0.01"), col="orange", lty=2) +
  geom_jitter(aes(color=controls), width = 0.2, alpha=0.5) +
  scale_color_manual(values = c("brown", "red", "cyan4","pink"), na.value = "darkgrey") +
  facet_wrap(~variable+controls, scale="free_y", ncol=4) +
  theme_bw() +
  scale_y_log10() +
  labs(x="MOTU pcr : total abundance filtering threshold", y="# Reads/MOTUs") +
  theme(panel.grid = element_blank(),
        strip.background = element_blank(),
        axis.text.x = element_text(angle=40, h=1),
         legend.position = "none")
# A threshold of 0.01 leads to a drop in both the number of reads and of MOTUs in sequencing negative controls. This drop is also noticeable in terms of the number of MOTUs in pcrs obtained from other controls as compared to those obtained from samples. The former are expected to be void of environmental MOTUs, and tag-jumps should be more visible/important in these pcrs. Note that this procedure primarily affects MOTU diversity in pcrs, and poorly the number of reads in pcrs.
# 
# As for above, pcrs containing large amounts of MOTUs identified as potentially artifactual or where tag-jumps filtering strongly affects the number of reads in pcrs can be flagged as potentially failed
```

## summarizing the noise in the serre dataset

We can now get an overview of the amount of noise identified with the criteria used above, for both the number of MOTUs and their associated readcount.


```{r}
# Create a table of MOTUs quality criteria
# noise is identified as FALSE in serre, the "!" transforms it to TRUE
#motus.qual <- !serre$motus[,c("not_an_extraction_conta", "not_a_pcr_conta", "not_a_sterilisation_conta", "target_taxon")]
motus.qual <- !serre$motus[,c("not_an_extraction_conta", "not_a_pcr_conta", "not_a_sterilisation_conta")]

#colnames(motus.qual) <- c("extraction_conta", "pcr_conta", "sterilisation_conta", "untargeted_taxon")
colnames(motus.qual) <- c("extraction_conta", "pcr_conta", "sterilisation_conta")

# Proportion of MOTUs potentially artifactual (TRUE) based on the criteria used
prop.table(table(apply(motus.qual, 1, sum) > 0))

# Corresponding proportion of artifactual reads (TRUE)
#get the number of reads per motus
serre$motus$count <- colSums(serre$reads>0)
prop.table(xtabs(serre$motus$count~apply(motus.qual, 1, sum) > 0))

# Proportion of MOTUs and reads potentially artifactual for each criterion
apply(motus.qual, 2, sum) / nrow(motus.qual)
apply(motus.qual, 2, function(x) sum(serre$motus$count[x])/sum(serre$motus$count))

tmp.motus <-
  apply(sapply(1:ncol(motus.qual), function(x) {
    ifelse(motus.qual[,x]==T, colnames(motus.qual)[x], NA)}), 1, function(x) {
      paste(sort(unique(x)), collapse = "|")
      })
tmp.motus <- as.data.frame(gsub("^$", "not_artefactual", tmp.motus))
colnames(tmp.motus) <-  "artefact_type"

ggplot(tmp.motus, aes(x=1, fill=artefact_type)) +
    geom_bar() +  xlim(0, 2) +
    labs(fill="Artifact type") +
    coord_polar(theta="y") + theme_void() +
    scale_fill_brewer(palette = "Set3") +
    theme(legend.direction = "vertical") +
    ggtitle("Bacteria - MOTUs artefacts overview")

# 
# The above shows that MOTUs flagged as potentially artefactual account for ca. 10% of the dataset’s diversity and roughly the same in terms of readcount. Most of these artifact MOTUs are rare and correspond to sequences which are potentially highly degraded, with very low sequence similarity against the EMBL reference database. The most abundant artifacts MOTUs were identified as contaminants.
#let's do the same for pcrs
# Create a table of pcrs quality criteria
# noise is identified as FALSE in serre, the "!" transforms it to TRUE

pcrs.qual <- !serre$pcrs[,c("low_contamination_level", "seqdepth_ok")]
colnames(pcrs.qual) <- c("high_contamination_level", "low_seqdepth")


# Proportion of pcrs potentially artifactual (TRUE) based on the criteria used
# excluding controls
prop.table(table(apply(pcrs.qual[serre$pcrs$type=="sample",], 1, sum) > 0))

# Proportion of MOTUs and reads potentially artifactual for each criterion
apply(pcrs.qual[serre$pcrs$type=="sample",], 2, sum) / nrow(pcrs.qual[serre$pcrs$type=="sample",])

tmp.pcrs <-
  apply(sapply(1:ncol(pcrs.qual), function(x) {
    ifelse(pcrs.qual[serre$pcrs$type=="sample",x]==T,
           colnames(pcrs.qual)[x], NA)}), 1, function(x) {
      paste(sort(unique(x)), collapse = "|")
      })
tmp.pcrs <- as.data.frame(gsub("^$", "not_artefactual", tmp.pcrs))

colnames(tmp.pcrs) <- "artefact_type"

ggplot(tmp.pcrs, aes(x=1, fill=artefact_type)) +
    geom_bar() +  xlim(0, 2) +
    labs(fill="Artifact type") +
    coord_polar(theta="y") + theme_void() +
    scale_fill_brewer(palette = "Set3") +
    theme(legend.direction = "vertical") +
    ggtitle("Bacteria - PCR artefacts overview")
```


high_contamination_level             low_seqdepth 
              0.00000000               0.05650685 
              
              
# Data cleaning and aggregation

The final stage of the analysis consists in removing data based on the criteria defined above and aggregating pcrs to get rid of technical replicates. Here, the decision of which of these flagged criteria to use rests with the user, depending on how they feel the tagged errors impact on the characteristics of their dataset.

## removing spurious signal

First, we will remove suprious MOTUs, PCRs and adjusting read counts by removing tag-jumps . At this stage of the analysis, *controls are no longer necessary*, and so can also be removed from the dataset.

```{r}
# Use tag-jump corrected metabarlist with the threshold identified above
tmp <- tests[["t_0.01"]]

#for pcrs
# Subset on pcrs and keep only controls
tmp_clean <- subset_metabarlist(tmp, "pcrs",
                          indices = rowSums(tmp$pcrs[,c("low_contamination_level","seqdepth_ok")]) == 2 &
                                    tmp$pcrs$type == "sample")
summary_metabarlist(tmp_clean)


serre_clean <- subset_metabarlist(tmp_clean, "motus",
                          indices = rowSums(tmp_clean$motus[,c("not_an_extraction_conta", "target_taxon", "not_a_pcr_conta", "not_a_sterilisation_conta")]) == 4)

summary_metabarlist(serre_clean)

#check if previsous subsetting lead to NAs
if(sum(colSums(tmp_clean$reads)==0)>0){print("empty motus present")}
if(sum(rowSums(serre_clean$reads)==0)>0){print("empty pcrs present")}

serre_clean$motus$count = colSums(serre_clean$reads)
serre_clean$pcrs$nb_reads_postmetabaR = rowSums(serre_clean$reads)
serre_clean$pcrs$nb_motus_postmetabaR = rowSums(ifelse(serre_clean$reads>0, T, F))

serre_clean_final <- subset_metabarlist(serre_clean, "pcrs",
                          indices = serre_clean$pcrs[,c("nb_reads_postmetabaR")] > 1000)

check <- melt(serre_clean_final$pcrs[,c("nb_reads", "nb_reads_postmetabaR",
                               "nb_motus", "nb_motus_postmetabaR")])
check$type <- ifelse(grepl("motus", check$variable), "richness", "abundance")

ggplot(data = check, aes(x = variable, y = value)) +
  geom_boxplot( color = "darkgrey") +
  geom_jitter(alpha=0.1, color = "darkgrey") +
  theme_bw() +
  facet_wrap(~type, scales = "free", ncol = 5) +
  theme(axis.text.x = element_text(angle=45, h=1)) 

#The sequencing depth or the richness was not greatly affected by the trimming.
```



```{r}
AMVG51_230505_16S_clean <- serre_clean_final
```

# Add information on the plants
```{r}
Indv <- googlesheets4::read_sheet("https://docs.google.com/spreadsheets/d/149M2FuvhAEsOysrsdpfIEvobh7W2pM7e105BK0y58G8/edit#gid=0", sheet = "Individuals") %>% dplyr::select(Code, Treatment, Species)

Time <- googlesheets4::read_sheet("https://docs.google.com/spreadsheets/d/149M2FuvhAEsOysrsdpfIEvobh7W2pM7e105BK0y58G8/edit#gid=0", sheet = "Extraction_Leaves") %>% dplyr::select(Code, Time) %>% filter(Time != "t0")

plant_info <- left_join(Indv, Time) %>% drop_na(Time)

plant_info$Code <- as.character(plant_info$Code)

plant_info <- plant_info %>% rename(code = Code)

AMVG51_230505_16S_clean$samples <- left_join(AMVG51_230505_16S_clean$samples, plant_info, by= "code")
```

##save
```{r}
save(AMVG51_230505_16S_clean, file = "./results/Metabarlist_serre_16S_AMVG51-230505_clean.Rdata")

load("./results/Metabarlist_serre_16S_AMVG51-230505_clean.Rdata")

rownames(AMVG51_230505_16S_clean$samples) <- AMVG51_230505_16S_clean$samples$sample_id

summary_metabarlist(AMVG51_230505_16S_clean)
```


#get final summary
```{r}

AMVG51_230505_16S_clean$samples$nb_reads <- rowSums(AMVG51_230505_16S_clean$reads) #get number of reads per sample
AMVG51_230505_16S_clean$samples$nb_motus <- rowSums(AMVG51_230505_16S_clean$reads>0) #get number of OTUs per sample

#to get number of motus and reads
leaf2 <- melt(AMVG51_230505_16S_clean$samples,id = c( "code", "Species", "Time", "Treatment", "nb_reads", "nb_motus")) %>% filter(variable =="sample_id")


total <- leaf2 %>%
  summarise(sum_nb_reads = sum(nb_reads))


leaf3 <- leaf2 %>%
  group_by(Species, Time, Treatment) %>%
  summarise(mean_nb_reads = round(mean(nb_reads)), 
            mean_nb_motus = round(mean(nb_motus)))

#to get number of samples (leaf) per species
leaf <- leaf2 %>%
  filter(variable == "sample_id") %>%
  group_by(Species, Time, Treatment) %>%
  summarise(n= n())

#table of number of reads, OTU and samples per species
table_leaf <- left_join(leaf3, leaf) 
table_leaf$organ <- "leaf"

write.csv(table_leaf,"./results/summary_16S_AMVG41_230505.csv")
table_leaf<-read.csv("./results/summary_16S_AMVG41_230505.csv") %>% dplyr::select(-X)

library(kableExtra)
table_leaf %>%
  dplyr::select(-organ) %>% #remove it otherwise redundant
  kbl(caption = "Curated data for AMVG41-16S-230505 : number of replicates, mean number of reads and OTUs per species.", escape = FALSE) %>% #title
  row_spec(0, bold = TRUE) %>% #first row is in bold
  column_spec(1, italic = T) %>% #make species name in italic
  kable_classic(full_width = F, html_font = "Cambria") %>%
  save_kable(file = "./results/summary_AMVG51_230505_16S.png", zoom = 10) #make it bigger add zoom = 5 in save_kable


```

##rarefaction curves

```{r}
leaf <- AMVG51_230505_16S_clean
rownames(leaf$samples) <- leaf$samples$sample_id
raref_bacteria = hill_rarefaction(leaf, nboot = 20, nsteps = 10)
head(raref_bacteria$hill_table)
gghill_rarefaction(raref_bacteria)


#with treatment
material <- leaf$samples$Treatment[match(leaf$pcrs$sample_id,                                rownames(leaf$samples))] 
material <- setNames(material,rownames(leaf$pcrs))
t <- gghill_rarefaction(raref_bacteria, group=material) 
rarefaction_leaf_bacteria<- t + ggtitle("A. Rarefaction curves - Leaf bacteria per treatment")+  theme(legend.text = element_text(size = 15, face = "italic"), legend.title = element_blank())

#per species
material <- leaf$samples$Species[match(leaf$pcrs$sample_id,                                rownames(leaf$samples))] 
material <- setNames(material,rownames(leaf$pcrs))s
p <- gghill_rarefaction(raref_bacteria, group=material) 
rarefaction_leaf_bacteria<- p + ggtitle("B. Rarefaction curves - Leaf bacteria per species")+  theme(legend.text = element_text(size = 15, face = "italic"), legend.title = element_blank())

#arrange
bacteria_rarefy <- ggarrange(t, p, nrow=2, ncol=1)
bacteria_rarefy

ggsave("rarefaction_bacteria.jpeg", plot = bacteria_rarefy, width = 10, height = 12)
```



